// @atlaskit/atlassian-navigation/dist/cjs/components/AtlassianNavigation/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.AtlassianNavigation = void 0;
var _react = require("@emotion/react");
var _analyticsNamespacedContext = require("@atlaskit/analytics-namespaced-context");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _colors = require("@atlaskit/theme/colors");
var _constants = require("../../common/constants");
var _theme = require("../../theme");
var _PrimaryItemsContainer = require("../PrimaryItemsContainer");
var _PrimaryItemsContainerV = require("../PrimaryItemsContainerV2");
/** @jsx jsx */

var containerStyles = (0, _react.css)({
  display: 'flex',
  boxSizing: 'border-box',
  height: _constants.HORIZONTAL_GLOBAL_NAV_HEIGHT,
  paddingRight: "var(--ds-space-150, 12px)",
  paddingLeft: "var(--ds-space-150, 12px)",
  position: 'relative',
  alignItems: 'center',
  justifyContent: 'space-between',
  flexShrink: 0,
  borderBottom: "1px solid ".concat("var(--ds-border, ".concat(_colors.N30, ")"))
});
var leftStyles = (0, _react.css)({
  display: 'flex',
  minWidth: 0,
  height: 'inherit',
  alignItems: 'center',
  flexGrow: 1,
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  '& > *': {
    flexShrink: 0
  }
});
var rightStyles = (0, _react.css)({
  display: 'flex',
  alignItems: 'center',
  flexShrink: 0,
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  '& > *': {
    marginRight: "var(--ds-space-050, 4px)",
    flexShrink: 0
  }
});
var analyticsData = {
  attributes: {
    navigationLayer: 'global'
  },
  componentName: 'atlassianNavigation'
};

/**
 * __Atlassian navigation__
 *
 * The horizontal navigation component for Atlassian products.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples)
 * - [Code](https://atlassian.design/components/atlassian-navigation/examples)
 */
var AtlassianNavigation = exports.AtlassianNavigation = function AtlassianNavigation(props) {
  var label = props.label,
    _props$primaryItems = props.primaryItems,
    primaryItems = _props$primaryItems === void 0 ? [] : _props$primaryItems,
    AppSwitcher = props.renderAppSwitcher,
    create = props.renderCreate,
    Help = props.renderHelp,
    Notifications = props.renderNotifications,
    ProductHome = props.renderProductHome,
    Profile = props.renderProfile,
    Search = props.renderSearch,
    SignIn = props.renderSignIn,
    Settings = props.renderSettings,
    _props$moreLabel = props.moreLabel,
    moreLabel = _props$moreLabel === void 0 ? '…' : _props$moreLabel,
    _props$theme = props.theme,
    theme = _props$theme === void 0 ? _theme.defaultTheme : _props$theme,
    testId = props.testId;
  return (0, _react.jsx)(_theme.ThemeProvider, {
    value: theme
  }, (0, _react.jsx)(_analyticsNamespacedContext.NavigationAnalyticsContext, {
    data: analyticsData
  }, (0, _react.jsx)("header", {
    style: theme.mode.navigation,
    css: containerStyles,
    "data-testid": testId && "".concat(testId, "-header"),
    role: "banner"
  }, (0, _react.jsx)("nav", {
    css: leftStyles,
    "aria-label": label
  }, AppSwitcher && (0, _react.jsx)(AppSwitcher, null), ProductHome && (0, _react.jsx)(ProductHome, null), (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.navigation-v2-no-jank_5yhbd') ? (0, _react.jsx)(_PrimaryItemsContainerV.PrimaryItemsContainer, {
    testId: testId,
    moreLabel: moreLabel,
    items: primaryItems,
    create: create
    // eslint-disable-next-line @repo/internal/react/no-unsafe-overrides
    ,
    theme: theme
  }) : (0, _react.jsx)(_PrimaryItemsContainer.PrimaryItemsContainer, {
    testId: testId,
    moreLabel: moreLabel,
    items: primaryItems,
    create: create
    // eslint-disable-next-line @repo/internal/react/no-unsafe-overrides
    ,
    theme: theme
  })), (0, _react.jsx)("div", {
    css: rightStyles,
    "data-testid": testId && "".concat(testId, "-secondary-actions")
  }, Search && (0, _react.jsx)(Search, null), Notifications && (0, _react.jsx)(Notifications, null), Help && (0, _react.jsx)(Help, null), Settings && (0, _react.jsx)(Settings, null), SignIn && (0, _react.jsx)(SignIn, null), Profile && (0, _react.jsx)(Profile, null)))));
};
var _default = exports.default = AtlassianNavigation;
