// @atlaskit/tabs/dist/cjs/components/tab-panel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _react2 = require("@emotion/react");
var _primitives = require("@atlaskit/primitives");
var _hooks = require("../hooks");
/** @jsx jsx */

// Note this is not being memoized as children is an unstable reference
/**
 * __TabPanel__
 *
 * A TabPanel houses the contents of a Tab.
 *
 * - [Examples](https://atlassian.design/components/tabs/examples)
 * - [Code](https://atlassian.design/components/tabs/code)
 * - [Usage](https://atlassian.design/components/tabs/usage)
 */
var TabPanel = function TabPanel(_ref) {
  var children = _ref.children,
    testId = _ref.testId;
  var _useTabPanel = (0, _hooks.useTabPanel)(),
    role = _useTabPanel.role,
    id = _useTabPanel.id,
    hidden = _useTabPanel.hidden,
    ariaLabelledBy = _useTabPanel['aria-labelledby'],
    onMouseDown = _useTabPanel.onMouseDown,
    tabIndex = _useTabPanel.tabIndex;
  return (0, _react2.jsx)(_primitives.Box, {
    testId: testId,
    role: role,
    id: id,
    hidden: hidden,
    "aria-labelledby": ariaLabelledBy,
    onMouseDown: onMouseDown,
    tabIndex: tabIndex
  }, (0, _react2.jsx)(_react.Fragment, null, children));
};
var _default = exports.default = TabPanel;
