// @atlaskit/tabs/dist/cjs/components/tab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Tab;
var _react = require("@emotion/react");
var _dsExplorations = require("@atlaskit/ds-explorations");
var _primitives = require("@atlaskit/primitives");
var _hooks = require("../hooks");
/** @jsx jsx */

/**
 * __Tab__
 *
 * Tab represents an indivudal Tab displayed in a TabList.
 *
 * - [Examples](https://atlassian.design/components/tabs/examples)
 * - [Code](https://atlassian.design/components/tabs/code)
 * - [Usage](https://atlassian.design/components/tabs/usage)
 */
function Tab(_ref) {
  var children = _ref.children,
    testId = _ref.testId;
  var _useTab = (0, _hooks.useTab)(),
    onClick = _useTab.onClick,
    id = _useTab.id,
    ariaControls = _useTab['aria-controls'],
    ariaPosinset = _useTab['aria-posinset'],
    ariaSelected = _useTab['aria-selected'],
    ariaSetsize = _useTab['aria-setsize'],
    onMouseDown = _useTab.onMouseDown,
    onKeyDown = _useTab.onKeyDown,
    role = _useTab.role,
    tabIndex = _useTab.tabIndex;
  return (0, _react.jsx)(_primitives.Box, {
    testId: testId,
    onClick: onClick,
    id: id,
    "aria-controls": ariaControls,
    "aria-posinset": ariaPosinset,
    "aria-selected": ariaSelected,
    "aria-setsize": ariaSetsize,
    onMouseDown: onMouseDown,
    onKeyDown: onKeyDown,
    role: role,
    tabIndex: tabIndex
  }, (0, _react.jsx)(_dsExplorations.UNSAFE_Text, {
    shouldTruncate: true,
    UNSAFE_style: {
      color: 'inherit'
    }
  }, children));
}
