// @atlaskit/page-layout/dist/cjs/components/slots/banner-slot.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _constants = require("../../common/constants");
var _utils = require("../../common/utils");
var _controllers = require("../../controllers");
var _slotFocusRing = _interopRequireDefault(require("./internal/slot-focus-ring"));
var _slotDimensions = _interopRequireDefault(require("./slot-dimensions"));
/** @jsx jsx */

var bannerStyles = (0, _react2.css)({
  height: _constants.BANNER_HEIGHT,
  gridArea: _constants.BANNER
});
var bannerFixedStyles = (0, _react2.css)({
  position: 'fixed',
  zIndex: 2,
  top: 0,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  right: _constants.RIGHT_PANEL_WIDTH,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: _constants.LEFT_PANEL_WIDTH
});

/**
 * __Banner__
 *
 * Provides a slot for a Banner within the PageLayout.
 *
 * - [Examples](https://atlassian.design/components/page-layout/examples)
 * - [Code](https://atlassian.design/components/page-layout/code)
 */
var Banner = function Banner(props) {
  var children = props.children,
    _props$height = props.height,
    height = _props$height === void 0 ? _constants.DEFAULT_BANNER_HEIGHT : _props$height,
    _props$isFixed = props.isFixed,
    isFixed = _props$isFixed === void 0 ? true : _props$isFixed,
    shouldPersistHeight = props.shouldPersistHeight,
    testId = props.testId,
    skipLinkTitle = props.skipLinkTitle,
    id = props.id;
  var bannerHeight = (0, _utils.resolveDimension)(_constants.VAR_BANNER_HEIGHT, height, shouldPersistHeight);
  (0, _react.useEffect)(function () {
    (0, _controllers.publishGridState)((0, _defineProperty2.default)({}, _constants.VAR_BANNER_HEIGHT, bannerHeight));
    return function () {
      (0, _controllers.publishGridState)((0, _defineProperty2.default)({}, _constants.VAR_BANNER_HEIGHT, 0));
    };
  }, [bannerHeight]);
  (0, _controllers.useSkipLink)(id, skipLinkTitle);
  return (0, _react2.jsx)(_slotFocusRing.default, null, function (_ref) {
    var className = _ref.className;
    return (0, _react2.jsx)("div", (0, _extends2.default)({
      css: [bannerStyles, isFixed && bannerFixedStyles],
      className: className,
      "data-testid": testId,
      id: id
    }, (0, _utils.getPageLayoutSlotSelector)('banner')), (0, _react2.jsx)(_slotDimensions.default, {
      variableName: _constants.VAR_BANNER_HEIGHT,
      value: bannerHeight
    }), children);
  });
};
var _default = exports.default = Banner;
