// @atlaskit/page-layout/dist/cjs/common/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VAR_TOP_NAVIGATION_HEIGHT = exports.VAR_RIGHT_SIDEBAR_WIDTH = exports.VAR_RIGHT_PANEL_WIDTH = exports.VAR_LEFT_SIDEBAR_WIDTH = exports.VAR_LEFT_SIDEBAR_FLYOUT = exports.VAR_LEFT_PANEL_WIDTH = exports.VAR_BANNER_HEIGHT = exports.TRANSITION_DURATION = exports.TOP_NAVIGATION_HEIGHT = exports.TOP_NAVIGATION = exports.RIGHT_SIDEBAR_WIDTH = exports.RIGHT_SIDEBAR = exports.RIGHT_PANEL_WIDTH = exports.RIGHT_PANEL = exports.RESIZE_CONTROL_SELECTOR = exports.RESIZE_BUTTON_SELECTOR = exports.PAGE_LAYOUT_SLOT_SELECTOR = exports.PAGE_LAYOUT_LS_KEY = exports.PAGE_LAYOUT_CONTAINER_SELECTOR = exports.MOBILE_COLLAPSED_LEFT_SIDEBAR_WIDTH = exports.MIN_LEFT_SIDEBAR_WIDTH = exports.MIN_LEFT_SIDEBAR_DRAG_THRESHOLD = exports.MAX_MOBILE_SIDEBAR_FLYOUT_WIDTH = exports.MAIN = exports.LEFT_SIDEBAR_WIDTH = exports.LEFT_SIDEBAR_FLYOUT_WIDTH = exports.LEFT_SIDEBAR_EXPANDED_WIDTH = exports.LEFT_SIDEBAR = exports.LEFT_PANEL_WIDTH = exports.LEFT_PANEL = exports.IS_SIDEBAR_DRAGGING = exports.IS_SIDEBAR_COLLAPSING = exports.IS_FLYOUT_OPEN = exports.GRAB_AREA_SELECTOR = exports.GRAB_AREA_LINE_SELECTOR = exports.FLYOUT_DELAY = exports.DIMENSIONS = exports.DEFAULT_TOP_NAVIGATION_HEIGHT = exports.DEFAULT_RIGHT_SIDEBAR_WIDTH = exports.DEFAULT_RIGHT_PANEL_WIDTH = exports.DEFAULT_LEFT_SIDEBAR_WIDTH = exports.DEFAULT_LEFT_SIDEBAR_FLYOUT_WIDTH = exports.DEFAULT_LEFT_PANEL_WIDTH = exports.DEFAULT_I18N_PROPS_SKIP_LINKS = exports.DEFAULT_BANNER_HEIGHT = exports.CONTENT = exports.COLLAPSED_LEFT_SIDEBAR_WIDTH = exports.BANNER_HEIGHT = exports.BANNER = void 0;
// dimension vars
var VAR_LEFT_PANEL_WIDTH = exports.VAR_LEFT_PANEL_WIDTH = 'leftPanelWidth';
var VAR_LEFT_SIDEBAR_WIDTH = exports.VAR_LEFT_SIDEBAR_WIDTH = 'leftSidebarWidth';
var VAR_RIGHT_SIDEBAR_WIDTH = exports.VAR_RIGHT_SIDEBAR_WIDTH = 'rightSidebarWidth';
var VAR_RIGHT_PANEL_WIDTH = exports.VAR_RIGHT_PANEL_WIDTH = 'rightPanelWidth';
var VAR_TOP_NAVIGATION_HEIGHT = exports.VAR_TOP_NAVIGATION_HEIGHT = 'topNavigationHeight';
var VAR_BANNER_HEIGHT = exports.VAR_BANNER_HEIGHT = 'bannerHeight';
var VAR_LEFT_SIDEBAR_FLYOUT = exports.VAR_LEFT_SIDEBAR_FLYOUT = 'leftSidebarFlyoutWidth';
var DIMENSIONS = exports.DIMENSIONS = [VAR_LEFT_PANEL_WIDTH, VAR_RIGHT_PANEL_WIDTH, VAR_BANNER_HEIGHT, VAR_TOP_NAVIGATION_HEIGHT, VAR_LEFT_SIDEBAR_WIDTH, VAR_RIGHT_SIDEBAR_WIDTH];

// Grid area names
var LEFT_PANEL = exports.LEFT_PANEL = 'left-panel';
var RIGHT_PANEL = exports.RIGHT_PANEL = 'right-panel';
var BANNER = exports.BANNER = 'banner';
var TOP_NAVIGATION = exports.TOP_NAVIGATION = 'top-navigation';
var CONTENT = exports.CONTENT = 'content';
var MAIN = exports.MAIN = 'main';
var LEFT_SIDEBAR = exports.LEFT_SIDEBAR = 'left-sidebar';
var RIGHT_SIDEBAR = exports.RIGHT_SIDEBAR = 'right-sidebar';

// Default slot dimension values
var DEFAULT_BANNER_HEIGHT = exports.DEFAULT_BANNER_HEIGHT = 56;
var DEFAULT_TOP_NAVIGATION_HEIGHT = exports.DEFAULT_TOP_NAVIGATION_HEIGHT = 56;
var DEFAULT_LEFT_SIDEBAR_WIDTH = exports.DEFAULT_LEFT_SIDEBAR_WIDTH = 240;
var DEFAULT_RIGHT_SIDEBAR_WIDTH = exports.DEFAULT_RIGHT_SIDEBAR_WIDTH = 280;
var DEFAULT_RIGHT_PANEL_WIDTH = exports.DEFAULT_RIGHT_PANEL_WIDTH = 368;
var DEFAULT_LEFT_PANEL_WIDTH = exports.DEFAULT_LEFT_PANEL_WIDTH = 368;

// Other constants
var COLLAPSED_LEFT_SIDEBAR_WIDTH = exports.COLLAPSED_LEFT_SIDEBAR_WIDTH = 20;
var MOBILE_COLLAPSED_LEFT_SIDEBAR_WIDTH = exports.MOBILE_COLLAPSED_LEFT_SIDEBAR_WIDTH = 16;
var MIN_LEFT_SIDEBAR_WIDTH = exports.MIN_LEFT_SIDEBAR_WIDTH = 80;
var DEFAULT_LEFT_SIDEBAR_FLYOUT_WIDTH = exports.DEFAULT_LEFT_SIDEBAR_FLYOUT_WIDTH = 240;
var MIN_LEFT_SIDEBAR_DRAG_THRESHOLD = exports.MIN_LEFT_SIDEBAR_DRAG_THRESHOLD = 200;
var MAX_MOBILE_SIDEBAR_FLYOUT_WIDTH = exports.MAX_MOBILE_SIDEBAR_FLYOUT_WIDTH = 350;
var TRANSITION_DURATION = exports.TRANSITION_DURATION = 300;
var FLYOUT_DELAY = exports.FLYOUT_DELAY = 200;
var LEFT_SIDEBAR_EXPANDED_WIDTH = exports.LEFT_SIDEBAR_EXPANDED_WIDTH = 'expandedLeftSidebarWidth';
var PAGE_LAYOUT_LS_KEY = exports.PAGE_LAYOUT_LS_KEY = 'DS_PAGE_LAYOUT_UI_STATE';

// Data attributes
var IS_SIDEBAR_DRAGGING = exports.IS_SIDEBAR_DRAGGING = 'data-is-sidebar-dragging';
var IS_SIDEBAR_COLLAPSING = exports.IS_SIDEBAR_COLLAPSING = 'data-is-sidebar-collapsing';
var IS_FLYOUT_OPEN = exports.IS_FLYOUT_OPEN = 'data-is-flyout-open';
var GRAB_AREA_LINE_SELECTOR = exports.GRAB_AREA_LINE_SELECTOR = 'data-grab-area-line';
var GRAB_AREA_SELECTOR = exports.GRAB_AREA_SELECTOR = 'data-grab-area';
var RESIZE_BUTTON_SELECTOR = exports.RESIZE_BUTTON_SELECTOR = 'data-resize-button';
var RESIZE_CONTROL_SELECTOR = exports.RESIZE_CONTROL_SELECTOR = 'data-resize-control';
var PAGE_LAYOUT_SLOT_SELECTOR = exports.PAGE_LAYOUT_SLOT_SELECTOR = 'data-ds--page-layout--slot';
var DEFAULT_I18N_PROPS_SKIP_LINKS = exports.DEFAULT_I18N_PROPS_SKIP_LINKS = 'Skip to:';
var PAGE_LAYOUT_CONTAINER_SELECTOR = exports.PAGE_LAYOUT_CONTAINER_SELECTOR = 'data-layout-container';
var LEFT_PANEL_WIDTH = exports.LEFT_PANEL_WIDTH = "var(--".concat(VAR_LEFT_PANEL_WIDTH, ", 0px)");
var RIGHT_PANEL_WIDTH = exports.RIGHT_PANEL_WIDTH = "var(--".concat(VAR_RIGHT_PANEL_WIDTH, ", 0px)");
var LEFT_SIDEBAR_WIDTH = exports.LEFT_SIDEBAR_WIDTH = "var(--".concat(VAR_LEFT_SIDEBAR_WIDTH, ", 0px)");
var RIGHT_SIDEBAR_WIDTH = exports.RIGHT_SIDEBAR_WIDTH = "var(--".concat(VAR_RIGHT_SIDEBAR_WIDTH, ", 0px)");
var TOP_NAVIGATION_HEIGHT = exports.TOP_NAVIGATION_HEIGHT = "var(--".concat(VAR_TOP_NAVIGATION_HEIGHT, ", 0px)");
var BANNER_HEIGHT = exports.BANNER_HEIGHT = "var(--".concat(VAR_BANNER_HEIGHT, ", 0px)");
var LEFT_SIDEBAR_FLYOUT_WIDTH = exports.LEFT_SIDEBAR_FLYOUT_WIDTH = "var(--".concat(VAR_LEFT_SIDEBAR_FLYOUT, ", ").concat(DEFAULT_LEFT_SIDEBAR_FLYOUT_WIDTH, "px)");
