// @atlaskit/page-layout/dist/cjs/common/utils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.resolveDimension = exports.removeFromGridStateInStorage = exports.mergeGridStateIntoStorage = exports.getPageLayoutSlotSelector = exports.getPageLayoutSlotCSSSelector = exports.getLeftSidebarPercentage = exports.getLeftPanelWidth = exports.getGridStateFromStorage = exports.emptyGridState = void 0;
var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _constants = require("./constants");
var _safeLocalStorage = _interopRequireDefault(require("./safe-local-storage"));
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var emptyGridState = exports.emptyGridState = _constants.DIMENSIONS.reduce(function (acc, currentValue) {
  return _objectSpread(_objectSpread({}, acc), {}, (0, _defineProperty2.default)({}, currentValue, 0));
}, {});
var mergeGridStateIntoStorage = exports.mergeGridStateIntoStorage = function mergeGridStateIntoStorage(key, value) {
  var storageValue = JSON.parse((0, _safeLocalStorage.default)().getItem(_constants.PAGE_LAYOUT_LS_KEY) || '{}');
  if (value !== null && (0, _typeof2.default)(value) === 'object') {
    storageValue[key] = _objectSpread(_objectSpread({}, storageValue[key]), value);
  } else {
    storageValue[key] = value;
  }
  (0, _safeLocalStorage.default)().setItem(_constants.PAGE_LAYOUT_LS_KEY, JSON.stringify(storageValue));
};
var getGridStateFromStorage = exports.getGridStateFromStorage = function getGridStateFromStorage(key) {
  var storageValue = JSON.parse((0, _safeLocalStorage.default)().getItem(_constants.PAGE_LAYOUT_LS_KEY) || '{}');
  return storageValue[key];
};
var removeFromGridStateInStorage = exports.removeFromGridStateInStorage = function removeFromGridStateInStorage(key, secondKey) {
  var storageValue = JSON.parse((0, _safeLocalStorage.default)().getItem(_constants.PAGE_LAYOUT_LS_KEY) || '{}');
  if (secondKey && storageValue[key]) {
    delete storageValue[key][secondKey];
  } else {
    delete storageValue[key];
  }
  (0, _safeLocalStorage.default)().setItem(_constants.PAGE_LAYOUT_LS_KEY, JSON.stringify(storageValue));
};
var resolveDimension = exports.resolveDimension = function resolveDimension(key) {
  var dimension = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
  var shouldPersist = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
  if (shouldPersist) {
    var cachedGridState = getGridStateFromStorage('gridState');
    return cachedGridState && Object.keys(cachedGridState).length > 0 && cachedGridState[key] ? cachedGridState[key] : dimension;
  }
  return dimension;
};
var getLeftPanelWidth = exports.getLeftPanelWidth = function getLeftPanelWidth() {
  if (typeof window === 'undefined') {
    return 0;
  }
  return parseInt(window.getComputedStyle(document.documentElement).getPropertyValue("--".concat(_constants.LEFT_PANEL_WIDTH)), 10) || 0;
};
var getLeftSidebarPercentage = exports.getLeftSidebarPercentage = function getLeftSidebarPercentage(currentWidth, maxWidth) {
  var total = (currentWidth - _constants.DEFAULT_LEFT_SIDEBAR_WIDTH) / (maxWidth - _constants.DEFAULT_LEFT_SIDEBAR_WIDTH);
  if (total < 0) {
    return 0;
  }
  if (total > 1) {
    return 100;
  }
  return Math.floor(total * 100);
};
var getPageLayoutSlotSelector = exports.getPageLayoutSlotSelector = function getPageLayoutSlotSelector(slotName) {
  return (0, _defineProperty2.default)({}, _constants.PAGE_LAYOUT_SLOT_SELECTOR, slotName);
};
var getPageLayoutSlotCSSSelector = exports.getPageLayoutSlotCSSSelector = function getPageLayoutSlotCSSSelector(slotName) {
  return "[".concat(_constants.PAGE_LAYOUT_SLOT_SELECTOR, "='").concat(slotName, "']");
};
