// @atlaskit/react-ufo/dist/esm/vc/vc-observer-new/window-event-observer/index.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import { bind } from 'bind-event-listener';
import { fg } from '@atlaskit/platform-feature-flags';
var WindowEventObserver = /*#__PURE__*/function () {
  function WindowEventObserver(opts) {
    _classCallCheck(this, WindowEventObserver);
    _defineProperty(this, "unbindFns", []);
    this.onEvent = opts.onEvent;
  }
  return _createClass(WindowEventObserver, [{
    key: "bindEvent",
    value: function bindEvent(type) {
      var _this = this;
      var unbindCallback = bind(window, {
        type: type,
        listener: function listener(event) {
          if (!event.isTrusted) {
            return;
          }
          _this.onEvent({
            time: event.timeStamp,
            type: type,
            event: event
          });
        },
        options: {
          passive: true,
          once: true
        }
      });
      this.unbindFns.push(unbindCallback);
    }

    /**
     * Binds a scroll listener on `document` using the capture phase.
     * The `scroll` event does not bubble, so listening on `window` only catches
     * document-level scrolls. By using `{ capture: true }` on `document`, we can
     * detect scroll events from any element in the DOM tree, including inner
     * scrollable containers (e.g. when the user drags a scrollbar).
     */
  }, {
    key: "bindCaptureScrollEvent",
    value: function bindCaptureScrollEvent() {
      var _this2 = this;
      var unbindCallback = bind(document, {
        type: 'scroll',
        listener: function listener(event) {
          if (!event.isTrusted) {
            return;
          }
          _this2.onEvent({
            time: event.timeStamp,
            type: 'scroll-container',
            event: event
          });
        },
        options: {
          capture: true,
          passive: true,
          once: true
        }
      });
      this.unbindFns.push(unbindCallback);
    }
  }, {
    key: "start",
    value: function start() {
      this.bindEvent('wheel');
      this.bindEvent('scroll');
      this.bindEvent('keydown');
      this.bindEvent('resize');
      if (fg('platform_ufo_detect_container_scroll')) {
        this.bindCaptureScrollEvent();
      }
    }
  }, {
    key: "stop",
    value: function stop() {
      this.unbindFns.forEach(function (cb) {
        cb();
      });
      this.unbindFns = [];
    }
  }]);
}();
export { WindowEventObserver as default };
