// @atlaskit/react-ufo/dist/es2019/create-interaction-extra-metrics-payload/index.js
import { fg } from '@atlaskit/platform-feature-flags';
import coinflip from '../coinflip';
import { DEFAULT_TTVC_REVISION, getConfig, getExtraInteractionRate, getMostRecentVCRevision } from '../config';
import { buildSegmentTree, getOldSegmentsLabelStack, optimizeLabelStack, sanitizeUfoName } from '../create-payload/common/utils';
import { getMoreAccuratePageVisibilityUpToTTAI } from '../create-payload/utils/get-more-accurate-page-visibility-up-to-ttai';
import getPageVisibilityUpToTTAI from '../create-payload/utils/get-page-visibility-up-to-ttai';
import getPayloadSize from '../create-payload/utils/get-payload-size';
import { getReactUFOPayloadVersion } from '../create-payload/utils/get-react-ufo-payload-version';
import getTTAI from '../create-payload/utils/get-ttai';
import getVCMetrics from '../create-payload/utils/get-vc-metrics';
import { optimizeApdex } from '../create-payload/utils/optimize-apdex';
import { optimizeCustomTimings } from '../create-payload/utils/optimize-custom-timings';
import { optimizeHoldInfo } from '../create-payload/utils/optimize-hold-info';
import { optimizeMarks } from '../create-payload/utils/optimize-marks';
import { optimizeReactProfilerTimings } from '../create-payload/utils/optimize-react-profiler-timings';
import { optimizeRequestInfo } from '../create-payload/utils/optimize-request-info';
import { optimizeSpans } from '../create-payload/utils/optimize-spans';
import { interactionSpans as atlaskitInteractionSpans } from '../interaction-metrics';
async function createInteractionExtraLogPayload(interactionId, interaction, lastInteractionFinish, lastInteractionFinishVCResult) {
  var _getTTAI, _getMostRecentVCRevis, _effectiveVCRevisionP, _effectiveVCRevisionP2, _lastInteractionFinis, _getTTAI2, _window$location;
  const config = getConfig();
  if (!config) {
    throw Error('UFO Configuration not provided');
  }
  const {
    end,
    start,
    ufoName,
    rate,
    type,
    abortReason,
    routeName,
    previousInteractionName,
    isPreviousInteractionAborted,
    abortedByInteractionName,
    knownSegments,
    minorInteractions
  } = interaction;
  const sanitisedUfoName = sanitizeUfoName(ufoName);
  const configRate = getExtraInteractionRate(sanitisedUfoName, type);
  if (!coinflip(configRate)) {
    return null;
  }
  if (fg('platform_ufo_disable_ufo_names_config')) {
    const config = getConfig();
    if (config !== null && config !== void 0 && config.disabledUfoNames && config !== null && config !== void 0 && config.disabledUfoNames.includes(sanitisedUfoName)) {
      return null;
    }
  }
  const pageVisibilityAtTTAI = getPageVisibilityUpToTTAI(interaction);
  const isPageLoad = type === 'page_load' || type === 'transition';
  const hasMinorInteractions = minorInteractions !== undefined && minorInteractions.length > 0;
  if (!isPageLoad) {
    // Only create payload for page load or transition
    return null;
  }
  if (hasMinorInteractions && !fg('platform_ufo_send_extra_metrics_on_dirty_vc')) {
    // Not send if aborted by minor interaction, unless feature gate is on
    return null;
  }
  const calculatePageVisibilityFromTheStartOfPageLoad = config.enableBetterPageVisibilityApi && isPageLoad;
  const moreAccuratePageVisibilityAtTTAI = calculatePageVisibilityFromTheStartOfPageLoad ? getMoreAccuratePageVisibilityUpToTTAI(interaction) : null;
  const extraTTAI = (_getTTAI = getTTAI(interaction)) !== null && _getTTAI !== void 0 ? _getTTAI : undefined;
  const newUFOName = sanitizeUfoName(ufoName);
  const finalVCMetrics = await getVCMetrics(interaction, true);

  // Check if VC is clean and has valid metric
  const vcRevisionPayload = finalVCMetrics === null || finalVCMetrics === void 0 ? void 0 : finalVCMetrics['ufo:vc:rev'];
  const effectiveRevision = (_getMostRecentVCRevis = getMostRecentVCRevision(ufoName)) !== null && _getMostRecentVCRevis !== void 0 ? _getMostRecentVCRevis : DEFAULT_TTVC_REVISION;
  const effectiveVCRevisionPayload = vcRevisionPayload === null || vcRevisionPayload === void 0 ? void 0 : vcRevisionPayload.find(({
    revision
  }) => revision === effectiveRevision);
  const isVCCleanFor3p = (_effectiveVCRevisionP = effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload.clean) !== null && _effectiveVCRevisionP !== void 0 ? _effectiveVCRevisionP : false;
  const vcAbortReasonFor3p = (_effectiveVCRevisionP2 = effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload.abortReason) !== null && _effectiveVCRevisionP2 !== void 0 ? _effectiveVCRevisionP2 : undefined;
  const allowDirtyVC = fg('platform_ufo_send_extra_metrics_on_dirty_vc');

  // Always require valid TTAI
  if (extraTTAI === undefined || typeof extraTTAI !== 'number') {
    return null;
  }

  // When feature gate is off, require no errors; when on, allow errors through
  if (interaction.errors.length > 0 && !fg('platform_ufo_send_extra_metrics_on_dirty_vc')) {
    return null;
  }
  if (isVCCleanFor3p) {
    // When VC is clean, require a valid vc90 metric
    if ((effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload['metric:vc90']) === undefined || typeof (effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload['metric:vc90']) !== 'number') {
      return null;
    }
  } else if (!allowDirtyVC) {
    // When VC is dirty and feature gate is off, preserve existing behavior
    return null;
  }
  // When VC is dirty and feature gate is on, continue to send payload

  // Get normal TTAI & VC90 for last finished interaction (without 3p)
  if (!lastInteractionFinish || lastInteractionFinish !== null && lastInteractionFinish !== void 0 && lastInteractionFinish.abortReason || lastInteractionFinish !== null && lastInteractionFinish !== void 0 && (_lastInteractionFinis = lastInteractionFinish.errors) !== null && _lastInteractionFinis !== void 0 && _lastInteractionFinis.length) {
    return null;
  }
  const normalTTAI = (_getTTAI2 = getTTAI(lastInteractionFinish)) !== null && _getTTAI2 !== void 0 ? _getTTAI2 : undefined;
  const lastInteractionFinishStart = typeof lastInteractionFinish.start === 'number' ? Math.round(lastInteractionFinish.start) : undefined;
  const lastInteractionFinishEnd = typeof lastInteractionFinish.end === 'number' ? Math.round(lastInteractionFinish.end) : undefined;
  let lastInteractionFinishVC90 = null;
  let lastInteractionFinishVCClean = false;
  if (lastInteractionFinishVCResult) {
    const lastInteractionFinishVCRev = lastInteractionFinishVCResult['ufo:vc:rev'];
    const lastInteractionFinishRevision = lastInteractionFinishVCRev === null || lastInteractionFinishVCRev === void 0 ? void 0 : lastInteractionFinishVCRev.find(({
      revision
    }) => revision === effectiveRevision);
    if (lastInteractionFinishRevision !== null && lastInteractionFinishRevision !== void 0 && lastInteractionFinishRevision.clean) {
      lastInteractionFinishVCClean = true;
      lastInteractionFinishVC90 = lastInteractionFinishRevision['metric:vc90'];
    } else if (!allowDirtyVC) {
      return null;
    }
  } else if (normalTTAI !== undefined && typeof normalTTAI === 'number' && normalTTAI === extraTTAI) {
    var _effectiveVCRevisionP3, _effectiveVCRevisionP4;
    // Because TTAI is equal between with and without 3p, we can assume VC90 is also equal
    lastInteractionFinishVC90 = (_effectiveVCRevisionP3 = effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload['metric:vc90']) !== null && _effectiveVCRevisionP3 !== void 0 ? _effectiveVCRevisionP3 : null;
    lastInteractionFinishVCClean = (_effectiveVCRevisionP4 = effectiveVCRevisionPayload === null || effectiveVCRevisionPayload === void 0 ? void 0 : effectiveVCRevisionPayload.clean) !== null && _effectiveVCRevisionP4 !== void 0 ? _effectiveVCRevisionP4 : false;
  }

  // Helper function to check if labelStack contains third-party type
  const isThirdParty = labelStack => {
    var _labelStack$some;
    return (_labelStack$some = labelStack === null || labelStack === void 0 ? void 0 : labelStack.some(entry => 'type' in entry && entry.type === 'third-party')) !== null && _labelStack$some !== void 0 ? _labelStack$some : false;
  };

  // Pre-filter 3p data
  const filteredData = {
    errors: interaction.errors.filter(error => isThirdParty(error.labelStack)),
    spans: [...interaction.spans, ...atlaskitInteractionSpans].filter(span => isThirdParty(span.labelStack)),
    requestInfo: interaction.requestInfo.filter(req => isThirdParty(req.labelStack)),
    customTimings: interaction.customTimings.filter(timing => isThirdParty(timing.labelStack)),
    apdex: interaction.apdex.filter(apdex => isThirdParty(apdex.labelStack)),
    reactProfilerTimings: interaction.reactProfilerTimings.filter(timing => isThirdParty(timing.labelStack)),
    customData: interaction.customData.filter(data => isThirdParty(data.labelStack)),
    segments: knownSegments.filter(segment => isThirdParty(segment.labelStack)),
    marks: interaction.marks.filter(mark => isThirdParty(mark.labelStack))
  };
  // Clear atlaskit spans after filtering
  atlaskitInteractionSpans.length = 0;

  // Detailed payload
  const getDetailedInteractionMetrics = () => {
    var _interaction$hold3pIn;
    return {
      errors: filteredData.errors.map(({
        labelStack,
        ...others
      }) => ({
        ...others,
        labelStack: labelStack && optimizeLabelStack(labelStack, getReactUFOPayloadVersion(interaction.type))
      })),
      holdActive: interaction.hold3pActive ? [...interaction.hold3pActive.values()] : [],
      holdInfo: optimizeHoldInfo((_interaction$hold3pIn = interaction.hold3pInfo) !== null && _interaction$hold3pIn !== void 0 ? _interaction$hold3pIn : [], start, getReactUFOPayloadVersion(interaction.type)),
      spans: optimizeSpans(filteredData.spans, start, getReactUFOPayloadVersion(interaction.type)),
      requestInfo: optimizeRequestInfo(filteredData.requestInfo, start, getReactUFOPayloadVersion(interaction.type)),
      customTimings: optimizeCustomTimings(filteredData.customTimings, start)
    };
  };
  const segments3p = !fg('platform_ufo_remove_deprecated_config_fields') && config.killswitchNestedSegments ? [] : filteredData.segments;
  const segmentTree = getReactUFOPayloadVersion(interaction.type) === '2.0.0' ? buildSegmentTree(segments3p.map(segment => segment.labelStack)) : {};
  const payload = {
    actionSubject: 'experience',
    action: 'measured',
    eventType: 'operational',
    source: 'measured',
    tags: ['observability'],
    attributes: {
      properties: {
        // basic
        'event:hostname': ((_window$location = window.location) === null || _window$location === void 0 ? void 0 : _window$location.hostname) || 'unknown',
        'event:product': config.product,
        'event:schema': '1.0.0',
        'event:sizeInKb': 0,
        'event:source': {
          name: 'react-ufo/web',
          version: getReactUFOPayloadVersion(interaction.type)
        },
        'event:region': config.region || 'unknown',
        'experience:key': 'custom.interaction-extra-metrics',
        'experience:name': newUFOName,
        interactionMetrics: {
          namePrefix: config.namePrefix || '',
          segmentPrefix: config.segmentPrefix || '',
          interactionId,
          pageVisibilityAtTTAI,
          experimental__pageVisibilityAtTTAI: moreAccuratePageVisibilityAtTTAI,
          // raw interaction metrics
          rate,
          routeName,
          type,
          abortReason,
          previousInteractionName,
          isPreviousInteractionAborted,
          abortedByInteractionName,
          // performance
          end: Math.round(end),
          start: Math.round(start),
          'metric:ttai:3p': extraTTAI,
          ...finalVCMetrics,
          segments: getReactUFOPayloadVersion(interaction.type) === '2.0.0' ? segmentTree : getOldSegmentsLabelStack(segments3p, interaction.type),
          marks: optimizeMarks(filteredData.marks, getReactUFOPayloadVersion(interaction.type)),
          apdex: optimizeApdex(filteredData.apdex, getReactUFOPayloadVersion(interaction.type)),
          reactProfilerTimings: optimizeReactProfilerTimings(filteredData.reactProfilerTimings, start, getReactUFOPayloadVersion(interaction.type)),
          customData: filteredData.customData,
          ...getDetailedInteractionMetrics()
        },
        vcClean: isVCCleanFor3p,
        ...(vcAbortReasonFor3p !== undefined && {
          vcAbortReason: vcAbortReasonFor3p
        }),
        hasMinorInteractions,
        hasErrors: interaction.errors.length > 0,
        'vc:effective:revision': effectiveRevision,
        lastInteractionFinish: {
          start: lastInteractionFinishStart,
          end: lastInteractionFinishEnd,
          ttai: normalTTAI,
          vc90: lastInteractionFinishVC90,
          vcClean: lastInteractionFinishVCClean
        }
      }
    }
  };
  payload.attributes.properties['event:sizeInKb'] = getPayloadSize(payload.attributes.properties);
  return payload;
}
export default createInteractionExtraLogPayload;
