// @atlaskit/button/dist/cjs/entry-points/unsafe.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "UNSAFE_BUTTON", {
  enumerable: true,
  get: function get() {
    return _button.default;
  }
});
Object.defineProperty(exports, "UNSAFE_DIVIDER", {
  enumerable: true,
  get: function get() {
    return _splitButton.Divider;
  }
});
Object.defineProperty(exports, "UNSAFE_GET_ACTIONS", {
  enumerable: true,
  get: function get() {
    return _splitButton.getActions;
  }
});
Object.defineProperty(exports, "UNSAFE_ICON_BUTTON", {
  enumerable: true,
  get: function get() {
    return _button2.default;
  }
});
Object.defineProperty(exports, "UNSAFE_LINK_BUTTON", {
  enumerable: true,
  get: function get() {
    return _link.default;
  }
});
Object.defineProperty(exports, "UNSAFE_LINK_ICON_BUTTON", {
  enumerable: true,
  get: function get() {
    return _link2.default;
  }
});
Object.defineProperty(exports, "UNSAFE_SPLIT_BUTTON", {
  enumerable: true,
  get: function get() {
    return _splitButton.SplitButton;
  }
});
Object.defineProperty(exports, "UNSAFE_SPLIT_BUTTON_CONTAINER", {
  enumerable: true,
  get: function get() {
    return _splitButton.SplitButtonContainer;
  }
});
Object.defineProperty(exports, "UNSAFE_SPLIT_BUTTON_CONTEXT", {
  enumerable: true,
  get: function get() {
    return _splitButtonContext.SplitButtonContext;
  }
});
var _button = _interopRequireDefault(require("../new-button/variants/default/button"));
var _link = _interopRequireDefault(require("../new-button/variants/default/link"));
var _button2 = _interopRequireDefault(require("../new-button/variants/icon/button"));
var _link2 = _interopRequireDefault(require("../new-button/variants/icon/link"));
var _splitButton = require("../new-button/containers/split-button");
var _splitButtonContext = require("../new-button/containers/split-button/split-button-context");
