// @atlaskit/atlassian-navigation/dist/esm/components/PrimarySplitButton/index.js
/** @jsx jsx */

import { css, jsx } from '@emotion/react';
import { UNSAFE_DIVIDER as Divider, UNSAFE_GET_ACTIONS as getActions, UNSAFE_SPLIT_BUTTON_CONTAINER as SplitButtonContainer, UNSAFE_SPLIT_BUTTON_CONTEXT as SplitButtonContext } from '@atlaskit/button/unsafe';
// Copied from packages/design-system/atlassian-navigation/src/components/PrimaryButton/index.tsx
var buttonBaseStyles = css({
  display: 'flex',
  height: '100%',
  position: 'relative',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column'
});

// Mostly copied from packages/design-system/atlassian-navigation/src/components/PrimaryButton/index.tsx
var buttonHighlightedStyles = css({
  '&:after': {
    height: 3,
    position: 'absolute',
    backgroundColor: "var(--ds-border-selected, #0052cc)",
    borderStartEndRadius: "var(--ds-border-radius-050, 1px)",
    borderStartStartRadius: "var(--ds-border-radius-050, 1px)",
    content: '""',
    insetBlockEnd: 0,
    insetInlineEnd: "var(--ds-space-050, 4px)",
    insetInlineStart: "var(--ds-space-050, 4px)"
  }
});
/**
 * TODO: Add JSDoc
 */
export var PrimarySplitButton = function PrimarySplitButton(_ref) {
  var children = _ref.children,
    _ref$isHighlighted = _ref.isHighlighted,
    isHighlighted = _ref$isHighlighted === void 0 ? false : _ref$isHighlighted;
  var _getActions = getActions(children),
    PrimaryAction = _getActions.PrimaryAction,
    SecondaryAction = _getActions.SecondaryAction;
  return jsx("div", {
    css: [buttonBaseStyles, isHighlighted && buttonHighlightedStyles]
  }, jsx(SplitButtonContainer, null, jsx(SplitButtonContext.Provider, {
    value: {
      appearance: 'navigation',
      isHighlighted: isHighlighted
    }
  }, PrimaryAction, jsx(Divider, {
    appearance: "navigation",
    spacing: "default",
    isDisabled: false
  }), SecondaryAction)));
};
