// @atlaskit/page-header/dist/cjs/PageHeader/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TitleWrapper = exports.TitleContainer = exports.StyledTitleWrapper = exports.OuterWrapper = exports.BottomBarWrapper = exports.ActionsWrapper = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _primitives = require("@atlaskit/primitives");
// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. The prop name concerned here is truncateTitle.
// This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var truncateStyles = (0, _react2.css)({
  overflowX: 'hidden',
  overflowY: 'hidden',
  // `overflow-y` necessary to prevent a scrollbar from showing in FF: https://product-fabric.atlassian.net/browse/PYX-1035
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});
var outerStyles = (0, _react2.css)({
  margin: "var(--ds-space-300, 24px)".concat(" 0 ", "var(--ds-space-200, 16px)", " 0")
});
var titleStyles = (0, _react2.css)({
  color: "var(--ds-text, #172B4D)",
  font: "var(--ds-font-heading-large, normal 500 24px/28px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  letterSpacing: 'normal',
  marginBlockStart: 0,
  outline: 'none'
});
var titleWrapperStyles = (0, _react2.css)({
  display: 'flex',
  alignItems: 'flex-start',
  flexWrap: 'wrap'
});
var titleWrapperTruncateStyles = (0, _react2.css)({
  flexWrap: 'nowrap'
});
var titleContainerStyles = (0, _react2.css)({
  minWidth: 0,
  maxWidth: '100%',
  flex: '1 0 auto',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  flexShrink: undefined,
  marginBlockEnd: "var(--ds-space-100, 8px)"
});
var actionStyles = (0, _react2.css)({
  maxWidth: '100%',
  flex: '0 0 auto',
  marginBlockEnd: "var(--ds-space-100, 8px)",
  marginInlineStart: 'auto',
  paddingInlineStart: "var(--ds-space-400, 32px)",
  whiteSpace: 'nowrap',
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles, @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '>': {
    textAlign: 'right'
  }
});
var titleContainerTruncateStyles = (0, _react2.css)({
  flexShrink: 1
});
var bottomBarStyles = (0, _primitives.xcss)({
  marginBlockStart: 'space.200'
});

/**
 * __Outer wrapper__.
 *
 * An outer wrapper that is the outermost component of the PageHeader component. It wraps around the PageHeader, its Actions,
 * the BottomBar and its Breadcrumbs.
 *
 */
var OuterWrapper = exports.OuterWrapper = function OuterWrapper(_ref) {
  var children = _ref.children;
  return (0, _react2.jsx)("div", {
    css: outerStyles
  }, children);
};
/**
 * __Styled title wrapper__.
 *
 * A styled title wrapper is a wrapper around the title that controls its the styles exclusively.
 *
 */
var StyledTitleWrapper = exports.StyledTitleWrapper = /*#__PURE__*/_react.default.forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    id = _ref2.id,
    truncateTitle = _ref2.truncateTitle;
  return (0, _react2.jsx)("h1", {
    css: [titleStyles, truncateTitle && truncateStyles],
    ref: ref,
    tabIndex: -1,
    id: id
  }, children);
});
/**
 * __Title wrapper__.
 *
 * A title wrapper is a wrapper around the title and the actions.
 *
 */
var TitleWrapper = exports.TitleWrapper = function TitleWrapper(_ref3) {
  var children = _ref3.children,
    truncateTitle = _ref3.truncateTitle;
  return (0, _react2.jsx)("div", {
    css: [titleWrapperStyles, truncateTitle && titleWrapperTruncateStyles]
  }, children);
};

/**
 * Title container.
 *
 * A title container is a container that wraps around the title and its styles (if applied).
 *
 */
var TitleContainer = exports.TitleContainer = function TitleContainer(_ref4) {
  var children = _ref4.children,
    truncateTitle = _ref4.truncateTitle;
  return (0, _react2.jsx)("div", {
    css: [titleContainerStyles, truncateTitle && titleContainerTruncateStyles]
  }, children);
};

/**
 * __Actions wrapper__.
 *
 * An actions wrapper is a wrapper for the actions, which appear on the top right of the PageHeader component.
 *
 */
var ActionsWrapper = exports.ActionsWrapper = function ActionsWrapper(_ref5) {
  var children = _ref5.children;
  return (0, _react2.jsx)("div", {
    css: actionStyles
  }, children);
};

/**
 * __Bottom bar wrapper__.
 *
 * A bottom bar wrapper is a wrapper for the bottom bar, which appears at the bottom of the PageHeader component.
 *
 */
var BottomBarWrapper = exports.BottomBarWrapper = function BottomBarWrapper(_ref6) {
  var children = _ref6.children;
  return (0, _react2.jsx)(_primitives.Box, {
    xcss: bottomBarStyles
  }, children);
};
