// @atlaskit/progress-tracker/dist/cjs/progress-tracker.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _constants = require("./constants");
var _constants2 = require("./internal/constants");
var _link = _interopRequireDefault(require("./internal/link"));
var _linkNew = _interopRequireDefault(require("./internal/link-new"));
var _stage = _interopRequireDefault(require("./internal/stage"));
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; } /* eslint-disable @atlaskit/design-system/no-nested-styles */ /**
 * @jsxRuntime classic
 * @jsx jsx
 */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var containerStyles = (0, _react2.css)({
  display: 'grid',
  width: '100%',
  margin: '0 auto',
  padding: "var(--ds-space-0, 0px)",
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  gap: "var(".concat(_constants2.varSpacing, ")"),
  listStyleType: 'none',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-nested-selectors, @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  '&&': {
    marginBlockStart: "var(--ds-space-500, 40px)"
  }
});
/**
 * __Progress tracker__
 *
 * A progress tracker displays the steps and progress through a journey.
 *
 * - [Examples](https://atlassian.design/components/progress-tracker/examples)
 * - [Code](https://atlassian.design/components/progress-tracker/code)
 * - [Usage](https://atlassian.design/components/progress-tracker/usage)
 */
var ProgressTracker = function ProgressTracker(_ref) {
  var _listInlineStyles;
  var _ref$items = _ref.items,
    items = _ref$items === void 0 ? [] : _ref$items,
    _ref$spacing = _ref.spacing,
    spacing = _ref$spacing === void 0 ? 'cosy' : _ref$spacing,
    _ref$render = _ref.render,
    render = _ref$render === void 0 ? {
      link: function link(_ref2) {
        var item = _ref2.item;
        return (
          // Anchor content is coming from another location
          // eslint-disable-next-line jsx-a11y/anchor-has-content
          (0, _platformFeatureFlags.fg)('platform_progress_tracker_link_migration') ? (0, _react2.jsx)(_linkNew.default, item) : (0, _react2.jsx)(_link.default, item)
        );
      }
    } : _ref$render,
    _ref$animated = _ref.animated,
    animated = _ref$animated === void 0 ? true : _ref$animated,
    testId = _ref.testId,
    _ref$label = _ref.label,
    label = _ref$label === void 0 ? 'Progress' : _ref$label;
  var prevItems = (0, _react.useRef)(items.map(function (stage) {
    return _objectSpread(_objectSpread({}, stage), {}, {
      percentageComplete: 0
    });
  }));
  var previousStages = items.map(function (stage) {
    var _prevItems$current;
    var oldStage = (_prevItems$current = prevItems.current) === null || _prevItems$current === void 0 ? void 0 : _prevItems$current.find(function (st) {
      return st.id === stage.id;
    });
    return !!oldStage ? oldStage : stage;
  });
  (0, _react.useEffect)(function () {
    prevItems.current = items;
  }, [items]);
  var progressChanges = items.filter(function (stage, index) {
    return stage.percentageComplete !== previousStages[index].percentageComplete;
  }).length;
  var totalStepsForward = items.filter(function (stage, index) {
    return stage.percentageComplete > previousStages[index].percentageComplete;
  }).length;
  var totalStepsBack = items.filter(function (stage, index) {
    return stage.percentageComplete < previousStages[index].percentageComplete;
  }).length;
  var stepsForward = totalStepsForward;
  var stepsBack = totalStepsBack;
  var progressItems = items.map(function (stage, index) {
    var transitionSpeed = 0;
    var transitionDelay = 0;
    var transitionEasing = progressChanges > 1 ? 'linear' : _constants2.ANIMATION_EASE_OUT;
    if (animated) {
      transitionSpeed = progressChanges > 1 ? _constants2.LINEAR_TRANSITION_SPEED : _constants2.TRANSITION_SPEED;
      if (stage.percentageComplete < previousStages[index].percentageComplete) {
        /**
         * Load each transition sequentially in reverse.
         */
        transitionDelay = (stepsBack - 1) * transitionSpeed;
        stepsBack -= 1;
      } else if (stage.percentageComplete > previousStages[index].percentageComplete) {
        /**
         * Load each transition sequentially.
         */
        transitionDelay = (totalStepsForward - stepsForward) * transitionSpeed;
        stepsForward -= 1;
      }
    }
    return (0, _react2.jsx)(_stage.default, {
      transitionSpeed: transitionSpeed,
      transitionDelay: transitionDelay,
      transitionEasing: transitionEasing,
      key: stage.id,
      item: stage,
      render: render
    });
  });
  var listInlineStyles = (_listInlineStyles = {
    gridTemplateColumns: "repeat(".concat(items.length, ", 1fr)")
  }, (0, _defineProperty2.default)(_listInlineStyles, _constants2.varSpacing, _constants.spacing[spacing]), (0, _defineProperty2.default)(_listInlineStyles, "maxWidth", 8 * 10 * items.length * 2), _listInlineStyles);
  return (0, _react2.jsx)("ul", {
    "data-testid": testId
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
    ,
    style: listInlineStyles,
    css: containerStyles,
    "aria-label": label
  }, progressItems);
};
var _default = exports.default = ProgressTracker;
