// @atlaskit/atlassian-navigation/dist/es2019/components/Search/index.js
import _extends from "@babel/runtime/helpers/extends";
/** @jsx jsx */
import { Fragment } from 'react';
import { css, jsx } from '@emotion/react';
import SearchIcon from '@atlaskit/icon/glyph/search';
import { getBooleanFF } from '@atlaskit/platform-feature-flags';
import { CREATE_BREAKPOINT, fontSize, varSearchBackgroundColor, varSearchBorderColor, varSearchColor, varSearchFocusBorderColor, varSearchHoverBackgroundColor } from '../../common/constants';
import { useTheme } from '../../theme';
import { IconButton } from '../IconButton';
const searchInputContainerStyles = css({
  marginRight: "var(--ds-space-100, 8px)",
  marginLeft: "var(--ds-space-250, 20px)",
  position: 'relative',
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  [`@media (max-width: ${CREATE_BREAKPOINT - 1}px)`]: {
    display: 'none !important'
  }
});
const searchInputIconStyles = css({
  width: '20px',
  height: '20px',
  position: 'absolute',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  top: '5px',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: '10px',
  pointerEvents: 'none'
});
const searchInputStyles = css({
  boxSizing: 'border-box',
  width: '220px',
  height: "var(--ds-space-400, 32px)",
  padding: `0 ${"var(--ds-space-100, 8px)"} 0 ${"var(--ds-space-500, 40px)"}`,
  backgroundColor: `var(${varSearchBackgroundColor})`,
  border: `${getBooleanFF('platform.design-system-team.border-checkbox_nyoiu') ? "var(--ds-border-width, 1px)" : '2px'} solid`,
  borderColor: `var(${varSearchBorderColor})`,
  borderRadius: "var(--ds-border-radius-200, 6px)",
  color: `var(${varSearchColor})`,
  fontSize: `${fontSize}px`,
  outline: 'none',
  '::placeholder': {
    color: 'inherit'
  },
  '&:focus': {
    borderColor: `var(${varSearchFocusBorderColor})`,
    boxShadow: getBooleanFF('platform.design-system-team.border-checkbox_nyoiu') ? `inset 0 0 0 ${"var(--ds-border-width, 1px)"} var(${varSearchFocusBorderColor})` : undefined
  },
  '&:hover': {
    backgroundColor: `var(${varSearchHoverBackgroundColor})`
  }
});
const searchIconStyles = css({
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  [`@media (min-width: ${CREATE_BREAKPOINT}px)`]: {
    display: 'none !important'
  }
});
const SearchComponent = props => {
  const {
    onClick,
    placeholder,
    label,
    value
  } = props;
  const theme = useTheme();
  const search = theme.mode.search;
  const onChange = (...args) => {
    // @ts-expect-error
    onClick && onClick(...args);
  };
  const onInputClick = (...args) => {
    // @ts-expect-error
    onClick && onClick(...args);
  };
  const searchInputDynamicStyles = {
    [varSearchBackgroundColor]: search.default.backgroundColor,
    [varSearchColor]: search.default.color,
    [varSearchBorderColor]: search.default.borderColor,
    [varSearchFocusBorderColor]: search.focus.borderColor,
    [varSearchHoverBackgroundColor]: search.hover.backgroundColor
  };
  return jsx("div", {
    css: searchInputContainerStyles,
    role: "search"
  }, jsx("div", {
    css: searchInputIconStyles
  }, jsx(SearchIcon, {
    label: ""
  })), jsx("input", {
    style: searchInputDynamicStyles,
    css: searchInputStyles,
    "aria-label": label,
    placeholder: placeholder,
    onChange: onChange,
    onClick: onInputClick,
    value: value
  }));
};

/**
 * __Search__
 *
 * A search input that can be passed into `AtlassianNavigation`'s `renderSearch` prop.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples#search)
 * - [Code](https://atlassian.design/components/atlassian-navigation/code)
 */
export const Search = props => {
  const {
    placeholder,
    tooltip,
    label,
    value,
    ...iconButtonProps
  } = props;
  return jsx(Fragment, null, jsx(SearchComponent, {
    onClick: iconButtonProps.onClick,
    placeholder: placeholder,
    label: label,
    value: value || ''
  }), jsx(IconButton
  // @ts-ignore Overriding styles is not supported.
  , _extends({
    css: searchIconStyles,
    icon: jsx(SearchIcon, {
      label: label
    }),
    tooltip: tooltip
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, iconButtonProps)));
};
export default Search;
