// @atlaskit/page-layout/dist/cjs/controllers/sidebar-resize-context.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.usePageLayoutResize = exports.useLeftSidebarFlyoutLock = exports.SidebarResizeContext = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _noop = _interopRequireDefault(require("@atlaskit/ds-lib/noop"));
var _excluded = ["setLeftSidebarState"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var leftSidebarState = {
  isFlyoutOpen: false,
  isResizing: false,
  isLeftSidebarCollapsed: false,
  leftSidebarWidth: 0,
  lastLeftSidebarWidth: 0,
  flyoutLockCount: 0,
  isFixed: true
};
// eslint-disable-next-line @repo/internal/react/require-jsdoc
var SidebarResizeContext = exports.SidebarResizeContext = /*#__PURE__*/(0, _react.createContext)({
  isLeftSidebarCollapsed: false,
  expandLeftSidebar: _noop.default,
  collapseLeftSidebar: _noop.default,
  leftSidebarState: leftSidebarState,
  setLeftSidebarState: _noop.default,
  toggleLeftSidebar: _noop.default
});
var usePageLayoutResize = exports.usePageLayoutResize = function usePageLayoutResize() {
  var _useContext = (0, _react.useContext)(SidebarResizeContext),
    setLeftSidebarState = _useContext.setLeftSidebarState,
    context = (0, _objectWithoutProperties2.default)(_useContext, _excluded);
  return context;
};

/**
 * _**WARNING:**_ This hook is intended as a temporary solution and
 * is likely to be removed in a future version of page-layout.
 *
 * ---
 *
 * This hook will prevent the left sidebar from automatically collapsing
 * when it is in a flyout state.
 *
 * The intended use case for this hook is to allow popup menus in the
 * left sidebar to be usable while it is in a flyout state.
 *
 * ## Usage
 * The intended usage is to use this hook within the popup component
 * you are rendering. This way the left sidebar will be locked for
 * as long as the popup is open.
 */
var useLeftSidebarFlyoutLock = exports.useLeftSidebarFlyoutLock = function useLeftSidebarFlyoutLock() {
  var _useContext2 = (0, _react.useContext)(SidebarResizeContext),
    setLeftSidebarState = _useContext2.setLeftSidebarState;
  (0, _react.useEffect)(function () {
    setLeftSidebarState(function (current) {
      return _objectSpread(_objectSpread({}, current), {}, {
        flyoutLockCount: current.flyoutLockCount + 1
      });
    });
    return function () {
      setLeftSidebarState(function (current) {
        return _objectSpread(_objectSpread({}, current), {}, {
          flyoutLockCount: current.flyoutLockCount - 1
        });
      });
    };
  }, [setLeftSidebarState]);
};
