// @atlaskit/atlassian-navigation/dist/cjs/components/Search/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Search = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _search = _interopRequireDefault(require("@atlaskit/icon/glyph/search"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _constants = require("../../common/constants");
var _theme = require("../../theme");
var _IconButton = require("../IconButton");
var _excluded = ["placeholder", "tooltip", "label", "value"];
/** @jsx jsx */
var searchInputContainerStyles = (0, _react2.css)((0, _defineProperty2.default)({
  marginRight: "var(--ds-space-100, 8px)",
  marginLeft: "var(--ds-space-250, 20px)",
  position: 'relative'
}, "@media (max-width: ".concat(_constants.CREATE_BREAKPOINT - 1, "px)"), {
  display: 'none !important'
}));
var searchInputIconStyles = (0, _react2.css)({
  width: '20px',
  height: '20px',
  position: 'absolute',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  top: '5px',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: '10px',
  pointerEvents: 'none'
});
var searchInputStyles = (0, _react2.css)({
  boxSizing: 'border-box',
  width: '220px',
  height: "var(--ds-space-400, 32px)",
  padding: "0 ".concat("var(--ds-space-100, 8px)", " 0 ", "var(--ds-space-500, 40px)"),
  backgroundColor: "var(".concat(_constants.varSearchBackgroundColor, ")"),
  border: "".concat((0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.border-checkbox_nyoiu') ? "var(--ds-border-width, 1px)" : '2px', " solid"),
  borderColor: "var(".concat(_constants.varSearchBorderColor, ")"),
  borderRadius: "var(--ds-border-radius-200, 6px)",
  color: "var(".concat(_constants.varSearchColor, ")"),
  fontSize: "".concat(_constants.fontSize, "px"),
  outline: 'none',
  '::placeholder': {
    color: 'inherit'
  },
  '&:focus': {
    borderColor: "var(".concat(_constants.varSearchFocusBorderColor, ")"),
    boxShadow: (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.border-checkbox_nyoiu') ? "inset 0 0 0 ".concat("var(--ds-border-width, 1px)", " var(", _constants.varSearchFocusBorderColor, ")") : undefined
  },
  '&:hover': {
    backgroundColor: "var(".concat(_constants.varSearchHoverBackgroundColor, ")")
  }
});
var searchIconStyles = (0, _react2.css)((0, _defineProperty2.default)({}, "@media (min-width: ".concat(_constants.CREATE_BREAKPOINT, "px)"), {
  display: 'none !important'
}));
var SearchComponent = function SearchComponent(props) {
  var _searchInputDynamicSt;
  var onClick = props.onClick,
    placeholder = props.placeholder,
    label = props.label,
    value = props.value;
  var theme = (0, _theme.useTheme)();
  var search = theme.mode.search;
  var onChange = function onChange() {
    // @ts-expect-error
    onClick && onClick.apply(void 0, arguments);
  };
  var onInputClick = function onInputClick() {
    // @ts-expect-error
    onClick && onClick.apply(void 0, arguments);
  };
  var searchInputDynamicStyles = (_searchInputDynamicSt = {}, (0, _defineProperty2.default)(_searchInputDynamicSt, _constants.varSearchBackgroundColor, search.default.backgroundColor), (0, _defineProperty2.default)(_searchInputDynamicSt, _constants.varSearchColor, search.default.color), (0, _defineProperty2.default)(_searchInputDynamicSt, _constants.varSearchBorderColor, search.default.borderColor), (0, _defineProperty2.default)(_searchInputDynamicSt, _constants.varSearchFocusBorderColor, search.focus.borderColor), (0, _defineProperty2.default)(_searchInputDynamicSt, _constants.varSearchHoverBackgroundColor, search.hover.backgroundColor), _searchInputDynamicSt);
  return (0, _react2.jsx)("div", {
    css: searchInputContainerStyles,
    role: "search"
  }, (0, _react2.jsx)("div", {
    css: searchInputIconStyles
  }, (0, _react2.jsx)(_search.default, {
    label: ""
  })), (0, _react2.jsx)("input", {
    style: searchInputDynamicStyles,
    css: searchInputStyles,
    "aria-label": label,
    placeholder: placeholder,
    onChange: onChange,
    onClick: onInputClick,
    value: value
  }));
};

/**
 * __Search__
 *
 * A search input that can be passed into `AtlassianNavigation`'s `renderSearch` prop.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples#search)
 * - [Code](https://atlassian.design/components/atlassian-navigation/code)
 */
var Search = exports.Search = function Search(props) {
  var placeholder = props.placeholder,
    tooltip = props.tooltip,
    label = props.label,
    value = props.value,
    iconButtonProps = (0, _objectWithoutProperties2.default)(props, _excluded);
  return (0, _react2.jsx)(_react.Fragment, null, (0, _react2.jsx)(SearchComponent, {
    onClick: iconButtonProps.onClick,
    placeholder: placeholder,
    label: label,
    value: value || ''
  }), (0, _react2.jsx)(_IconButton.IconButton
  // @ts-ignore Overriding styles is not supported.
  , (0, _extends2.default)({
    css: searchIconStyles,
    icon: (0, _react2.jsx)(_search.default, {
      label: label
    }),
    tooltip: tooltip
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, iconButtonProps)));
};
var _default = exports.default = Search;
