// @atlaskit/table-tree/dist/es2019/components/internal/chevron.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
/* eslint-disable @repo/internal/react/consistent-props-definitions */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { IconButton } from '@atlaskit/button/new';
import ChevronDownIcon from '@atlaskit/icon/glyph/chevron-down';
import ChevronRightIcon from '@atlaskit/icon/glyph/chevron-right';
import { ChevronContainer } from './styled';
export default class Chevron extends Component {
  constructor(...args) {
    super(...args);
    _defineProperty(this, "handleClick", () => {
      if (this.props.onExpandToggle) {
        this.props.onExpandToggle();
      }
    });
  }
  render() {
    const {
      isExpanded,
      ariaControls,
      collapseLabel,
      expandLabel,
      rowId,
      extendedLabel
    } = this.props;
    const getLabel = defaultLabel => extendedLabel ? `${defaultLabel} ${extendedLabel} row` : `${defaultLabel} row ${rowId}`;
    return /*#__PURE__*/React.createElement(ChevronContainer, null, /*#__PURE__*/React.createElement(IconButton, {
      appearance: "subtle",
      onClick: this.handleClick,
      spacing: "compact",
      icon: isExpanded ? ChevronDownIcon : ChevronRightIcon,
      "aria-controls": ariaControls,
      label: isExpanded ? getLabel(collapseLabel) : getLabel(expandLabel)
    }));
  }
}
_defineProperty(Chevron, "defaultProps", {
  expandLabel: 'Expand',
  collapseLabel: 'Collapse'
});
