// @atlaskit/table-tree/dist/es2019/components/internal/styled.js
import _extends from "@babel/runtime/helpers/extends";
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { N30 } from '@atlaskit/theme/colors';
export const indentBase = "var(--ds-space-300, 25px)";
const treeRowContainerStyles = css({
  display: 'flex',
  borderBlockEnd: `1px solid ${`var(--ds-border, ${N30})`}`
});

/**
 * __Tree row container__
 */
export const TreeRowContainer = props => jsx("div", _extends({
  role: "row",
  css: treeRowContainerStyles
  // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
}, props));
const commonCellElementStyles = css({
  display: 'flex',
  position: 'absolute',
  alignItems: 'center',
  // indentBase is re-used elsewhere and is primarily used as positive value; we need to negate it here
  marginInlineStart: `calc(${indentBase} * -1)`
});
const commonChevronContainerStyles = css({
  // Aligns position:absolute chevron button with the adjacent text. Any future visual breaking changes
  // should consider setting this to `-2px` for better alignment, or refactor completely
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  marginBlockStart: -3
});
/**
 * __Chevron container__
 *
 * A wrapper container around the expand table tree button.
 */
export const ChevronContainer = props => jsx("span", _extends({}, props, {
  css: [commonCellElementStyles, commonChevronContainerStyles]
}));
const loadingItemContainerStyles = css({
  width: '100%'
});
const paddingLeftStyles = css({
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  paddingInlineStart: '50%'
});
/**
 * __Loader item container__
 *
 * A loader item container.
 */
export const LoaderItemContainer = ({
  isRoot,
  ...props
}) => jsx("span", _extends({
  css: [commonCellElementStyles, loadingItemContainerStyles, isRoot && paddingLeftStyles]
  // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
}, props));
