// @atlaskit/side-navigation/dist/cjs/components/LoadingItems/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("@emotion/react");
var _motion = require("@atlaskit/motion");
var _context = require("../NestableNavigationContent/context");
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var baseMotionStyles = (0, _react.css)({
  position: 'absolute',
  zIndex: 1,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @atlaskit/design-system/no-physical-properties
  top: 0,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @atlaskit/design-system/no-physical-properties
  right: 0,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @atlaskit/design-system/no-physical-properties
  left: 0
});
var enteringStyles = (0, _react.css)({
  position: 'static',
  zIndex: 2
});

/**
 * __Loading items__
 *
 * Loading items conditionally render based on the useShouldNestedElementRender() hook.
 */
var LoadingItems = function LoadingItems(_ref) {
  var children = _ref.children,
    isLoading = _ref.isLoading,
    fallback = _ref.fallback,
    testId = _ref.testId;
  var _useShouldNestedEleme = (0, _context.useShouldNestedElementRender)(),
    shouldRender = _useShouldNestedEleme.shouldRender;
  if (!shouldRender) {
    return children;
  }
  return (0, _react.jsx)(_motion.ExitingPersistence, null, (0, _react.jsx)(_motion.FadeIn, {
    key: "loading-section-".concat(isLoading),
    duration: _motion.mediumDurationMs
  }, function (motion, state) {
    return (0, _react.jsx)("span", (0, _extends2.default)({}, motion, {
      "data-testid": testId && "".concat(testId, "--").concat(state),
      css: [baseMotionStyles, state === 'entering' && enteringStyles]
    }), isLoading ? fallback : children);
  }));
};
var _default = exports.default = LoadingItems;
