// @atlaskit/react-ufo/dist/es2019/set-terminal-error/index.js
import { useEffect, useRef } from 'react';
import { fg } from '@atlaskit/platform-feature-flags';
import { getActiveTrace } from '../experience-trace-id-context';
import { useInteractionContext } from '../interaction-context';
import { getActiveInteraction, PreviousInteractionLog } from '../interaction-metrics';
import UFORouteName from '../route-name-context';
let sinkHandlerFn = () => {};
export function sinkTerminalErrorHandler(fn) {
  sinkHandlerFn = fn;
}
const RELAY_NETWORK_ERRORS_NAME = 'RelayNetwork';
/**
 * Relay error structure
 */

export const isRelayNetworkError = error => error.name === RELAY_NETWORK_ERRORS_NAME;
const isErrorObject = error => typeof error === 'object' && error !== null;
const getErrorStatusCode = error => {
  if (!isErrorObject(error)) {
    return undefined;
  }
  if (isRelayNetworkError(error)) {
    var _error$source, _error$source$errors, _error$source$errors$, _error$source$errors$2;
    return (_error$source = error.source) === null || _error$source === void 0 ? void 0 : (_error$source$errors = _error$source.errors) === null || _error$source$errors === void 0 ? void 0 : (_error$source$errors$ = _error$source$errors[0]) === null || _error$source$errors$ === void 0 ? void 0 : (_error$source$errors$2 = _error$source$errors$.extensions) === null || _error$source$errors$2 === void 0 ? void 0 : _error$source$errors$2.statusCode;
  }
  if ('statusCode' in error && typeof error.statusCode === 'number') {
    return error.statusCode;
  }
  return undefined;
};
const getErrorTraceId = error => isErrorObject(error) && 'traceId' in error && typeof error.traceId === 'string' ? error.traceId : undefined;
export function setTerminalError(error, additionalAttributes, labelStack) {
  var _error$message, _getActiveTrace$trace, _getActiveTrace, _activeInteraction$uf, _activeInteraction$id, _activeInteraction$ty, _PreviousInteractionL, _PreviousInteractionL2, _PreviousInteractionL3, _UFORouteName$current;
  if (additionalAttributes !== null && additionalAttributes !== void 0 && additionalAttributes.isClientNetworkError) {
    // Exclude client network errors from being reported to UFO
    return;
  }
  const activeInteraction = getActiveInteraction();
  const currentTime = performance.now();
  const baseErrorData = {
    errorType: error.name || 'Error',
    errorMessage: ((_error$message = error.message) === null || _error$message === void 0 ? void 0 : _error$message.slice(0, 100)) || 'Unknown error',
    timestamp: currentTime
  };
  const errorData = fg('platform_ufo_terminal_errors_fix_missing_data') ? {
    ...baseErrorData,
    statusCode: getErrorStatusCode(error),
    // Fallback to traceId from error object if it exists (e.g. FetchError)
    traceId: (_getActiveTrace$trace = (_getActiveTrace = getActiveTrace()) === null || _getActiveTrace === void 0 ? void 0 : _getActiveTrace.traceId) !== null && _getActiveTrace$trace !== void 0 ? _getActiveTrace$trace : getErrorTraceId(error),
    teamName: additionalAttributes === null || additionalAttributes === void 0 ? void 0 : additionalAttributes.teamName,
    packageName: additionalAttributes === null || additionalAttributes === void 0 ? void 0 : additionalAttributes.packageName,
    errorBoundaryId: additionalAttributes === null || additionalAttributes === void 0 ? void 0 : additionalAttributes.errorBoundaryId,
    errorHash: additionalAttributes === null || additionalAttributes === void 0 ? void 0 : additionalAttributes.errorHash,
    fallbackType: additionalAttributes === null || additionalAttributes === void 0 ? void 0 : additionalAttributes.fallbackType
  } : {
    ...baseErrorData,
    ...additionalAttributes
  };

  // Calculate time since previous interaction
  const timeSincePreviousInteraction = PreviousInteractionLog.timestamp != null ? currentTime - PreviousInteractionLog.timestamp : null;
  const context = {
    labelStack: labelStack !== null && labelStack !== void 0 ? labelStack : null,
    activeInteractionName: (_activeInteraction$uf = activeInteraction === null || activeInteraction === void 0 ? void 0 : activeInteraction.ufoName) !== null && _activeInteraction$uf !== void 0 ? _activeInteraction$uf : null,
    activeInteractionId: (_activeInteraction$id = activeInteraction === null || activeInteraction === void 0 ? void 0 : activeInteraction.id) !== null && _activeInteraction$id !== void 0 ? _activeInteraction$id : null,
    activeInteractionType: (_activeInteraction$ty = activeInteraction === null || activeInteraction === void 0 ? void 0 : activeInteraction.type) !== null && _activeInteraction$ty !== void 0 ? _activeInteraction$ty : null,
    previousInteractionId: (_PreviousInteractionL = PreviousInteractionLog.id) !== null && _PreviousInteractionL !== void 0 ? _PreviousInteractionL : null,
    previousInteractionName: (_PreviousInteractionL2 = PreviousInteractionLog.name) !== null && _PreviousInteractionL2 !== void 0 ? _PreviousInteractionL2 : null,
    previousInteractionType: (_PreviousInteractionL3 = PreviousInteractionLog.type) !== null && _PreviousInteractionL3 !== void 0 ? _PreviousInteractionL3 : null,
    timeSincePreviousInteraction,
    routeName: (_UFORouteName$current = UFORouteName.current) !== null && _UFORouteName$current !== void 0 ? _UFORouteName$current : null
  };
  sinkHandlerFn(errorData, context);
}
export function useReportTerminalError(error, additionalAttributes) {
  const interactionContext = useInteractionContext();
  const hasReportedRef = useRef(false);
  useEffect(() => {
    if (error && !hasReportedRef.current) {
      hasReportedRef.current = true;
      setTerminalError(error, additionalAttributes, interactionContext === null || interactionContext === void 0 ? void 0 : interactionContext.labelStack);
    }
  }, [error, additionalAttributes, interactionContext === null || interactionContext === void 0 ? void 0 : interactionContext.labelStack]);
}
