// @compiled/react/dist/cjs/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.jsx = exports.cx = exports.styled = exports.keyframes = exports.cssMap = exports.css = exports.createStrictAPI = exports.ClassNames = void 0;
const react_1 = require("react");
var index_js_1 = require("./class-names/index.js");
Object.defineProperty(exports, "ClassNames", { enumerable: true, get: function () { return index_js_1.ClassNames; } });
var index_js_2 = require("./create-strict-api/index.js");
Object.defineProperty(exports, "createStrictAPI", { enumerable: true, get: function () { return index_js_2.createStrictAPI; } });
var index_js_3 = require("./css/index.js");
Object.defineProperty(exports, "css", { enumerable: true, get: function () { return __importDefault(index_js_3).default; } });
var index_js_4 = require("./css-map/index.js");
Object.defineProperty(exports, "cssMap", { enumerable: true, get: function () { return __importDefault(index_js_4).default; } });
var index_js_5 = require("./keyframes/index.js");
Object.defineProperty(exports, "keyframes", { enumerable: true, get: function () { return index_js_5.keyframes; } });
var index_js_6 = require("./styled/index.js");
Object.defineProperty(exports, "styled", { enumerable: true, get: function () { return index_js_6.styled; } });
var index_js_7 = require("./xcss-prop/index.js");
Object.defineProperty(exports, "cx", { enumerable: true, get: function () { return index_js_7.cx; } });
// Pass through the (classic) jsx runtime.
// Compiled currently doesn't define its own and uses this purely to enable a local jsx namespace.
exports.jsx = react_1.createElement;
