// @atlaskit/page-layout/dist/cjs/components/resize-control/resize-button.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("@emotion/react");
var _chevronRight = _interopRequireDefault(require("@atlaskit/icon/glyph/chevron-right"));
var _curves = require("@atlaskit/motion/curves");
var _durations = require("@atlaskit/motion/durations");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _responsive = require("@atlaskit/primitives/responsive");
var _colors = require("@atlaskit/theme/colors");
var _constants = require("../../common/constants");
var _excluded = ["isLeftSidebarCollapsed", "label", "testId"];
/** @jsx jsx */
var increaseHitAreaStyles = (0, _react.css)({
  position: 'absolute',
  insetBlockEnd: "var(--ds-space-negative-100, -8px)",
  insetBlockStart: "var(--ds-space-negative-100, -8px)",
  insetInlineEnd: "var(--ds-space-negative-150, -12px)",
  insetInlineStart: "var(--ds-space-negative-100, -8px)"
});
var mobileStyles = (0, _react.css)((0, _defineProperty2.default)({}, _responsive.UNSAFE_media.below.sm, {
  opacity: 1
}));
var resizeIconButtonStyles = (0, _react.css)({
  width: 24,
  height: 24,
  padding: "var(--ds-space-0, 0px)",
  position: 'absolute',
  backgroundColor: "var(--ds-surface-overlay, ".concat(_colors.N0, ")"),
  border: 0,
  borderRadius: "var(--ds-border-radius-circle, 50%)",
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage -- TODO: https://product-fabric.atlassian.net/browse/DSP-3392; this boxShadow needs further investigation, along with the hover and active background colors.
  boxShadow: "0 0 0 1px ".concat(_colors.N30A, ", 0 2px 4px 1px ").concat(_colors.N30A),
  color: "var(--ds-text-subtle, ".concat(_colors.N200, ")"),
  cursor: 'pointer',
  insetBlockStart: "var(--ds-space-400, 32px)",
  insetInlineStart: 0,
  /**
   * The fallback value of 0 ensures that the button is hidden by default,
   * unless some parent (or the button itself) overrides it.
   */
  opacity: "var(--ds--resize-button--opacity,0)",
  outline: 0,
  transform: 'translateX(-50%)',
  transition: "\n    background-color ".concat(_durations.smallDurationMs, "ms linear,\n    color ").concat(_durations.smallDurationMs, "ms linear,\n    opacity ").concat(_durations.mediumDurationMs, "ms ").concat(_curves.easeOut, "\n  "),
  ':hover': {
    backgroundColor: "var(--ds-background-selected-bold, ".concat(_colors.B100, ")"),
    color: "var(--ds-text-inverse, ".concat(_colors.N0, ")"),
    opacity: 1
  },
  ':active, :focus': {
    backgroundColor: "var(--ds-background-selected-bold-hovered, ".concat(_colors.B200, ")"),
    color: "var(--ds-text-inverse, ".concat(_colors.N0, ")"),
    opacity: 1
  }
});
var resizeIconButtonExpandedStyles = (0, _react.css)({
  transform: 'rotate(180deg)',
  transformOrigin: 7
});
var preventDefault = function preventDefault(event) {
  return event.preventDefault();
};
var cssSelector = (0, _defineProperty2.default)({}, _constants.RESIZE_BUTTON_SELECTOR, true);
var ResizeButton = function ResizeButton(_ref) {
  var isLeftSidebarCollapsed = _ref.isLeftSidebarCollapsed,
    label = _ref.label,
    testId = _ref.testId,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (0, _react.jsx)("button", (0, _extends2.default)({}, cssSelector, {
    // DO NOT remove. used as a CSS selector.
    "aria-expanded": !isLeftSidebarCollapsed,
    "aria-label": label,
    type: "button",
    css: [resizeIconButtonStyles, (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.responsive-page-layout-left-sidebar_p8r7g') && mobileStyles, !isLeftSidebarCollapsed && resizeIconButtonExpandedStyles],
    "data-testid": testId
    // Prevents focus staying attached to the button when pressed
    ,
    onMouseDown: preventDefault
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, props), (0, _react.jsx)(_chevronRight.default, {
    label: ""
  }), (0, _react.jsx)("div", {
    css: increaseHitAreaStyles
  }));
};

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = ResizeButton;
