// @atlaskit/drawer/dist/es2019/components/primitives/index.js
import _extends from "@babel/runtime/helpers/extends";
import React, { useCallback, useRef } from 'react';
import { IconButton } from '@atlaskit/button/new';
import ArrowLeft from '@atlaskit/icon/core/migration/arrow-left';
import { ExitingPersistence, SlideIn } from '@atlaskit/motion';
import { createExtender } from '../utils';
import ContentOverrides from './content';
import DrawerWrapper from './drawer-wrapper';
import FocusLock from './focus-lock';
import SidebarOverrides from './sidebar';

// Misc.
// ------------------------------

const defaults = {
  Sidebar: SidebarOverrides,
  Content: ContentOverrides
};

/**
 * This wrapper is used to specify separate durations for enter and exit.
 */
const CustomSlideIn = ({
  children,
  onFinish,
  enterFrom
}) => {
  return /*#__PURE__*/React.createElement(SlideIn, {
    animationTimingFunction: "ease-out",
    duration: "small",
    enterFrom: enterFrom,
    exitTo: enterFrom,
    fade: "none",
    onFinish: onFinish
  }, children);
};
const DrawerPrimitive = ({
  children,
  icon: Icon,
  closeLabel = 'Close drawer',
  scrollContentLabel,
  onClose,
  onCloseComplete,
  onOpenComplete,
  overrides,
  testId,
  in: isOpen,
  shouldReturnFocus,
  autoFocusFirstElem,
  isFocusLockEnabled,
  width,
  label,
  titleId,
  enterFrom = 'left'
}) => {
  const getOverrides = createExtender(defaults, overrides);
  const {
    component: Sidebar,
    ...sideBarOverrides
  } = getOverrides('Sidebar');
  const {
    component: Content,
    ...contentOverrides
  } = getOverrides('Content');

  /**
   * A ref to point to our wrapper, passed to `onCloseComplete` and `onOpenComplete` callbacks.
   */
  const drawerRef = useRef(null);
  const onFinish = useCallback(state => {
    if (state === 'entering') {
      onOpenComplete === null || onOpenComplete === void 0 ? void 0 : onOpenComplete(drawerRef.current);
    } else if (state === 'exiting') {
      onCloseComplete === null || onCloseComplete === void 0 ? void 0 : onCloseComplete(drawerRef.current);
    }
  }, [onCloseComplete, onOpenComplete]);
  return /*#__PURE__*/React.createElement(ExitingPersistence, {
    appear: true
  }, isOpen && /*#__PURE__*/React.createElement(CustomSlideIn, {
    onFinish: onFinish,
    enterFrom: enterFrom
  }, ({
    className
  }) => /*#__PURE__*/React.createElement(FocusLock, {
    autoFocusFirstElem: autoFocusFirstElem,
    isFocusLockEnabled: isFocusLockEnabled,
    shouldReturnFocus: shouldReturnFocus
  }, /*#__PURE__*/React.createElement(DrawerWrapper
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
  , {
    className: className,
    width: width,
    testId: testId,
    drawerRef: drawerRef,
    label: label,
    titleId: titleId
  }, /*#__PURE__*/React.createElement(Sidebar, sideBarOverrides, /*#__PURE__*/React.createElement(IconButton, {
    onClick: onClose,
    testId: testId && 'DrawerPrimitiveSidebarCloseButton',
    icon: Icon ? iconProps => /*#__PURE__*/React.createElement(Icon, _extends({}, iconProps, {
      size: "large",
      LEGACY_size: "large"
    })) : ArrowLeft,
    label: closeLabel,
    shape: "circle",
    appearance: "subtle"
  })), /*#__PURE__*/React.createElement(Content, _extends({
    scrollContentLabel: scrollContentLabel
  }, contentOverrides), children)))));
};

// eslint-disable-next-line @repo/internal/react/require-jsdoc
export default DrawerPrimitive;
