// @atlaskit/page-layout/dist/cjs/components/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Banner", {
  enumerable: true,
  get: function get() {
    return _bannerSlot.default;
  }
});
Object.defineProperty(exports, "Content", {
  enumerable: true,
  get: function get() {
    return _content.default;
  }
});
Object.defineProperty(exports, "LeftPanel", {
  enumerable: true,
  get: function get() {
    return _leftPanel.default;
  }
});
Object.defineProperty(exports, "LeftSidebar", {
  enumerable: true,
  get: function get() {
    return _leftSidebar.default;
  }
});
Object.defineProperty(exports, "LeftSidebarWithoutResize", {
  enumerable: true,
  get: function get() {
    return _leftSidebarWithoutResize.default;
  }
});
Object.defineProperty(exports, "Main", {
  enumerable: true,
  get: function get() {
    return _main.default;
  }
});
Object.defineProperty(exports, "PageLayout", {
  enumerable: true,
  get: function get() {
    return _pageLayout.default;
  }
});
Object.defineProperty(exports, "ResizeControl", {
  enumerable: true,
  get: function get() {
    return _resizeControl.default;
  }
});
Object.defineProperty(exports, "RightPanel", {
  enumerable: true,
  get: function get() {
    return _rightPanel.default;
  }
});
Object.defineProperty(exports, "RightSidebar", {
  enumerable: true,
  get: function get() {
    return _rightSidebar.default;
  }
});
Object.defineProperty(exports, "TopNavigation", {
  enumerable: true,
  get: function get() {
    return _topNavigation.default;
  }
});
Object.defineProperty(exports, "useCustomSkipLink", {
  enumerable: true,
  get: function get() {
    return _skipLinks.useCustomSkipLink;
  }
});
var _pageLayout = _interopRequireDefault(require("./slots/page-layout"));
var _main = _interopRequireDefault(require("./slots/main"));
var _rightPanel = _interopRequireDefault(require("./slots/right-panel"));
var _leftPanel = _interopRequireDefault(require("./slots/left-panel"));
var _rightSidebar = _interopRequireDefault(require("./slots/right-sidebar"));
var _leftSidebar = _interopRequireDefault(require("./slots/left-sidebar"));
var _leftSidebarWithoutResize = _interopRequireDefault(require("./slots/left-sidebar-without-resize"));
var _bannerSlot = _interopRequireDefault(require("./slots/banner-slot"));
var _topNavigation = _interopRequireDefault(require("./slots/top-navigation"));
var _content = _interopRequireDefault(require("./slots/content"));
var _resizeControl = _interopRequireDefault(require("./resize-control"));
var _skipLinks = require("./skip-links");
