// @atlaskit/progress-indicator/dist/cjs/components/indicator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PresentationalIndicator = exports.ButtonIndicator = void 0;
var _react = _interopRequireDefault(require("react"));
var _primitives = require("@atlaskit/primitives");
var _visuallyHidden = _interopRequireDefault(require("@atlaskit/visually-hidden"));
var _constants = require("./constants");
// TODO Token usages are not consistent for dots https://product-fabric.atlassian.net/browse/DSP-3180
var colorMap = {
  default: (0, _primitives.xcss)({
    backgroundColor: 'color.background.neutral'
  }),
  help: (0, _primitives.xcss)({
    backgroundColor: 'color.background.neutral'
  }),
  inverted: (0, _primitives.xcss)({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-subtle, #626F86)"
  }),
  primary: (0, _primitives.xcss)({
    backgroundColor: 'color.background.neutral'
  })
};
var selectedColorMap = {
  default: (0, _primitives.xcss)({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon, #44546F)"
  }),
  help: (0, _primitives.xcss)({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-discovery, #8270DB)"
  }),
  inverted: (0, _primitives.xcss)({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-inverse, #FFFFFF)"
  }),
  primary: (0, _primitives.xcss)({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-brand, #0C66E4)"
  })
};
var commonStyles = (0, _primitives.xcss)({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  width: "var(".concat(_constants.varDotsSize, ")"),
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  height: "var(".concat(_constants.varDotsSize, ")"),
  position: 'relative',
  borderRadius: 'border.radius.circle',
  '::before': {
    display: 'block',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    width: "calc(var(".concat(_constants.varDotsSize, ") + var(").concat(_constants.varDotsMargin, "))"),
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    height: "calc(var(".concat(_constants.varDotsSize, ") + var(").concat(_constants.varDotsMargin, "))"),
    position: 'absolute',
    content: '""',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    insetBlockStart: "calc(-1 * var(".concat(_constants.varDotsMargin, ") / 2)"),
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    insetInlineStart: "calc(-1 * var(".concat(_constants.varDotsMargin, ") / 2)")
  }
});
var buttonStyles = (0, _primitives.xcss)({
  padding: 'space.0',
  border: 0,
  cursor: 'pointer',
  outline: 0
});
/**
 * __Presentational indicator__
 *
 * A presentational indicator with no interactivity
 */
var PresentationalIndicator = exports.PresentationalIndicator = function PresentationalIndicator(_ref) {
  var appearance = _ref.appearance,
    isSelected = _ref.isSelected,
    testId = _ref.testId;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    testId: testId,
    xcss: [commonStyles, appearance === 'default' && !isSelected && colorMap['default'], appearance === 'help' && !isSelected && colorMap['help'], appearance === 'inverted' && !isSelected && colorMap['inverted'], appearance === 'primary' && !isSelected && colorMap['primary'], appearance === 'default' && isSelected && selectedColorMap['default'], appearance === 'help' && isSelected && selectedColorMap['help'], appearance === 'inverted' && isSelected && selectedColorMap['inverted'], appearance === 'primary' && isSelected && selectedColorMap['primary']]
  });
};
/**
 * __Button indicator__
 *
 * An interactive indicator.
 */
var ButtonIndicator = exports.ButtonIndicator = function ButtonIndicator(_ref2) {
  var appearance = _ref2.appearance,
    panelId = _ref2.panelId,
    tabId = _ref2.tabId,
    isSelected = _ref2.isSelected,
    onClick = _ref2.onClick,
    testId = _ref2.testId;
  return /*#__PURE__*/_react.default.createElement(_primitives.Pressable, {
    role: "tab",
    xcss: [commonStyles, buttonStyles, appearance === 'default' && !isSelected && colorMap['default'], appearance === 'help' && !isSelected && colorMap['help'], appearance === 'inverted' && !isSelected && colorMap['inverted'], appearance === 'primary' && !isSelected && colorMap['primary'], appearance === 'default' && isSelected && selectedColorMap['default'], appearance === 'help' && isSelected && selectedColorMap['help'], appearance === 'inverted' && isSelected && selectedColorMap['inverted'], appearance === 'primary' && isSelected && selectedColorMap['primary']],
    "aria-controls": panelId,
    "aria-selected": isSelected,
    id: tabId,
    onClick: onClick,
    tabIndex: isSelected ? -1 : undefined,
    testId: testId
  }, /*#__PURE__*/_react.default.createElement(_visuallyHidden.default, null, tabId));
};
