// @atlaskit/table-tree/dist/cjs/components/header.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _commonCell = _interopRequireDefault(require("./internal/common-cell"));
var _withColumnWidth = _interopRequireDefault(require("./internal/with-column-width"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /**
 * @jsxRuntime classic
 * @jsx jsx
 */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var headerStyles = (0, _react2.css)({
  color: "var(--ds-text-subtle, ".concat(_colors.N300, ")"),
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  fontWeight: "var(--ds-font-weight-bold, 700)"
});

/**
 * This is hard-coded here because our actual <Header /> has no typings
 * for its props.
 *
 * Adding types for real *might* break things so will need a little care.
 *
 * Defining it here for now lets us provide *something* without much headache.
 */
var HeaderComponent = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(HeaderComponent, _Component);
  var _super = _createSuper(HeaderComponent);
  function HeaderComponent() {
    (0, _classCallCheck2.default)(this, HeaderComponent);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(HeaderComponent, [{
    key: "render",
    value: function render() {
      var props = this.props;
      return (
        // TODO: Determine whether proper `th` elements can be used instead of
        // roles (DSP-11588)
        (0, _react2.jsx)(_commonCell.default, (0, _extends2.default)({
          css: headerStyles,
          role: "columnheader",
          style: {
            width: props.width
          }
        }, props), props.children)
      );
    }
  }]);
  return HeaderComponent;
}(_react.Component);
var Header = (0, _withColumnWidth.default)(HeaderComponent);

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = Header;
