// @atlaskit/table-tree/dist/cjs/components/headers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = require("react");
var _react2 = require("@emotion/react");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /**
 * @jsxRuntime classic
 * @jsx jsx
 */ /* eslint-disable @repo/internal/react/no-clone-element */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var containerStyles = (0, _react2.css)({
  display: 'flex',
  borderBlockEnd: "solid 2px ".concat("var(--ds-border, #dfe1e6)")
});
var Headers = exports.default = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(Headers, _Component);
  var _super = _createSuper(Headers);
  function Headers() {
    (0, _classCallCheck2.default)(this, Headers);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(Headers, [{
    key: "render",
    value: function render() {
      return (
        // TODO: Determine whether proper `tr` elements can be used instead of
        // roles (DSP-11588)
        (0, _react2.jsx)("div", {
          css: containerStyles,
          role: "row"
        }, _react.Children.map(this.props.children, function (header, index) {
          return (
            /*#__PURE__*/
            // eslint-disable-next-line react/no-array-index-key
            (0, _react.cloneElement)(header, {
              key: index,
              columnIndex: index
            })
          );
        }))
      );
    }
  }]);
  return Headers;
}(_react.Component);
