// @atlaskit/table-tree/dist/cjs/components/row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RowWithoutAnalytics = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _react2 = require("@emotion/react");
var _analyticsNext = require("@atlaskit/analytics-next");
var _toItemId = _interopRequireDefault(require("../utils/to-item-id"));
var _chevron = _interopRequireDefault(require("./internal/chevron"));
var _styled = require("./internal/styled");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /**
 * @jsxRuntime classic
 * @jsx jsx
 */ /* eslint-disable @repo/internal/react/no-clone-element */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var treeRowClickableStyles = (0, _react2.css)({
  cursor: 'pointer'
});
var packageName = "@atlaskit/table-tree";
var packageVersion = "9.12.2";

/**
 * This is hard-coded here because our actual <TableTree /> has no typings
 * for its props.
 *
 * Adding types for real *might* break things so will need a little care.
 *
 * Defining it here for now lets us provide *something* without much headache.
 */
var RowComponent = exports.RowWithoutAnalytics = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(RowComponent, _Component);
  var _super = _createSuper(RowComponent);
  function RowComponent() {
    var _this;
    (0, _classCallCheck2.default)(this, RowComponent);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      isExpanded: _this.props.isDefaultExpanded || false
    });
    /**
     * This ensures a user won't trigger a click event and expand the accordion
     * when making a text selection.
     */
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onClickHandler", function (e) {
      var _window$getSelection;
      var selection = ((_window$getSelection = window.getSelection()) === null || _window$getSelection === void 0 ? void 0 : _window$getSelection.toString()) || '';
      if ((selection === null || selection === void 0 ? void 0 : selection.length) === 0) {
        _this.onExpandToggle();
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onExpandToggle", function () {
      var isExpanded = _this.props.isExpanded;
      if (isExpanded !== undefined) {
        _this.onExpandStateChange(!isExpanded);
      } else {
        _this.setState({
          isExpanded: !_this.state.isExpanded
        });
        _this.onExpandStateChange(!_this.state.isExpanded);
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "getExtendedLabel", function (cellContent, cellIndex, mainColumnForExpandCollapseLabel) {
      /**
       * First condition - when we pass data via `items` property in `<TableTree />`
       * Second condition - when we pass data via `<Rows />` as children in `<TableTree />`.
       */
      if (cellContent.hasOwnProperty('props')) {
        return cellContent === null || cellContent === void 0 ? void 0 : cellContent.props[mainColumnForExpandCollapseLabel === null || mainColumnForExpandCollapseLabel === void 0 ? void 0 : mainColumnForExpandCollapseLabel.toLowerCase()];
      } else if (cellIndex === mainColumnForExpandCollapseLabel) {
        return cellContent;
      }
      return undefined;
    });
    return _this;
  }
  (0, _createClass2.default)(RowComponent, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props = this.props,
        isDefaultExpanded = _this$props.isDefaultExpanded,
        isExpanded = _this$props.isExpanded;
      if (isExpanded === undefined && isDefaultExpanded !== undefined && prevProps.isDefaultExpanded !== isDefaultExpanded && this.state.isExpanded !== isDefaultExpanded) {
        // eslint-disable-next-line react/no-did-update-set-state
        this.setState({
          isExpanded: isDefaultExpanded
        });
      }
    }
  }, {
    key: "onExpandStateChange",
    value: function onExpandStateChange(isExpanded) {
      if (this.props.data) {
        if (isExpanded && this.props.onExpand) {
          this.props.onExpand(this.props.data);
        } else if (!isExpanded && this.props.onCollapse) {
          this.props.onCollapse(this.props.data);
        }
      }
    }
  }, {
    key: "isExpanded",
    value: function isExpanded() {
      var isExpanded = this.props.isExpanded;
      return isExpanded !== undefined ? isExpanded : this.state.isExpanded;
    }
  }, {
    key: "renderCell",
    value: function renderCell(cell, cellIndex) {
      var props = this.props;
      var isExpanded = this.isExpanded();
      var hasChildren = props.hasChildren,
        depth = props.depth,
        mainColumnForExpandCollapseLabel = props.mainColumnForExpandCollapseLabel;
      var isFirstCell = cellIndex === 0;
      var indentLevel = isFirstCell ? depth : 0;
      var cellContent = cell.props.children || [];
      var extendedLabel = this.getExtendedLabel(cellContent, cellIndex, mainColumnForExpandCollapseLabel);
      if (isFirstCell && hasChildren) {
        cellContent = [(0, _react2.jsx)(_chevron.default, {
          key: "chevron",
          expandLabel: props.expandLabel,
          collapseLabel: props.collapseLabel,
          extendedLabel: extendedLabel,
          isExpanded: isExpanded,
          onExpandToggle: this.onExpandToggle,
          ariaControls: isExpanded ? (0, _toItemId.default)(props.itemId) : undefined,
          rowId: props.itemId
        })].concat(cellContent);
      }
      return /*#__PURE__*/_react.default.cloneElement(cell, {
        key: cellIndex,
        columnIndex: cellIndex,
        indentLevel: indentLevel
      }, cellContent);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;
      var _this$props2 = this.props,
        shouldExpandOnClick = _this$props2.shouldExpandOnClick,
        hasChildren = _this$props2.hasChildren,
        depth = _this$props2.depth,
        renderChildren = _this$props2.renderChildren;
      var isExpanded = this.isExpanded();
      var ariaAttrs = {};
      if (hasChildren) {
        ariaAttrs['aria-expanded'] = isExpanded;
      }
      if (depth !== undefined) {
        ariaAttrs['aria-level'] = depth;
      }
      return (0, _react2.jsx)(_react.Fragment, null, (0, _react2.jsx)(_styled.TreeRowContainer, (0, _extends2.default)({
        css: hasChildren && shouldExpandOnClick ? treeRowClickableStyles : undefined,
        onClick: hasChildren && shouldExpandOnClick ? this.onClickHandler : undefined
      }, ariaAttrs), _react.default.Children.map(this.props.children, function (cell, index) {
        return _this2.renderCell(cell, index);
      })), hasChildren && isExpanded && renderChildren && renderChildren());
    }
  }]);
  return RowComponent;
}(_react.Component);
var createAndFireEventOnAtlaskit = (0, _analyticsNext.createAndFireEvent)('atlaskit');
var Row = (0, _analyticsNext.withAnalyticsContext)({
  componentName: 'row',
  packageName: packageName,
  packageVersion: packageVersion
})((0, _analyticsNext.withAnalyticsEvents)({
  onExpand: createAndFireEventOnAtlaskit({
    action: 'expanded',
    actionSubject: 'tableTree',
    attributes: {
      componentName: 'row',
      packageName: packageName,
      packageVersion: packageVersion
    }
  }),
  onCollapse: createAndFireEventOnAtlaskit({
    action: 'collapsed',
    actionSubject: 'tableTree',
    attributes: {
      componentName: 'row',
      packageName: packageName,
      packageVersion: packageVersion
    }
  })
})(RowComponent));

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = Row;
