// @atlaskit/table-tree/dist/cjs/components/table-tree.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _wrapNativeSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/wrapNativeSuper"));
var _react = _interopRequireWildcard(require("react"));
var _cell = _interopRequireDefault(require("./cell"));
var _header = _interopRequireDefault(require("./header"));
var _headers = _interopRequireDefault(require("./headers"));
var _context = require("./internal/context");
var _row = _interopRequireDefault(require("./row"));
var _rows = _interopRequireDefault(require("./rows"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /* eslint-disable react/prop-types */
var Content = /*#__PURE__*/function (_Object) {
  (0, _inherits2.default)(Content, _Object);
  var _super = _createSuper(Content);
  function Content() {
    (0, _classCallCheck2.default)(this, Content);
    return _super.apply(this, arguments);
  }
  return (0, _createClass2.default)(Content);
}( /*#__PURE__*/(0, _wrapNativeSuper2.default)(Object));
/**
 * This is hard-coded here because our actual <TableTree /> has no typings
 * for its props.
 *
 * Adding types for real *might* break things so will need a little care.
 *
 * Defining it here for now lets us provide *something* without much headache.
 */
var TableTree = exports.default = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(TableTree, _Component);
  var _super2 = _createSuper(TableTree);
  function TableTree() {
    var _this;
    (0, _classCallCheck2.default)(this, TableTree);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super2.call.apply(_super2, [this].concat(args));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      columnWidths: []
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "setColumnWidth", function (columnIndex, width) {
      var columnWidths = _this.state.columnWidths;
      if (width === columnWidths[columnIndex]) {
        return;
      }
      columnWidths[columnIndex] = width;
      _this.setState({
        columnWidths: columnWidths
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "getColumnWidth", function (columnIndex) {
      return _this.state && _this.state.columnWidths[columnIndex] || null;
    });
    return _this;
  }
  (0, _createClass2.default)(TableTree, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var widths = this.props.columnWidths;
      if (widths) {
        this.setState({
          columnWidths: widths
        }); // eslint-disable-line
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
        items = _this$props.items,
        label = _this$props.label,
        referencedLabel = _this$props.referencedLabel,
        shouldExpandOnClick = _this$props.shouldExpandOnClick,
        headers = _this$props.headers,
        columns = _this$props.columns,
        _this$props$columnWid = _this$props.columnWidths,
        columnWidths = _this$props$columnWid === void 0 ? [] : _this$props$columnWid,
        mainColumnForExpandCollapseLabel = _this$props.mainColumnForExpandCollapseLabel;
      var heads = headers && /*#__PURE__*/_react.default.createElement(_headers.default, null, headers.map(function (header, index) {
        return (
          /*#__PURE__*/
          // eslint-disable-next-line react/no-array-index-key
          _react.default.createElement(_header.default, {
            key: index,
            columnIndex: index,
            width: columnWidths[index]
          }, header)
        );
      }));
      var rows = null;
      if (columns && items) {
        rows = /*#__PURE__*/_react.default.createElement(_rows.default, {
          items: items,
          render: function render(_ref) {
            var id = _ref.id,
              children = _ref.children,
              hasChildren = _ref.hasChildren,
              content = _ref.content;
            return /*#__PURE__*/_react.default.createElement(_row.default, {
              itemId: id,
              items: children,
              hasChildren: hasChildren,
              shouldExpandOnClick: shouldExpandOnClick,
              mainColumnForExpandCollapseLabel: mainColumnForExpandCollapseLabel
            }, columns.map(function (CellContent, index) {
              return /*#__PURE__*/_react.default.createElement(_cell.default
              // eslint-disable-next-line react/no-array-index-key
              , {
                key: index,
                columnIndex: index,
                width: columnWidths[index]
              }, /*#__PURE__*/_react.default.createElement(CellContent, content));
            }));
          }
        });
      }
      return /*#__PURE__*/_react.default.createElement(_context.TableTreeContext.Provider, {
        value: {
          setColumnWidth: this.setColumnWidth,
          getColumnWidth: this.getColumnWidth
        }
      }, /*#__PURE__*/_react.default.createElement("div", {
        role: "treegrid",
        "aria-readonly": true,
        "aria-label": label,
        "aria-labelledby": referencedLabel
      }, heads, rows, this.props.children));
    }
  }]);
  return TableTree;
}(_react.Component);
