// @atlaskit/atlassian-navigation/dist/cjs/components/PrimarySplitButton/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrimarySplitButton = void 0;
var _react = require("@emotion/react");
var _unsafe = require("@atlaskit/button/unsafe");
/** @jsx jsx */

// Copied from packages/design-system/atlassian-navigation/src/components/PrimaryButton/index.tsx
var buttonBaseStyles = (0, _react.css)({
  display: 'flex',
  height: '100%',
  position: 'relative',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column'
});

// Mostly copied from packages/design-system/atlassian-navigation/src/components/PrimaryButton/index.tsx
var buttonHighlightedStyles = (0, _react.css)({
  '&:after': {
    height: 3,
    position: 'absolute',
    backgroundColor: "var(--ds-border-selected, #0052cc)",
    borderStartEndRadius: "var(--ds-border-radius-050, 1px)",
    borderStartStartRadius: "var(--ds-border-radius-050, 1px)",
    content: '""',
    insetBlockEnd: 0,
    insetInlineEnd: "var(--ds-space-050, 4px)",
    insetInlineStart: "var(--ds-space-050, 4px)"
  }
});
/**
 * TODO: Add JSDoc
 */
var PrimarySplitButton = exports.PrimarySplitButton = function PrimarySplitButton(_ref) {
  var children = _ref.children,
    _ref$isHighlighted = _ref.isHighlighted,
    isHighlighted = _ref$isHighlighted === void 0 ? false : _ref$isHighlighted;
  var _getActions = (0, _unsafe.UNSAFE_GET_ACTIONS)(children),
    PrimaryAction = _getActions.PrimaryAction,
    SecondaryAction = _getActions.SecondaryAction;
  return (0, _react.jsx)("div", {
    css: [buttonBaseStyles, isHighlighted && buttonHighlightedStyles]
  }, (0, _react.jsx)(_unsafe.UNSAFE_SPLIT_BUTTON_CONTAINER, null, (0, _react.jsx)(_unsafe.UNSAFE_SPLIT_BUTTON_CONTEXT.Provider, {
    value: {
      appearance: 'navigation',
      isHighlighted: isHighlighted
    }
  }, PrimaryAction, (0, _react.jsx)(_unsafe.UNSAFE_DIVIDER, {
    appearance: "navigation",
    spacing: "default",
    isDisabled: false
  }), SecondaryAction)));
};
