// @atlaskit/atlassian-navigation/dist/es2019/components/Create/skeleton.js
/** @jsx jsx */
import { Fragment } from 'react';
import { css, jsx } from '@emotion/react';
import { CREATE_BREAKPOINT } from '../../common/constants';
import { useTheme } from '../../theme';
import { IconButtonSkeleton } from '../IconButton/skeleton';
const buttonHeight = "var(--ds-space-400, 32px)";
const skeletonStyles = css({
  width: 68,
  height: buttonHeight,
  borderRadius: "var(--ds-border-radius, 3px)",
  marginInlineStart: "var(--ds-space-150, 12px)",
  opacity: 0.15
});
const mobileStyles = css({
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  [`@media (max-width: ${CREATE_BREAKPOINT - 1}px)`]: {
    display: 'none !important'
  }
});

// Not exported to consumers, only used in NavigationSkeleton
// eslint-disable-next-line @repo/internal/react/require-jsdoc
export const CreateSkeleton = () => {
  const theme = useTheme();
  return jsx(Fragment, null, jsx("div", {
    style: theme.mode.skeleton,
    css: [skeletonStyles, mobileStyles]
  }), jsx(IconButtonSkeleton, {
    css: mobileStyles,
    size: 26
  }));
};
