// @atlaskit/atlassian-navigation/dist/es2019/components/PrimaryDropdownButton/index.js
import _extends from "@babel/runtime/helpers/extends";
/** @jsx jsx */
import { forwardRef } from 'react';
import { jsx } from '@emotion/react';
import ChevronIcon from '@atlaskit/icon/glyph/chevron-down';
import { getBooleanFF } from '@atlaskit/platform-feature-flags';
import { Bleed, xcss } from '@atlaskit/primitives';
import { PrimaryButton } from '../PrimaryButton';
const chevronIconStyles = xcss({
  opacity: 0.51
});
const chevronIconStylesWithSpacingFixStyles = xcss({
  opacity: 0.51,
  marginInlineEnd: `${"var(--ds-space-negative-050, -0.25rem)"}`
});

/**
 * __Primary dropdown button__
 *
 * A primary dropdown button allows you to add dropdown menus to the navigation.
 * Should be passed into `AtlassianNavigation`'s `primaryItems` prop.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples#button)
 * - [Code](https://atlassian.design/components/atlassian-navigation/code)
 */
export const PrimaryDropdownButton = /*#__PURE__*/forwardRef((props, ref) => {
  return jsx(PrimaryButton, _extends({
    iconAfter: jsx(Bleed, {
      xcss: getBooleanFF('platform.design-system-team.icon-button-spacing-fix_o1zc5') ? chevronIconStylesWithSpacingFixStyles : chevronIconStyles,
      inline: "space.100"
    }, jsx(ChevronIcon, {
      label: ""
    })),
    ref: ref
  }, props));
});
export default PrimaryDropdownButton;
