// @atlaskit/page-layout/dist/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BANNER_HEIGHT", {
  enumerable: true,
  get: function get() {
    return _constants.BANNER_HEIGHT;
  }
});
Object.defineProperty(exports, "Banner", {
  enumerable: true,
  get: function get() {
    return _components.Banner;
  }
});
Object.defineProperty(exports, "Content", {
  enumerable: true,
  get: function get() {
    return _components.Content;
  }
});
Object.defineProperty(exports, "LEFT_PANEL_WIDTH", {
  enumerable: true,
  get: function get() {
    return _constants.LEFT_PANEL_WIDTH;
  }
});
Object.defineProperty(exports, "LEFT_SIDEBAR_WIDTH", {
  enumerable: true,
  get: function get() {
    return _constants.LEFT_SIDEBAR_WIDTH;
  }
});
Object.defineProperty(exports, "LeftPanel", {
  enumerable: true,
  get: function get() {
    return _components.LeftPanel;
  }
});
Object.defineProperty(exports, "LeftSidebar", {
  enumerable: true,
  get: function get() {
    return _components.LeftSidebar;
  }
});
Object.defineProperty(exports, "LeftSidebarWithoutResize", {
  enumerable: true,
  get: function get() {
    return _components.LeftSidebarWithoutResize;
  }
});
Object.defineProperty(exports, "Main", {
  enumerable: true,
  get: function get() {
    return _components.Main;
  }
});
Object.defineProperty(exports, "PageLayout", {
  enumerable: true,
  get: function get() {
    return _components.PageLayout;
  }
});
Object.defineProperty(exports, "RIGHT_PANEL_WIDTH", {
  enumerable: true,
  get: function get() {
    return _constants.RIGHT_PANEL_WIDTH;
  }
});
Object.defineProperty(exports, "RIGHT_SIDEBAR_WIDTH", {
  enumerable: true,
  get: function get() {
    return _constants.RIGHT_SIDEBAR_WIDTH;
  }
});
Object.defineProperty(exports, "RightPanel", {
  enumerable: true,
  get: function get() {
    return _components.RightPanel;
  }
});
Object.defineProperty(exports, "RightSidebar", {
  enumerable: true,
  get: function get() {
    return _components.RightSidebar;
  }
});
Object.defineProperty(exports, "TOP_NAVIGATION_HEIGHT", {
  enumerable: true,
  get: function get() {
    return _constants.TOP_NAVIGATION_HEIGHT;
  }
});
Object.defineProperty(exports, "TopNavigation", {
  enumerable: true,
  get: function get() {
    return _components.TopNavigation;
  }
});
Object.defineProperty(exports, "useCustomSkipLink", {
  enumerable: true,
  get: function get() {
    return _components.useCustomSkipLink;
  }
});
Object.defineProperty(exports, "useLeftSidebarFlyoutLock", {
  enumerable: true,
  get: function get() {
    return _controllers.useLeftSidebarFlyoutLock;
  }
});
Object.defineProperty(exports, "usePageLayoutResize", {
  enumerable: true,
  get: function get() {
    return _controllers.usePageLayoutResize;
  }
});
var _components = require("./components");
var _constants = require("./common/constants");
var _controllers = require("./controllers");
