// @atlaskit/motion/dist/cjs/entering/slide-in.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.slideInAnimation = exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _keyframesMotion = _interopRequireDefault(require("./keyframes-motion"));
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var slideInAnimation = exports.slideInAnimation = function slideInAnimation(from, state, fade) {
  var initial = state === 'entering' ? '0%' : '100%';
  var end = state === 'entering' ? '100%' : '0%';
  var fromMap = {
    top: 'translate3d(0, -100%, 0)',
    right: 'translate3d(100%, 0, 0)',
    bottom: 'translate3d(0, 100%, 0)',
    left: 'translate3d(-100%, 0, 0)'
  };
  var animateOpacity = fade === 'inout' || fade === 'in' && state === 'entering' || fade === 'out' && state === 'exiting';
  return (0, _defineProperty2.default)((0, _defineProperty2.default)({}, initial, {
    transform: fromMap[from],
    opacity: animateOpacity ? 0 : 1
  }), end, {
    transform: 'none',
    opacity: 1
  });
};
/**
 * __SlideIn__
 *
 * Will slide an element into position, generally used for things that appear from outside of the viewport into view.
 *
 * See [examples](https://atlaskit.atlassian.com/packages/design-system/motion/docs/entering-motions).
 */
var SlideIn = function SlideIn(_ref2) {
  var children = _ref2.children,
    enterFrom = _ref2.enterFrom,
    exitTo = _ref2.exitTo,
    _ref2$fade = _ref2.fade,
    fade = _ref2$fade === void 0 ? 'none' : _ref2$fade,
    _ref2$duration = _ref2.duration,
    duration = _ref2$duration === void 0 ? 'medium' : _ref2$duration,
    isPaused = _ref2.isPaused,
    onFinish = _ref2.onFinish,
    _ref2$animationTiming = _ref2.animationTimingFunction,
    animationTimingFunction = _ref2$animationTiming === void 0 ? 'ease-out' : _ref2$animationTiming,
    _ref2$animationTiming2 = _ref2.animationTimingFunctionExiting,
    animationTimingFunctionExiting = _ref2$animationTiming2 === void 0 ? 'ease-in' : _ref2$animationTiming2;
  return /*#__PURE__*/_react.default.createElement(_keyframesMotion.default, {
    duration: duration,
    enteringAnimation: slideInAnimation(enterFrom, 'entering', fade),
    exitingAnimation: slideInAnimation(exitTo || enterFrom, 'exiting', fade),
    animationTimingFunction: animationTimingFunction,
    animationTimingFunctionExiting: animationTimingFunctionExiting,
    isPaused: isPaused,
    onFinish: onFinish
  }, children);
};
var _default = exports.default = SlideIn;
