// @atlaskit/onboarding/dist/esm/components/modal.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
var _excluded = ["text", "key"],
  _excluded2 = ["actions", "children", "heading"];
function _createSuper(t) { var r = _isNativeReflectConstruct(); return function () { var e, o = _getPrototypeOf(t); if (r) { var s = _getPrototypeOf(this).constructor; e = Reflect.construct(o, arguments, s); } else e = o.apply(this, arguments); return _possibleConstructorReturn(this, e); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import { Component } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { jsx } from '@emotion/react';
import Button, { Theme as ButtonTheme } from '@atlaskit/button/custom-theme-button';
import Modal, { ModalBody, useModal } from '@atlaskit/modal-dialog';
import { ModalBody as Body, ModalHeading as Heading, ModalActionContainer, ModalActionItem, ModalImage } from '../styled/modal';
import { modalButtonTheme } from './theme';

// TODO: DSP-1250 - use a composable API consistent with normal modal dialog
/**
 * __Benefits modal__
 *
 * A benefits modal explains the benefits of a significant new feature or experience change.
 *
 * - [Examples](https://atlassian.design/components/onboarding/benefits-modal/examples)
 * - [Code](https://atlassian.design/components/onboarding/benefits-modal/code)
 * - [Usage](https://atlassian.design/components/onboarding/benefits-modal/usage)
 */
var BenefitsModal = /*#__PURE__*/function (_Component) {
  _inherits(BenefitsModal, _Component);
  var _super = _createSuper(BenefitsModal);
  function BenefitsModal() {
    var _this;
    _classCallCheck(this, BenefitsModal);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    _defineProperty(_assertThisInitialized(_this), "headerComponent", function (props) {
      var HeaderElement = props.header,
        src = props.image;
      var ImageElement = function ImageElement() {
        return jsx(ModalImage, {
          src: src,
          alt: ""
        });
      };
      return HeaderElement || ImageElement;
    });
    _defineProperty(_assertThisInitialized(_this), "footerComponent", function (props) {
      var FooterElement = props.footer,
        actionList = props.actions,
        _props$experimental_s = props.experimental_shouldShowPrimaryButtonOnRight,
        experimental_shouldShowPrimaryButtonOnRight = _props$experimental_s === void 0 ? false : _props$experimental_s;
      var ActionsElement = function ActionsElement() {
        return actionList ? jsx(ButtonTheme.Provider, {
          value: modalButtonTheme
        }, jsx(ModalActionContainer, {
          shouldReverseButtonOrder: experimental_shouldShowPrimaryButtonOnRight
        }, actionList.map(function (_ref, idx) {
          var text = _ref.text,
            key = _ref.key,
            rest = _objectWithoutProperties(_ref, _excluded);
          var variant = idx ? 'subtle-link' : 'primary';
          return jsx(ModalActionItem, {
            key: key || (typeof text === 'string' ? text : "".concat(idx))
          }, jsx(Button, _extends({
            appearance: variant,
            autoFocus: !idx
          }, rest), text));
        }))) : null;
      };
      return FooterElement || ActionsElement;
    });
    return _this;
  }
  _createClass(BenefitsModal, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        actions = _this$props.actions,
        children = _this$props.children,
        heading = _this$props.heading,
        props = _objectWithoutProperties(_this$props, _excluded2);
      var Header = this.headerComponent(this.props);
      var Footer = this.footerComponent(this.props);
      var CustomHeader = function CustomHeader() {
        var _useModal = useModal(),
          titleId = _useModal.titleId;
        return jsx(Heading, {
          id: titleId
        }, heading);
      };
      return jsx(Modal, _extends({
        autoFocus: true,
        shouldScrollInViewport: true,
        shouldCloseOnOverlayClick: false,
        shouldCloseOnEscapePress: false
      }, props), jsx(Header, null), jsx(ModalBody, null, jsx(Body, null, heading && jsx(CustomHeader, null), children)), jsx(Footer, null));
    }
  }]);
  return BenefitsModal;
}(Component);
export { BenefitsModal as default };
