// @atlaskit/table-tree/dist/cjs/components/internal/common-cell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _styled = require("./styled");
var _excluded = ["indent", "width"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var commonStyles = (0, _react.css)({
  display: 'flex',
  boxSizing: 'border-box',
  minHeight: 40,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  padding: "var(--ds-space-100, 10px)".concat(" ", _styled.indentBase, " ", "var(--ds-space-100, 10px)", " var(--indent, ").concat(_styled.indentBase, ")"),
  position: 'relative',
  alignItems: 'center',
  color: "var(--ds-text, ".concat(_colors.N800, ")"),
  font: "var(--ds-font-body, normal 400 14px/20px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  hyphens: 'auto',
  wordBreak: 'break-word'
});
/**
 * __Common cell__
 */
var CommonCell = function CommonCell(_ref) {
  var indent = _ref.indent,
    width = _ref.width,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (0, _react.jsx)("div", (0, _extends2.default)({
    role: "gridcell"
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, props, {
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop, @atlaskit/ui-styling-standard/no-imported-style-values -- Ignored via go/DSP-18766
    style: {
      '--indent': indent,
      width: width
    },
    css: commonStyles
  }));
};
var _default = exports.default = CommonCell;
