// @atlaskit/table-tree/dist/cjs/components/internal/items.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _item = _interopRequireDefault(require("./item"));
var _loaderItem = _interopRequireDefault(require("./loader-item"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /* eslint-disable react/prop-types */
var Items = exports.default = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(Items, _Component);
  var _super = _createSuper(Items);
  function Items() {
    var _this;
    (0, _classCallCheck2.default)(this, Items);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      isLoaderShown: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleLoaderComplete", function () {
      _this.setState({
        isLoaderShown: false
      });
    });
    return _this;
  }
  (0, _createClass2.default)(Items, [{
    key: "renderLoader",
    value: function renderLoader() {
      var _this$props = this.props,
        depth = _this$props.depth,
        items = _this$props.items,
        loadingLabel = _this$props.loadingLabel;
      return /*#__PURE__*/_react.default.createElement(_loaderItem.default, {
        isCompleting: !!(items && items.length),
        onComplete: this.handleLoaderComplete,
        depth: depth + 1,
        loadingLabel: loadingLabel
      });
    }
  }, {
    key: "renderItems",
    value: function renderItems() {
      var _this$props2 = this.props,
        render = _this$props2.render,
        items = _this$props2.items,
        loadingLabel = _this$props2.loadingLabel,
        _this$props2$depth = _this$props2.depth,
        depth = _this$props2$depth === void 0 ? 0 : _this$props2$depth;
      return items && items.map(function (itemData, index) {
        return /*#__PURE__*/_react.default.createElement(_item.default, {
          data: itemData,
          depth: depth + 1,
          key: itemData && itemData.id || index,
          render: render,
          loadingLabel: loadingLabel
        });
      });
    }
  }, {
    key: "render",
    value: function render() {
      var isLoaderShown = this.state.isLoaderShown;
      return isLoaderShown ? this.renderLoader() : this.renderItems();
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(nextProps, prevState) {
      if (!nextProps.items && !prevState.isLoaderShown) {
        return {
          isLoaderShown: true
        };
      }
      return null;
    }
  }]);
  return Items;
}(_react.Component);
(0, _defineProperty2.default)(Items, "defaultProps", {
  depth: 0
});
