// @atlaskit/table-tree/dist/cjs/components/internal/item.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _toItemId = _interopRequireDefault(require("../../utils/to-item-id"));
var _items = _interopRequireDefault(require("./items"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /**
 * @jsxRuntime classic
 * @jsx jsx
 */ /* eslint-disable @repo/internal/react/no-clone-element */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var Item = exports.default = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(Item, _Component);
  var _super = _createSuper(Item);
  function Item() {
    (0, _classCallCheck2.default)(this, Item);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(Item, [{
    key: "render",
    value: function render() {
      // eslint-disable-next-line react/prop-types
      var _this$props = this.props,
        depth = _this$props.depth,
        data = _this$props.data,
        render = _this$props.render,
        loadingLabel = _this$props.loadingLabel;
      var renderedRow = render(data);
      if (!renderedRow) {
        return null;
      }
      var _renderedRow$props = renderedRow.props,
        itemId = _renderedRow$props.itemId,
        items = _renderedRow$props.items;
      return /*#__PURE__*/(0, _react.cloneElement)(renderedRow, {
        depth: depth,
        data: data,
        loadingLabel: loadingLabel,
        renderChildren: function renderChildren() {
          return (0, _react2.jsx)("div", {
            id: (0, _toItemId.default)(itemId)
          }, (0, _react2.jsx)(_items.default, {
            parentData: data,
            depth: depth,
            items: items,
            render: render,
            loadingLabel: loadingLabel
          }));
        }
      });
    }
  }]);
  return Item;
}(_react.Component);
(0, _defineProperty2.default)(Item, "defaultProps", {
  depth: 0
});
