// @atlaskit/table-tree/dist/cjs/components/internal/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.indentBase = exports.TreeRowContainer = exports.LoaderItemContainer = exports.ChevronContainer = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _excluded = ["isRoot"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var indentBase = exports.indentBase = "var(--ds-space-300, 25px)";
var treeRowContainerStyles = (0, _react.css)({
  display: 'flex',
  borderBlockEnd: "1px solid ".concat("var(--ds-border, ".concat(_colors.N30, ")"))
});

/**
 * __Tree row container__
 */
var TreeRowContainer = exports.TreeRowContainer = function TreeRowContainer(props) {
  return (0, _react.jsx)("div", (0, _extends2.default)({
    role: "row",
    css: treeRowContainerStyles
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, props));
};
var commonCellElementStyles = (0, _react.css)({
  display: 'flex',
  position: 'absolute',
  alignItems: 'center',
  // indentBase is re-used elsewhere and is primarily used as positive value; we need to negate it here
  marginInlineStart: "calc(".concat(indentBase, " * -1)")
});
var commonChevronContainerStyles = (0, _react.css)({
  // Aligns position:absolute chevron button with the adjacent text. Any future visual breaking changes
  // should consider setting this to `-2px` for better alignment, or refactor completely
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  marginBlockStart: -3
});
/**
 * __Chevron container__
 *
 * A wrapper container around the expand table tree button.
 */
var ChevronContainer = exports.ChevronContainer = function ChevronContainer(props) {
  return (0, _react.jsx)("span", (0, _extends2.default)({}, props, {
    css: [commonCellElementStyles, commonChevronContainerStyles]
  }));
};
var loadingItemContainerStyles = (0, _react.css)({
  width: '100%'
});
var paddingLeftStyles = (0, _react.css)({
  // eslint-disable-next-line @atlaskit/design-system/use-tokens-space
  paddingInlineStart: '50%'
});
/**
 * __Loader item container__
 *
 * A loader item container.
 */
var LoaderItemContainer = exports.LoaderItemContainer = function LoaderItemContainer(_ref) {
  var isRoot = _ref.isRoot,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (0, _react.jsx)("span", (0, _extends2.default)({
    css: [commonCellElementStyles, loadingItemContainerStyles, isRoot && paddingLeftStyles]
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
  }, props));
};
