// @atlaskit/table-tree/dist/esm/components/internal/chevron.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
/* eslint-disable @repo/internal/react/consistent-props-definitions */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { IconButton } from '@atlaskit/button/new';
import ChevronDownIcon from '@atlaskit/icon/glyph/chevron-down';
import ChevronRightIcon from '@atlaskit/icon/glyph/chevron-right';
import { ChevronContainer } from './styled';
var Chevron = /*#__PURE__*/function (_Component) {
  _inherits(Chevron, _Component);
  var _super = _createSuper(Chevron);
  function Chevron() {
    var _this;
    _classCallCheck(this, Chevron);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    _defineProperty(_assertThisInitialized(_this), "handleClick", function () {
      if (_this.props.onExpandToggle) {
        _this.props.onExpandToggle();
      }
    });
    return _this;
  }
  _createClass(Chevron, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        isExpanded = _this$props.isExpanded,
        ariaControls = _this$props.ariaControls,
        collapseLabel = _this$props.collapseLabel,
        expandLabel = _this$props.expandLabel,
        rowId = _this$props.rowId,
        extendedLabel = _this$props.extendedLabel;
      var getLabel = function getLabel(defaultLabel) {
        return extendedLabel ? "".concat(defaultLabel, " ").concat(extendedLabel, " row") : "".concat(defaultLabel, " row ").concat(rowId);
      };
      return /*#__PURE__*/React.createElement(ChevronContainer, null, /*#__PURE__*/React.createElement(IconButton, {
        appearance: "subtle",
        onClick: this.handleClick,
        spacing: "compact",
        icon: isExpanded ? ChevronDownIcon : ChevronRightIcon,
        "aria-controls": ariaControls,
        label: isExpanded ? getLabel(collapseLabel) : getLabel(expandLabel)
      }));
    }
  }]);
  return Chevron;
}(Component);
_defineProperty(Chevron, "defaultProps", {
  expandLabel: 'Expand',
  collapseLabel: 'Collapse'
});
export { Chevron as default };
