// @atlaskit/table-tree/dist/esm/components/internal/items.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import Item from './item';
import LoaderItem from './loader-item';
var Items = /*#__PURE__*/function (_Component) {
  _inherits(Items, _Component);
  var _super = _createSuper(Items);
  function Items() {
    var _this;
    _classCallCheck(this, Items);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    _defineProperty(_assertThisInitialized(_this), "state", {
      isLoaderShown: false
    });
    _defineProperty(_assertThisInitialized(_this), "handleLoaderComplete", function () {
      _this.setState({
        isLoaderShown: false
      });
    });
    return _this;
  }
  _createClass(Items, [{
    key: "renderLoader",
    value: function renderLoader() {
      var _this$props = this.props,
        depth = _this$props.depth,
        items = _this$props.items,
        loadingLabel = _this$props.loadingLabel;
      return /*#__PURE__*/React.createElement(LoaderItem, {
        isCompleting: !!(items && items.length),
        onComplete: this.handleLoaderComplete,
        depth: depth + 1,
        loadingLabel: loadingLabel
      });
    }
  }, {
    key: "renderItems",
    value: function renderItems() {
      var _this$props2 = this.props,
        render = _this$props2.render,
        items = _this$props2.items,
        loadingLabel = _this$props2.loadingLabel,
        _this$props2$depth = _this$props2.depth,
        depth = _this$props2$depth === void 0 ? 0 : _this$props2$depth;
      return items && items.map(function (itemData, index) {
        return /*#__PURE__*/React.createElement(Item, {
          data: itemData,
          depth: depth + 1,
          key: itemData && itemData.id || index,
          render: render,
          loadingLabel: loadingLabel
        });
      });
    }
  }, {
    key: "render",
    value: function render() {
      var isLoaderShown = this.state.isLoaderShown;
      return isLoaderShown ? this.renderLoader() : this.renderItems();
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(nextProps, prevState) {
      if (!nextProps.items && !prevState.isLoaderShown) {
        return {
          isLoaderShown: true
        };
      }
      return null;
    }
  }]);
  return Items;
}(Component);
_defineProperty(Items, "defaultProps", {
  depth: 0
});
export { Items as default };
