// @atlaskit/atlassian-navigation/dist/esm/components/SkeletonCreateButton/index.js
/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { useTheme } from '../../theme';
var skeletonCreateButtonStyles = css({
  height: 32,
  padding: "0 ".concat("var(--ds-space-150, 12px)"),
  alignSelf: 'center',
  border: 0,
  borderRadius: "var(--ds-border-radius, 3px)",
  fontSize: "var(--ds-font-size-100, 14px)",
  fontWeight: "var(--ds-font-weight-medium, 500)",
  lineHeight: 1,
  pointerEvents: 'none',
  ':focus, :active, :hover': {
    appearance: 'none',
    border: 0,
    outline: 0
  },
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
  '&&': {
    marginLeft: "var(--ds-space-150, 12px)"
  }
});

/**
 * __Skeleton create button__
 *
 * Skeleton buttons are lightweight HTML button elements with CSS that represent
 * their heavier interactive counterparts, for use when elements of the
 * navigation are loaded dynamically. This one represents the Create button.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples#skeleton-button)
 * - [Code](https://atlassian.design/components/atlassian-navigation/code)
 */
export var SkeletonCreateButton = function SkeletonCreateButton(_ref) {
  var text = _ref.text,
    testId = _ref.testId;
  var theme = useTheme();
  return jsx("button", {
    style: theme.mode.create.default,
    css: skeletonCreateButtonStyles,
    "data-testid": testId,
    type: "button"
  }, text);
};
