// @atlaskit/react-ufo/dist/es2019/vc/vc-observer-new/window-event-observer/index.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import { bind } from 'bind-event-listener';
import { fg } from '@atlaskit/platform-feature-flags';
export default class WindowEventObserver {
  constructor(opts) {
    _defineProperty(this, "unbindFns", []);
    this.onEvent = opts.onEvent;
  }
  bindEvent(type) {
    const unbindCallback = bind(window, {
      type,
      listener: event => {
        if (!event.isTrusted) {
          return;
        }
        this.onEvent({
          time: event.timeStamp,
          type,
          event
        });
      },
      options: {
        passive: true,
        once: true
      }
    });
    this.unbindFns.push(unbindCallback);
  }

  /**
   * Binds a scroll listener on `document` using the capture phase.
   * The `scroll` event does not bubble, so listening on `window` only catches
   * document-level scrolls. By using `{ capture: true }` on `document`, we can
   * detect scroll events from any element in the DOM tree, including inner
   * scrollable containers (e.g. when the user drags a scrollbar).
   */
  bindCaptureScrollEvent() {
    const unbindCallback = bind(document, {
      type: 'scroll',
      listener: event => {
        if (!event.isTrusted) {
          return;
        }
        this.onEvent({
          time: event.timeStamp,
          type: 'scroll-container',
          event
        });
      },
      options: {
        capture: true,
        passive: true,
        once: true
      }
    });
    this.unbindFns.push(unbindCallback);
  }
  start() {
    this.bindEvent('wheel');
    this.bindEvent('scroll');
    this.bindEvent('keydown');
    this.bindEvent('resize');
    if (fg('platform_ufo_detect_container_scroll')) {
      this.bindCaptureScrollEvent();
    }
  }
  stop() {
    this.unbindFns.forEach(cb => {
      cb();
    });
    this.unbindFns = [];
  }
}
