// @atlaskit/atlassian-navigation/dist/cjs/components/PrimaryDropdownButton/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.PrimaryDropdownButton = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _chevronDown = _interopRequireDefault(require("@atlaskit/icon/glyph/chevron-down"));
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _primitives = require("@atlaskit/primitives");
var _PrimaryButton = require("../PrimaryButton");
/** @jsx jsx */

var chevronIconStyles = (0, _primitives.xcss)({
  opacity: 0.51
});
var chevronIconStylesWithSpacingFixStyles = (0, _primitives.xcss)({
  opacity: 0.51,
  marginInlineEnd: "var(--ds-space-negative-050, -0.25rem)"
});

/**
 * __Primary dropdown button__
 *
 * A primary dropdown button allows you to add dropdown menus to the navigation.
 * Should be passed into `AtlassianNavigation`'s `primaryItems` prop.
 *
 * - [Examples](https://atlassian.design/components/atlassian-navigation/examples#button)
 * - [Code](https://atlassian.design/components/atlassian-navigation/code)
 */
var PrimaryDropdownButton = exports.PrimaryDropdownButton = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  return (0, _react2.jsx)(_PrimaryButton.PrimaryButton, (0, _extends2.default)({
    iconAfter: (0, _react2.jsx)(_primitives.Bleed, {
      xcss: (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.icon-button-spacing-fix_o1zc5') ? chevronIconStylesWithSpacingFixStyles : chevronIconStyles,
      inline: "space.100"
    }, (0, _react2.jsx)(_chevronDown.default, {
      label: ""
    })),
    ref: ref
  }, props));
});
var _default = exports.default = PrimaryDropdownButton;
