// @atlaskit/side-navigation/dist/cjs/components/Footer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _warnOnce = _interopRequireDefault(require("@atlaskit/ds-lib/warn-once"));
var _primitives = require("@atlaskit/primitives");
var _colors = require("@atlaskit/theme/colors");
var _styles = require("../../common/styles");
var _Header = require("../Header");
var _Item = require("../Item");
/* eslint-disable @repo/internal/react/consistent-props-definitions */

var iconContainerStyles = (0, _primitives.xcss)({
  display: 'inline-block',
  width: '1.5rem',
  height: '1.5rem'
});
/**
 * __Footer__
 *
 * - [Examples](https://atlassian.design/components/side-navigation/examples#header-and-footer)
 * - [Code](https://atlassian.design/components/side-navigation/code)
 */
var OldFooter = function OldFooter(props) {
  var cssFn = (0, _styles.overrideStyleFunction)(function () {
    return {
      userSelect: 'auto',
      display: 'block',
      textAlign: 'center',
      minHeight: '24px',
      alignItems: 'center',
      width: '100%',
      '[data-item-elem-before]': {
        marginRight: "var(--ds-space-0, 0px)",
        marginBottom: "var(--ds-space-100, 8px)",
        display: 'inline-block'
      },
      '[data-item-title]': {
        textAlign: 'center',
        font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)"
      },
      '[data-item-description]': {
        textAlign: 'center',
        display: 'inline-block',
        margin: "var(--ds-space-075, 6px)"
      },
      // Will look interactive if the `component` is anything other than a div.
      'div&:hover': {
        backgroundColor: "var(--ds-background-neutral-subtle, transparent)",
        cursor: 'default'
      },
      'div&:active': {
        backgroundColor: "var(--ds-background-neutral-subtle, transparent)",
        color: "var(--ds-text-subtle, ".concat(_colors.N500, ")")
      }
    };
  }, props.cssFn);

  // https://stackoverflow.com/a/39333479
  var safeProps = function (_ref) {
    var iconBefore = _ref.iconBefore,
      onClick = _ref.onClick,
      description = _ref.description,
      children = _ref.children,
      testId = _ref.testId;
    return {
      iconBefore: iconBefore,
      onClick: onClick,
      description: description,
      children: children,
      testId: testId
    };
  }(props);
  return /*#__PURE__*/_react.default.createElement(_Item.CustomItem, (0, _extends2.default)({}, safeProps, {
    component: props.component || _Header.Container
    // eslint-disable-next-line @repo/internal/react/no-unsafe-overrides
    ,
    cssFn: cssFn
  }));
};

/**
 * __Footer__
 *
 * - [Examples](https://atlassian.design/components/side-navigation/examples#header-and-footer)
 * - [Code](https://atlassian.design/components/side-navigation/code)
 */
var Footer = function Footer(_ref2) {
  var description = _ref2.description,
    children = _ref2.children,
    iconBefore = _ref2.iconBefore,
    testId = _ref2.testId;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    padding: "space.100",
    testId: testId
  }, /*#__PURE__*/_react.default.createElement(_primitives.Stack, {
    space: "space.100",
    alignInline: "center"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    xcss: iconContainerStyles
  }, iconBefore), /*#__PURE__*/_react.default.createElement(_primitives.Stack, {
    space: "space.075"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Text, {
    size: "UNSAFE_small",
    align: "center",
    color: "inherit"
  }, children), description && /*#__PURE__*/_react.default.createElement(_primitives.Text, {
    size: "UNSAFE_small",
    as: "p",
    align: "center",
    color: "color.text.subtlest"
  }, description))));
};

/**
 * __Footer__
 *
 * - [Examples](https://atlassian.design/components/side-navigation/examples#header-and-footer)
 * - [Code](https://atlassian.design/components/side-navigation/code)
 */
var FooterFacade = function FooterFacade(_ref3) {
  var _ref3$useDeprecatedAp = _ref3.useDeprecatedApi,
    useDeprecatedApi = _ref3$useDeprecatedAp === void 0 ? true : _ref3$useDeprecatedAp,
    description = _ref3.description,
    iconBefore = _ref3.iconBefore,
    testId = _ref3.testId,
    children = _ref3.children,
    component = _ref3.component,
    cssFn = _ref3.cssFn,
    onClick = _ref3.onClick;
  if (!useDeprecatedApi) {
    return /*#__PURE__*/_react.default.createElement(Footer, {
      iconBefore: iconBefore,
      description: description,
      testId: testId
    }, children);
  }
  if (typeof process !== 'undefined' && process.env.NODE_ENV !== 'production' && process.env.NODE_ENV !== 'CI') {
    (0, _warnOnce.default)('The `@atlaskit/side-navigation` Footer has simplified its API, removing support for the `component`, `cssFn` and `onClick` props.\n\nThese props will stop working in a future major release. Reach out to #help-design-system on slack for help.');
  }
  return /*#__PURE__*/_react.default.createElement(OldFooter, {
    onClick: onClick,
    component: component
    // eslint-disable-next-line @repo/internal/react/no-unsafe-overrides
    ,
    cssFn: cssFn,
    iconBefore: iconBefore,
    description: description,
    testId: testId
  }, children);
};
var _default = exports.default = FooterFacade;
