// @atlaskit/react-ufo/dist/cjs/vc/vc-observer-new/window-event-observer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _bindEventListener = require("bind-event-listener");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var WindowEventObserver = exports.default = /*#__PURE__*/function () {
  function WindowEventObserver(opts) {
    (0, _classCallCheck2.default)(this, WindowEventObserver);
    (0, _defineProperty2.default)(this, "unbindFns", []);
    this.onEvent = opts.onEvent;
  }
  return (0, _createClass2.default)(WindowEventObserver, [{
    key: "bindEvent",
    value: function bindEvent(type) {
      var _this = this;
      var unbindCallback = (0, _bindEventListener.bind)(window, {
        type: type,
        listener: function listener(event) {
          if (!event.isTrusted) {
            return;
          }
          _this.onEvent({
            time: event.timeStamp,
            type: type,
            event: event
          });
        },
        options: {
          passive: true,
          once: true
        }
      });
      this.unbindFns.push(unbindCallback);
    }

    /**
     * Binds a scroll listener on `document` using the capture phase.
     * The `scroll` event does not bubble, so listening on `window` only catches
     * document-level scrolls. By using `{ capture: true }` on `document`, we can
     * detect scroll events from any element in the DOM tree, including inner
     * scrollable containers (e.g. when the user drags a scrollbar).
     */
  }, {
    key: "bindCaptureScrollEvent",
    value: function bindCaptureScrollEvent() {
      var _this2 = this;
      var unbindCallback = (0, _bindEventListener.bind)(document, {
        type: 'scroll',
        listener: function listener(event) {
          if (!event.isTrusted) {
            return;
          }
          _this2.onEvent({
            time: event.timeStamp,
            type: 'scroll-container',
            event: event
          });
        },
        options: {
          capture: true,
          passive: true,
          once: true
        }
      });
      this.unbindFns.push(unbindCallback);
    }
  }, {
    key: "start",
    value: function start() {
      this.bindEvent('wheel');
      this.bindEvent('scroll');
      this.bindEvent('keydown');
      this.bindEvent('resize');
      if ((0, _platformFeatureFlags.fg)('platform_ufo_detect_container_scroll')) {
        this.bindCaptureScrollEvent();
      }
    }
  }, {
    key: "stop",
    value: function stop() {
      this.unbindFns.forEach(function (cb) {
        cb();
      });
      this.unbindFns = [];
    }
  }]);
}();
