// @atlaskit/page-layout/dist/cjs/components/slots/internal/left-sidebar-inner.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("@emotion/react");
var _motion = require("@atlaskit/motion");
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _responsive = require("@atlaskit/primitives/responsive");
var _constants = require("../../../common/constants");
var _hooks = require("../../../common/hooks");
/** @jsx jsx */

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
var prefersReducedMotionStyles = (0, _react.css)((0, _motion.prefersReducedMotion)());

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage -- With a feature flag, this does not apply
var mobileStyles = (0, _react.css)((0, _defineProperty2.default)({}, _responsive.UNSAFE_media.below.sm, {
  width: "".concat(_constants.MOBILE_COLLAPSED_LEFT_SIDEBAR_WIDTH, "px"),
  position: 'fixed',
  insetBlockEnd: 0,
  insetBlockStart: "calc(".concat(_constants.BANNER_HEIGHT, " + ").concat(_constants.TOP_NAVIGATION_HEIGHT, ")"),
  insetInlineStart: "".concat(_constants.LEFT_PANEL_WIDTH),
  transition: "width ".concat(_constants.TRANSITION_DURATION, "ms ").concat(_motion.easeOut, " 0s")
}));
var mobileInnerFlyoutStyles = (0, _react.css)((0, _defineProperty2.default)({}, _responsive.UNSAFE_media.below.sm, {
  width: "min(90vw, ".concat(_constants.MAX_MOBILE_SIDEBAR_FLYOUT_WIDTH, "px)"),
  maxWidth: _constants.MAX_MOBILE_SIDEBAR_FLYOUT_WIDTH,
  transition: "width ".concat(_constants.TRANSITION_DURATION, "ms ").concat(_motion.easeOut, " 0s, box-shadow ").concat(_constants.TRANSITION_DURATION, "ms ").concat(_motion.easeOut, " 0s")
}));

/**
 * This inner wrapper is required to allow the sidebar to be `position: fixed`.
 *
 * If we were to apply `position: fixed` to the outer wrapper, it will be popped
 * out of its flex container and Main would stretch to occupy all the space.
 */
var fixedInnerStyles = (0, _react.css)({
  width: "".concat(_constants.LEFT_SIDEBAR_WIDTH),
  position: 'fixed',
  insetBlockEnd: 0,
  insetBlockStart: "calc(".concat(_constants.BANNER_HEIGHT, " + ").concat(_constants.TOP_NAVIGATION_HEIGHT, ")"),
  insetInlineStart: "".concat(_constants.LEFT_PANEL_WIDTH),
  transition: "width ".concat(_constants.TRANSITION_DURATION, "ms ").concat(_motion.easeOut, " 0s")
});
var fixedInnerFlyoutStyles = (0, _react.css)({
  width: _constants.LEFT_SIDEBAR_FLYOUT_WIDTH
});

/**
 * Static in the sense of `position: static`.
 *
 * It will expand the page height to fit its content.
 */
var staticInnerStyles = (0, _react.css)({
  height: '100%'
});
var draggingStyles = (0, _react.css)({
  cursor: 'ew-resize',
  // Make sure drag to resize does not animate as the user drags
  transition: 'none'
});
var LeftSidebarInner = function LeftSidebarInner(_ref) {
  var children = _ref.children,
    _ref$isFixed = _ref.isFixed,
    isFixed = _ref$isFixed === void 0 ? false : _ref$isFixed,
    _ref$isFlyoutOpen = _ref.isFlyoutOpen,
    isFlyoutOpen = _ref$isFlyoutOpen === void 0 ? false : _ref$isFlyoutOpen;
  var isDragging = (0, _hooks.useIsSidebarDragging)();
  return (0, _react.jsx)("div", {
    css: [
    // feature flagged mobile viewport styles
    (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.responsive-page-layout-left-sidebar_p8r7g') && mobileStyles, (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.responsive-page-layout-left-sidebar_p8r7g') && isFlyoutOpen && mobileInnerFlyoutStyles,
    // generic styles
    !isFixed && staticInnerStyles, isFixed && fixedInnerStyles, isFixed && isFlyoutOpen && fixedInnerFlyoutStyles, isDragging && draggingStyles, prefersReducedMotionStyles]
  }, children);
};

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = LeftSidebarInner;
