// @atlaskit/progress-indicator/dist/es2019/components/indicator.js
import React from 'react';
import { Box, Pressable, xcss } from '@atlaskit/primitives';
import VisuallyHidden from '@atlaskit/visually-hidden';
import { varDotsMargin, varDotsSize } from './constants';
// TODO Token usages are not consistent for dots https://product-fabric.atlassian.net/browse/DSP-3180
const colorMap = {
  default: xcss({
    backgroundColor: 'color.background.neutral'
  }),
  help: xcss({
    backgroundColor: 'color.background.neutral'
  }),
  inverted: xcss({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-subtle, #626F86)"
  }),
  primary: xcss({
    backgroundColor: 'color.background.neutral'
  })
};
const selectedColorMap = {
  default: xcss({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon, #44546F)"
  }),
  help: xcss({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-discovery, #8270DB)"
  }),
  inverted: xcss({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-inverse, #FFFFFF)"
  }),
  primary: xcss({
    // @ts-expect-error
    backgroundColor: "var(--ds-icon-brand, #0C66E4)"
  })
};
const commonStyles = xcss({
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  width: `var(${varDotsSize})`,
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  height: `var(${varDotsSize})`,
  position: 'relative',
  borderRadius: 'border.radius.circle',
  '::before': {
    display: 'block',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    width: `calc(var(${varDotsSize}) + var(${varDotsMargin}))`,
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    height: `calc(var(${varDotsSize}) + var(${varDotsMargin}))`,
    position: 'absolute',
    content: '""',
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    insetBlockStart: `calc(-1 * var(${varDotsMargin}) / 2)`,
    // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
    insetInlineStart: `calc(-1 * var(${varDotsMargin}) / 2)`
  }
});
const buttonStyles = xcss({
  padding: 'space.0',
  border: 0,
  cursor: 'pointer',
  outline: 0
});
/**
 * __Presentational indicator__
 *
 * A presentational indicator with no interactivity
 */
export const PresentationalIndicator = ({
  appearance,
  isSelected,
  testId
}) => /*#__PURE__*/React.createElement(Box, {
  testId: testId,
  xcss: [commonStyles, appearance === 'default' && !isSelected && colorMap['default'], appearance === 'help' && !isSelected && colorMap['help'], appearance === 'inverted' && !isSelected && colorMap['inverted'], appearance === 'primary' && !isSelected && colorMap['primary'], appearance === 'default' && isSelected && selectedColorMap['default'], appearance === 'help' && isSelected && selectedColorMap['help'], appearance === 'inverted' && isSelected && selectedColorMap['inverted'], appearance === 'primary' && isSelected && selectedColorMap['primary']]
});
/**
 * __Button indicator__
 *
 * An interactive indicator.
 */
export const ButtonIndicator = ({
  appearance,
  panelId,
  tabId,
  isSelected,
  onClick,
  testId
}) => {
  return /*#__PURE__*/React.createElement(Pressable, {
    role: "tab",
    xcss: [commonStyles, buttonStyles, appearance === 'default' && !isSelected && colorMap['default'], appearance === 'help' && !isSelected && colorMap['help'], appearance === 'inverted' && !isSelected && colorMap['inverted'], appearance === 'primary' && !isSelected && colorMap['primary'], appearance === 'default' && isSelected && selectedColorMap['default'], appearance === 'help' && isSelected && selectedColorMap['help'], appearance === 'inverted' && isSelected && selectedColorMap['inverted'], appearance === 'primary' && isSelected && selectedColorMap['primary']],
    "aria-controls": panelId,
    "aria-selected": isSelected,
    id: tabId,
    onClick: onClick,
    tabIndex: isSelected ? -1 : undefined,
    testId: testId
  }, /*#__PURE__*/React.createElement(VisuallyHidden, null, tabId));
};
