// @atlaskit/form/dist/cjs/form-section.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _components = require("@atlaskit/theme/components");
var _typography = require("@atlaskit/theme/typography");
/** @jsx jsx */

var formSectionDescriptionStyles = (0, _react.css)({
  marginTop: "var(--ds-space-100, 8px)"
});
var formSectionTitleStyles = (0, _react.css)({
  marginTop: 0,
  marginRight: "var(--ds-space-400, 32px)",
  lineHeight: "var(--ds-space-400, 32px)",
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});
var formSectionWrapperStyles = (0, _react.css)({
  marginTop: "var(--ds-space-300, 24px)"
});

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
var lightH600Styles = (0, _react.css)((0, _typography.h600)({
  theme: {
    mode: 'light'
  }
}));
// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
var darkH600Styles = (0, _react.css)((0, _typography.h600)({
  theme: {
    mode: 'dark'
  }
}));
var FormSectionWrapper = function FormSectionWrapper(_ref) {
  var children = _ref.children;
  return (0, _react.jsx)("div", {
    css: formSectionWrapperStyles
  }, children);
};
var FormSectionTitle = function FormSectionTitle(_ref2) {
  var children = _ref2.children;
  var _useGlobalTheme = (0, _components.useGlobalTheme)(),
    mode = _useGlobalTheme.mode;
  return (0, _react.jsx)("h3", {
    css: [formSectionTitleStyles, mode === 'light' ? lightH600Styles : darkH600Styles]
  }, children);
};
var FormSectionDescription = function FormSectionDescription(_ref3) {
  var children = _ref3.children;
  return (0, _react.jsx)("div", {
    css: formSectionDescriptionStyles
  }, children);
};

/**
 * __Form section__
 *
 * A form section is used to define a section of a form layout. This contains a section title, content
 * and a description of the section.
 *
 * - [Examples](https://atlaskit.atlassian.com/packages/design-system/form/docs/layout)
 * - [Code](https://atlaskit.atlassian.com/packages/design-system/form/docs/layout)
 * - [Usage](https://atlaskit.atlassian.com/packages/design-system/form/docs/layout)
 */
var FormSection = function FormSection(_ref4) {
  var children = _ref4.children,
    description = _ref4.description,
    title = _ref4.title;
  return (0, _react.jsx)(FormSectionWrapper, null, title && (0, _react.jsx)(FormSectionTitle, null, title), description && (0, _react.jsx)(FormSectionDescription, null, description), children);
};
var _default = exports.default = FormSection;
