// @atlaskit/form/dist/cjs/label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Legend = exports.Label = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _components = require("@atlaskit/theme/components");
var _constants = require("@atlaskit/theme/constants");
var _typography = require("@atlaskit/theme/typography");
/** @jsx jsx */

var fontFamily = (0, _constants.fontFamily)();
var fieldsetLabelStyles = (0, _react.css)({
  display: 'inline-block',
  marginTop: 0,
  marginBottom: "var(--ds-space-050, 4px)",
  fontFamily: fontFamily
});
var getFieldsetLabelDynamicStyles = function getFieldsetLabelDynamicStyles(mode) {
  return (0, _react.css)([(0, _typography.h200)({
    theme: {
      mode: mode
    }
  }), {
    color: (0, _components.themed)({
      dark: "var(--ds-text-subtle, ".concat(_colors.DN300, ")"),
      light: "var(--ds-text-subtle, ".concat(_colors.N200, ")")
    })({
      theme: {
        mode: mode
      }
    })
  }]);
};

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
var lightH200Styles = getFieldsetLabelDynamicStyles('light');

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
var darkH200Styles = getFieldsetLabelDynamicStyles('dark');

/**
 * __Label__
 *
 * A label represents a caption for an item in a user interface.
 *
 * It's recommended that a label has a `4px` spacing above its associated
 * control element.
 */
var Label = exports.Label = function Label(_ref) {
  var children = _ref.children,
    htmlFor = _ref.htmlFor,
    id = _ref.id,
    testId = _ref.testId;
  var _useGlobalTheme = (0, _components.useGlobalTheme)(),
    mode = _useGlobalTheme.mode;
  return (0, _react.jsx)("label", {
    css: [mode === 'light' ? lightH200Styles : darkH200Styles, fieldsetLabelStyles],
    id: id,
    htmlFor: htmlFor,
    "data-testid": testId
  }, children);
};

/**
 * __Legend__
 *
 * A Legend represents a caption for a fieldset in a user interface.
 */
var Legend = exports.Legend = function Legend(_ref2) {
  var children = _ref2.children;
  var _useGlobalTheme2 = (0, _components.useGlobalTheme)(),
    mode = _useGlobalTheme2.mode;
  return (0, _react.jsx)("legend", {
    css: [mode === 'light' ? lightH200Styles : darkH200Styles, fieldsetLabelStyles]
  }, children);
};
var _default = exports.default = Label;
