// @atlaskit/onboarding/dist/cjs/styled/dialog.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DialogImage = exports.DialogActionItemContainer = exports.DialogActionItem = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("@emotion/react");
var _excluded = ["alt"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var imageStyles = (0, _react.css)({
  maxWidth: '100%',
  height: 'auto'
});
var actionItemContainerStyles = (0, _react.css)({
  display: 'flex',
  margin: "var(--ds-space-0, 0px)".concat(" -4px"),
  /* When there is more than one action, place primary action visually on the
   right, but keep it's position as the first focusable element in the DOM */
  flexDirection: 'row-reverse'
});
var actionItemStyles = (0, _react.css)({
  margin: "var(--ds-space-0, 0px)".concat(" ", "var(--ds-space-050, 4px)")
});
/**
 * __Dialog image__
 *
 * An optional header image in spotlight dialogs.
 *
 * @internal
 */
var DialogImage = exports.DialogImage = function DialogImage(_ref) {
  var alt = _ref.alt,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
    (0, _react.jsx)("img", (0, _extends2.default)({
      css: imageStyles,
      alt: alt
    }, props))
  );
};

/**
 * __Dialog action item container__
 *
 * Flex wrapper around dialog action items.
 *
 * @internal
 */
var DialogActionItemContainer = exports.DialogActionItemContainer = function DialogActionItemContainer(_ref2) {
  var children = _ref2.children;
  return (0, _react.jsx)("div", {
    css: actionItemContainerStyles
  }, children);
};

/**
 * __Dialog action item__
 *
 * Action items shown inside of the dialog.
 *
 * @internal
 */
var DialogActionItem = exports.DialogActionItem = function DialogActionItem(_ref3) {
  var children = _ref3.children;
  return (0, _react.jsx)("div", {
    css: actionItemStyles
  }, children);
};
