// @atlaskit/onboarding/dist/cjs/styled/modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalImage = exports.ModalHeading = exports.ModalBody = exports.ModalActionItem = exports.ModalActionContainer = void 0;
var _react = require("@emotion/react");
var _typography = require("@atlaskit/theme/typography");
/* eslint-disable @atlaskit/design-system/no-nested-styles */
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var modalBodyStyles = (0, _react.css)({
  padding: "var(--ds-space-500, 40px)".concat(" ", "var(--ds-space-200, 16px)"),
  textAlign: 'center'
});

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-array-arguments -- Ignored via go/DSP-18766
var modalHeadingStyles = (0, _react.css)([
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
(0, _react.css)((0, _typography.h600)({
  theme: {
    mode: 'light'
  }
})), {
  marginBottom: "var(--ds-space-100, 8px)",
  color: 'inherit'
}]);
var modalImageStyles = (0, _react.css)({
  width: '100%',
  height: 'auto',
  borderStartEndRadius: "var(--ds-border-radius, 3px)",
  borderStartStartRadius: "var(--ds-border-radius, 3px)",
  '@media (min-width: 320px) and (max-width: 480px)': {
    borderRadius: 0
  }
});
var modalActionContainerStyles = (0, _react.css)({
  display: 'flex',
  padding: "var(--ds-space-0, 0px)".concat(" ", "var(--ds-space-500, 40px)", " 36px"),
  justifyContent: 'center',
  flexDirection: 'row',
  flexFlow: 'wrap'
});
var modalActionContainerReversedStyles = (0, _react.css)({
  flexDirection: 'row-reverse'
});
var modalActionItemStyles = (0, _react.css)({
  margin: "var(--ds-space-0, 0px)".concat(" ", "var(--ds-space-050, 4px)", " ", "var(--ds-space-050, 4px)")
});

/**
 * __Modal body__
 *
 * @internal
 */
var ModalBody = exports.ModalBody = function ModalBody(_ref) {
  var children = _ref.children;
  return (0, _react.jsx)("div", {
    css: modalBodyStyles
  }, children);
};

/**
 * __Modal heading__
 *
 * @internal
 */
var ModalHeading = exports.ModalHeading = function ModalHeading(_ref2) {
  var children = _ref2.children,
    id = _ref2.id;
  return (0, _react.jsx)("h1", {
    css: modalHeadingStyles,
    id: id
  }, children);
};

/**
 * __Modal image__
 *
 * @internal
 */
var ModalImage = exports.ModalImage = function ModalImage(_ref3) {
  var alt = _ref3.alt,
    src = _ref3.src;
  return (0, _react.jsx)("img", {
    css: modalImageStyles,
    alt: alt,
    src: src
  });
};

/**
 * __Modal action container__
 *
 * @internal
 */
var ModalActionContainer = exports.ModalActionContainer = function ModalActionContainer(_ref4) {
  var children = _ref4.children,
    shouldReverseButtonOrder = _ref4.shouldReverseButtonOrder;
  return (0, _react.jsx)("div", {
    css: [modalActionContainerStyles, shouldReverseButtonOrder && modalActionContainerReversedStyles]
  }, children);
};

/**
 * __Modal action item__
 *
 * @internal
 */
var ModalActionItem = exports.ModalActionItem = function ModalActionItem(_ref5) {
  var children = _ref5.children;
  return (0, _react.jsx)("div", {
    css: modalActionItemStyles
  }, children);
};
