// @atlaskit/table-tree/dist/esm/components/row.js
import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
/* eslint-disable @repo/internal/react/no-clone-element */
import React, { Component, Fragment } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { createAndFireEvent, withAnalyticsContext, withAnalyticsEvents } from '@atlaskit/analytics-next';
import toItemId from '../utils/to-item-id';
import Chevron from './internal/chevron';
import { TreeRowContainer } from './internal/styled';
var treeRowClickableStyles = css({
  cursor: 'pointer'
});
var packageName = "@atlaskit/table-tree";
var packageVersion = "9.12.2";

/**
 * This is hard-coded here because our actual <TableTree /> has no typings
 * for its props.
 *
 * Adding types for real *might* break things so will need a little care.
 *
 * Defining it here for now lets us provide *something* without much headache.
 */
var RowComponent = /*#__PURE__*/function (_Component) {
  _inherits(RowComponent, _Component);
  var _super = _createSuper(RowComponent);
  function RowComponent() {
    var _this;
    _classCallCheck(this, RowComponent);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    _defineProperty(_assertThisInitialized(_this), "state", {
      isExpanded: _this.props.isDefaultExpanded || false
    });
    /**
     * This ensures a user won't trigger a click event and expand the accordion
     * when making a text selection.
     */
    _defineProperty(_assertThisInitialized(_this), "onClickHandler", function (e) {
      var _window$getSelection;
      var selection = ((_window$getSelection = window.getSelection()) === null || _window$getSelection === void 0 ? void 0 : _window$getSelection.toString()) || '';
      if ((selection === null || selection === void 0 ? void 0 : selection.length) === 0) {
        _this.onExpandToggle();
      }
    });
    _defineProperty(_assertThisInitialized(_this), "onExpandToggle", function () {
      var isExpanded = _this.props.isExpanded;
      if (isExpanded !== undefined) {
        _this.onExpandStateChange(!isExpanded);
      } else {
        _this.setState({
          isExpanded: !_this.state.isExpanded
        });
        _this.onExpandStateChange(!_this.state.isExpanded);
      }
    });
    _defineProperty(_assertThisInitialized(_this), "getExtendedLabel", function (cellContent, cellIndex, mainColumnForExpandCollapseLabel) {
      /**
       * First condition - when we pass data via `items` property in `<TableTree />`
       * Second condition - when we pass data via `<Rows />` as children in `<TableTree />`.
       */
      if (cellContent.hasOwnProperty('props')) {
        return cellContent === null || cellContent === void 0 ? void 0 : cellContent.props[mainColumnForExpandCollapseLabel === null || mainColumnForExpandCollapseLabel === void 0 ? void 0 : mainColumnForExpandCollapseLabel.toLowerCase()];
      } else if (cellIndex === mainColumnForExpandCollapseLabel) {
        return cellContent;
      }
      return undefined;
    });
    return _this;
  }
  _createClass(RowComponent, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props = this.props,
        isDefaultExpanded = _this$props.isDefaultExpanded,
        isExpanded = _this$props.isExpanded;
      if (isExpanded === undefined && isDefaultExpanded !== undefined && prevProps.isDefaultExpanded !== isDefaultExpanded && this.state.isExpanded !== isDefaultExpanded) {
        // eslint-disable-next-line react/no-did-update-set-state
        this.setState({
          isExpanded: isDefaultExpanded
        });
      }
    }
  }, {
    key: "onExpandStateChange",
    value: function onExpandStateChange(isExpanded) {
      if (this.props.data) {
        if (isExpanded && this.props.onExpand) {
          this.props.onExpand(this.props.data);
        } else if (!isExpanded && this.props.onCollapse) {
          this.props.onCollapse(this.props.data);
        }
      }
    }
  }, {
    key: "isExpanded",
    value: function isExpanded() {
      var isExpanded = this.props.isExpanded;
      return isExpanded !== undefined ? isExpanded : this.state.isExpanded;
    }
  }, {
    key: "renderCell",
    value: function renderCell(cell, cellIndex) {
      var props = this.props;
      var isExpanded = this.isExpanded();
      var hasChildren = props.hasChildren,
        depth = props.depth,
        mainColumnForExpandCollapseLabel = props.mainColumnForExpandCollapseLabel;
      var isFirstCell = cellIndex === 0;
      var indentLevel = isFirstCell ? depth : 0;
      var cellContent = cell.props.children || [];
      var extendedLabel = this.getExtendedLabel(cellContent, cellIndex, mainColumnForExpandCollapseLabel);
      if (isFirstCell && hasChildren) {
        cellContent = [jsx(Chevron, {
          key: "chevron",
          expandLabel: props.expandLabel,
          collapseLabel: props.collapseLabel,
          extendedLabel: extendedLabel,
          isExpanded: isExpanded,
          onExpandToggle: this.onExpandToggle,
          ariaControls: isExpanded ? toItemId(props.itemId) : undefined,
          rowId: props.itemId
        })].concat(cellContent);
      }
      return /*#__PURE__*/React.cloneElement(cell, {
        key: cellIndex,
        columnIndex: cellIndex,
        indentLevel: indentLevel
      }, cellContent);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;
      var _this$props2 = this.props,
        shouldExpandOnClick = _this$props2.shouldExpandOnClick,
        hasChildren = _this$props2.hasChildren,
        depth = _this$props2.depth,
        renderChildren = _this$props2.renderChildren;
      var isExpanded = this.isExpanded();
      var ariaAttrs = {};
      if (hasChildren) {
        ariaAttrs['aria-expanded'] = isExpanded;
      }
      if (depth !== undefined) {
        ariaAttrs['aria-level'] = depth;
      }
      return jsx(Fragment, null, jsx(TreeRowContainer, _extends({
        css: hasChildren && shouldExpandOnClick ? treeRowClickableStyles : undefined,
        onClick: hasChildren && shouldExpandOnClick ? this.onClickHandler : undefined
      }, ariaAttrs), React.Children.map(this.props.children, function (cell, index) {
        return _this2.renderCell(cell, index);
      })), hasChildren && isExpanded && renderChildren && renderChildren());
    }
  }]);
  return RowComponent;
}(Component);
export { RowComponent as RowWithoutAnalytics };
var createAndFireEventOnAtlaskit = createAndFireEvent('atlaskit');
var Row = withAnalyticsContext({
  componentName: 'row',
  packageName: packageName,
  packageVersion: packageVersion
})(withAnalyticsEvents({
  onExpand: createAndFireEventOnAtlaskit({
    action: 'expanded',
    actionSubject: 'tableTree',
    attributes: {
      componentName: 'row',
      packageName: packageName,
      packageVersion: packageVersion
    }
  }),
  onCollapse: createAndFireEventOnAtlaskit({
    action: 'collapsed',
    actionSubject: 'tableTree',
    attributes: {
      componentName: 'row',
      packageName: packageName,
      packageVersion: packageVersion
    }
  })
})(RowComponent));

// eslint-disable-next-line @repo/internal/react/require-jsdoc
export default Row;
