// @atlaskit/table-tree/dist/es2019/components/cell.js
import _extends from "@babel/runtime/helpers/extends";
import React, { Component } from 'react';
import CommonCell from './internal/common-cell';
import OverflowContainer from './internal/overflow-container';
import { indentBase } from './internal/styled';
import withColumnWidth from './internal/with-column-width';
class CellComponent extends Component {
  render() {
    const {
      children,
      singleLine,
      indentLevel,
      width,
      className,
      ...props
    } = this.props;
    return /*#__PURE__*/React.createElement(CommonCell, _extends({
      indent: indentLevel ? `calc(${indentBase} * ${indentLevel})` : undefined,
      width: width
      // eslint-disable-next-line @atlaskit/ui-styling-standard/no-classname-prop -- Ignored via go/DSP-18766
      ,
      className: className
    }, props), /*#__PURE__*/React.createElement(OverflowContainer, {
      isSingleLine: singleLine
    }, children));
  }
}
const Cell = withColumnWidth(CellComponent);

// eslint-disable-next-line @repo/internal/react/require-jsdoc
export default Cell;
