// @atlaskit/table-tree/dist/es2019/components/header.js
import _extends from "@babel/runtime/helpers/extends";
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
import { Component } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { N300 } from '@atlaskit/theme/colors';
import ColumnCell from './internal/common-cell';
import withColumnWidth from './internal/with-column-width';
const headerStyles = css({
  color: `var(--ds-text-subtle, ${N300})`,
  font: "var(--ds-font-body-UNSAFE_small, normal 400 12px/16px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  fontWeight: "var(--ds-font-weight-bold, 700)"
});

/**
 * This is hard-coded here because our actual <Header /> has no typings
 * for its props.
 *
 * Adding types for real *might* break things so will need a little care.
 *
 * Defining it here for now lets us provide *something* without much headache.
 */

class HeaderComponent extends Component {
  render() {
    const {
      props
    } = this;
    return (
      // TODO: Determine whether proper `th` elements can be used instead of
      // roles (DSP-11588)
      jsx(ColumnCell, _extends({
        css: headerStyles,
        role: "columnheader",
        style: {
          width: props.width
        }
      }, props), props.children)
    );
  }
}
const Header = withColumnWidth(HeaderComponent);

// eslint-disable-next-line @repo/internal/react/require-jsdoc
export default Header;
