// @atlaskit/page-header/dist/es2019/PageHeader/styled.js
// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. The prop name concerned here is truncateTitle.
// This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

import React from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { Box, xcss } from '@atlaskit/primitives';
const truncateStyles = css({
  overflowX: 'hidden',
  overflowY: 'hidden',
  // `overflow-y` necessary to prevent a scrollbar from showing in FF: https://product-fabric.atlassian.net/browse/PYX-1035
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});
const outerStyles = css({
  margin: `${"var(--ds-space-300, 24px)"} 0 ${"var(--ds-space-200, 16px)"} 0`
});
const titleStyles = css({
  color: "var(--ds-text, #172B4D)",
  font: "var(--ds-font-heading-large, normal 500 24px/28px ui-sans-serif, -apple-system, BlinkMacSystemFont, \"Segoe UI\", Ubuntu, system-ui, \"Helvetica Neue\", sans-serif)",
  letterSpacing: 'normal',
  marginBlockStart: 0,
  outline: 'none'
});
const titleWrapperStyles = css({
  display: 'flex',
  alignItems: 'flex-start',
  flexWrap: 'wrap'
});
const titleWrapperTruncateStyles = css({
  flexWrap: 'nowrap'
});
const titleContainerStyles = css({
  minWidth: 0,
  maxWidth: '100%',
  flex: '1 0 auto',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  flexShrink: undefined,
  marginBlockEnd: "var(--ds-space-100, 8px)"
});
const actionStyles = css({
  maxWidth: '100%',
  flex: '0 0 auto',
  marginBlockEnd: "var(--ds-space-100, 8px)",
  marginInlineStart: 'auto',
  paddingInlineStart: "var(--ds-space-400, 32px)",
  whiteSpace: 'nowrap',
  // eslint-disable-next-line @atlaskit/design-system/no-nested-styles, @atlaskit/ui-styling-standard/no-nested-selectors -- Ignored via go/DSP-18766
  '>': {
    textAlign: 'right'
  }
});
const titleContainerTruncateStyles = css({
  flexShrink: 1
});
const bottomBarStyles = xcss({
  marginBlockStart: 'space.200'
});

/**
 * __Outer wrapper__.
 *
 * An outer wrapper that is the outermost component of the PageHeader component. It wraps around the PageHeader, its Actions,
 * the BottomBar and its Breadcrumbs.
 *
 */
export const OuterWrapper = ({
  children
}) => {
  return jsx("div", {
    css: outerStyles
  }, children);
};
/**
 * __Styled title wrapper__.
 *
 * A styled title wrapper is a wrapper around the title that controls its the styles exclusively.
 *
 */
export const StyledTitleWrapper = /*#__PURE__*/React.forwardRef(({
  children,
  id,
  truncateTitle
}, ref) => {
  return jsx("h1", {
    css: [titleStyles, truncateTitle && truncateStyles],
    ref: ref,
    tabIndex: -1,
    id: id
  }, children);
});
/**
 * __Title wrapper__.
 *
 * A title wrapper is a wrapper around the title and the actions.
 *
 */
export const TitleWrapper = ({
  children,
  truncateTitle
}) => {
  return jsx("div", {
    css: [titleWrapperStyles, truncateTitle && titleWrapperTruncateStyles]
  }, children);
};

/**
 * Title container.
 *
 * A title container is a container that wraps around the title and its styles (if applied).
 *
 */
export const TitleContainer = ({
  children,
  truncateTitle
}) => {
  return jsx("div", {
    css: [titleContainerStyles, truncateTitle && titleContainerTruncateStyles]
  }, children);
};

/**
 * __Actions wrapper__.
 *
 * An actions wrapper is a wrapper for the actions, which appear on the top right of the PageHeader component.
 *
 */
export const ActionsWrapper = ({
  children
}) => {
  return jsx("div", {
    css: actionStyles
  }, children);
};

/**
 * __Bottom bar wrapper__.
 *
 * A bottom bar wrapper is a wrapper for the bottom bar, which appears at the bottom of the PageHeader component.
 *
 */
export const BottomBarWrapper = ({
  children
}) => {
  return jsx(Box, {
    xcss: bottomBarStyles
  }, children);
};
