// @atlaskit/atlassian-navigation/dist/cjs/components/PrimaryItemsContainerV2/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrimaryItemsContainer = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireWildcard(require("react"));
var _popup = _interopRequireDefault(require("@atlaskit/popup"));
var _primitives = require("@atlaskit/primitives");
var _responsive = require("@atlaskit/primitives/responsive");
var _overflow = require("../../controllers/overflow");
var _PrimaryDropdownButton = require("../PrimaryDropdownButton");
var _xcss2;
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var sharedContainerStyles = (0, _primitives.xcss)({
  height: '100%',
  alignItems: 'stretch',
  paddingInlineEnd: 'space.050',
  gap: 'space.100'
});
var smallContainerStyles = (0, _primitives.xcss)((0, _defineProperty2.default)({}, _responsive.media.above.sm, {
  display: 'none'
}));
var mediumContainerStyles = (0, _primitives.xcss)((_xcss2 = {
  display: 'none'
}, (0, _defineProperty2.default)(_xcss2, _responsive.media.above.sm, {
  display: 'flex'
}), (0, _defineProperty2.default)(_xcss2, _responsive.media.above.lg, {
  display: 'none'
}), _xcss2));
var largeContainerStyles = (0, _primitives.xcss)((0, _defineProperty2.default)({
  display: 'none'
}, _responsive.media.above.lg, {
  display: 'flex'
}));
var MoreItemsPopup = function MoreItemsPopup(_ref) {
  var moreLabel = _ref.moreLabel,
    testId = _ref.testId,
    items = _ref.items;
  var _useState = (0, _react.useState)(false),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    isMoreOpen = _useState2[0],
    setIsMoreOpen = _useState2[1];
  var onMoreClose = (0, _react.useCallback)(function () {
    return setIsMoreOpen(false);
  }, []);
  var onMoreClick = (0, _react.useCallback)(function () {
    return setIsMoreOpen(function (current) {
      return !current;
    });
  }, []);
  var openOverflowMenu = (0, _react.useCallback)(function () {
    return setIsMoreOpen(true);
  }, []);
  var trigger = (0, _react.useCallback)(function (triggerProps) {
    return /*#__PURE__*/_react.default.createElement(_PrimaryDropdownButton.PrimaryDropdownButton, (0, _extends2.default)({
      onClick: onMoreClick,
      isSelected: isMoreOpen,
      testId: testId ? "".concat(testId, "-overflow-menu-trigger") : 'overflow-menu-trigger'
    }, triggerProps), moreLabel);
  }, [moreLabel, onMoreClick, isMoreOpen, testId]);
  var content = (0, _react.useCallback)(function () {
    return /*#__PURE__*/_react.default.createElement(_overflow.OverflowProvider, {
      isVisible: false,
      openOverflowMenu: openOverflowMenu,
      closeOverflowMenu: onMoreClose
    }, items);
  }, [items, openOverflowMenu, onMoreClose]);
  return /*#__PURE__*/_react.default.createElement(_popup.default, {
    placement: "bottom-start",
    isOpen: isMoreOpen,
    onClose: onMoreClose,
    trigger: trigger,
    content: content,
    testId: testId ? "".concat(testId, "-overflow-menu-popup") : 'overflow-menu-popup'
  });
};

// Internal only
// eslint-disable-next-line @repo/internal/react/require-jsdoc
var PrimaryItemsContainer = exports.PrimaryItemsContainer = /*#__PURE__*/(0, _react.memo)(function (_ref2) {
  var moreLabel = _ref2.moreLabel,
    items = _ref2.items,
    Create = _ref2.create,
    theme = _ref2.theme,
    testId = _ref2.testId;
  // We render a CSS media query based nav at first to handle SSR, then use
  // our useMediaQuery hook once we are hydrated so there is only one set of nav items
  var _useState3 = (0, _react.useState)(false),
    _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
    isClient = _useState4[0],
    setIsClient = _useState4[1];
  (0, _react.useEffect)(function () {
    setIsClient(true);
  }, []);

  // Setting up our media queries to use once app is hydrated
  var mqSm = (0, _responsive.UNSAFE_useMediaQuery)('above.sm', function (event) {
    return setIsAboveSm(event.matches);
  });
  var _useState5 = (0, _react.useState)(mqSm === null || mqSm === void 0 ? void 0 : mqSm.matches),
    _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
    isAboveSm = _useState6[0],
    setIsAboveSm = _useState6[1];
  var mqLg = (0, _responsive.UNSAFE_useMediaQuery)('above.lg', function (event) {
    return setIsAboveLg(event.matches);
  });
  var _useState7 = (0, _react.useState)(mqLg === null || mqLg === void 0 ? void 0 : mqLg.matches),
    _useState8 = (0, _slicedToArray2.default)(_useState7, 2),
    isAboveLg = _useState8[0],
    setIsAboveLg = _useState8[1];

  // Filter out any falsy items passed in
  var filteredItems = (0, _react.useMemo)(function () {
    return _react.default.Children.toArray(items).filter(function (item) {
      return !!item;
    });
  }, [items]);
  // NOTE: we could make these max items configurable in the future. For now
  // we are using sensible defaults. Anything over the max gets put in the
  // overflow menu for that screen size.
  // While it may be tempting to use more than 3 items for the medium nav when
  // you see the available space, you need to consider internationalistion.
  // 3 is the safe, defensive choice across many languages and scripts.
  var smallMaxItems = 0;
  var mediumMaxItems = 3;
  var largeMaxItems = 8;

  // We re-use this in both the CSS media query nav that loads in for SSR,
  // and the JS/hook media query nav that is used once hydrated
  var navItems = (0, _react.useMemo)(function () {
    return {
      small: {
        navBarItems: [],
        overflowItems: filteredItems
      },
      medium: {
        navBarItems: filteredItems.slice(0, mediumMaxItems),
        overflowItems: filteredItems.slice(mediumMaxItems, filteredItems.length)
      },
      large: {
        navBarItems: filteredItems.slice(0, largeMaxItems),
        overflowItems: filteredItems.slice(largeMaxItems, filteredItems.length)
      }
    };
  }, [filteredItems]);
  var hydratedNavItems = (0, _react.useMemo)(function () {
    if (isAboveLg) {
      return navItems.large;
    }
    if (isAboveSm && !isAboveLg) {
      return navItems.medium;
    }
    return navItems.small;
  }, [isAboveLg, isAboveSm, navItems.large, navItems.medium, navItems.small]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, isClient ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    testId: testId && "".concat(testId, "-primary-actions"),
    xcss: sharedContainerStyles
  }, hydratedNavItems.navBarItems, hydratedNavItems.overflowItems.length > 0 && /*#__PURE__*/_react.default.createElement(MoreItemsPopup, {
    moreLabel: moreLabel,
    items: hydratedNavItems.overflowItems,
    testId: testId
  }))) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    xcss: [sharedContainerStyles, smallContainerStyles]
  }, filteredItems.length > smallMaxItems &&
  /*#__PURE__*/
  // We don't need to pass items into popup, it won't be interactive (SSR only)
  _react.default.createElement(MoreItemsPopup, {
    moreLabel: moreLabel,
    testId: testId
  })), /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    xcss: [sharedContainerStyles, mediumContainerStyles]
  }, navItems.medium.navBarItems, navItems.medium.overflowItems.length > 0 && /*#__PURE__*/_react.default.createElement(MoreItemsPopup, {
    moreLabel: moreLabel,
    testId: testId
  })), /*#__PURE__*/_react.default.createElement(_primitives.Inline, {
    xcss: [sharedContainerStyles, largeContainerStyles]
  }, navItems.large.navBarItems, navItems.large.overflowItems.length > 0 && /*#__PURE__*/_react.default.createElement(MoreItemsPopup, {
    moreLabel: moreLabel,
    testId: testId
  }))), Create && /*#__PURE__*/_react.default.createElement(Create, null));
});
