// @atlaskit/onboarding/dist/cjs/components/modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _customThemeButton = _interopRequireWildcard(require("@atlaskit/button/custom-theme-button"));
var _modalDialog = _interopRequireWildcard(require("@atlaskit/modal-dialog"));
var _modal = require("../styled/modal");
var _theme = require("./theme");
var _excluded = ["text", "key"],
  _excluded2 = ["actions", "children", "heading"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _createSuper(t) { var r = _isNativeReflectConstruct(); return function () { var e, o = (0, _getPrototypeOf2.default)(t); if (r) { var s = (0, _getPrototypeOf2.default)(this).constructor; e = Reflect.construct(o, arguments, s); } else e = o.apply(this, arguments); return (0, _possibleConstructorReturn2.default)(this, e); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); } /**
 * @jsxRuntime classic
 * @jsx jsx
 */ // eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
// TODO: DSP-1250 - use a composable API consistent with normal modal dialog
/**
 * __Benefits modal__
 *
 * A benefits modal explains the benefits of a significant new feature or experience change.
 *
 * - [Examples](https://atlassian.design/components/onboarding/benefits-modal/examples)
 * - [Code](https://atlassian.design/components/onboarding/benefits-modal/code)
 * - [Usage](https://atlassian.design/components/onboarding/benefits-modal/usage)
 */
var BenefitsModal = exports.default = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(BenefitsModal, _Component);
  var _super = _createSuper(BenefitsModal);
  function BenefitsModal() {
    var _this;
    (0, _classCallCheck2.default)(this, BenefitsModal);
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "headerComponent", function (props) {
      var HeaderElement = props.header,
        src = props.image;
      var ImageElement = function ImageElement() {
        return (0, _react2.jsx)(_modal.ModalImage, {
          src: src,
          alt: ""
        });
      };
      return HeaderElement || ImageElement;
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "footerComponent", function (props) {
      var FooterElement = props.footer,
        actionList = props.actions,
        _props$experimental_s = props.experimental_shouldShowPrimaryButtonOnRight,
        experimental_shouldShowPrimaryButtonOnRight = _props$experimental_s === void 0 ? false : _props$experimental_s;
      var ActionsElement = function ActionsElement() {
        return actionList ? (0, _react2.jsx)(_customThemeButton.Theme.Provider, {
          value: _theme.modalButtonTheme
        }, (0, _react2.jsx)(_modal.ModalActionContainer, {
          shouldReverseButtonOrder: experimental_shouldShowPrimaryButtonOnRight
        }, actionList.map(function (_ref, idx) {
          var text = _ref.text,
            key = _ref.key,
            rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
          var variant = idx ? 'subtle-link' : 'primary';
          return (0, _react2.jsx)(_modal.ModalActionItem, {
            key: key || (typeof text === 'string' ? text : "".concat(idx))
          }, (0, _react2.jsx)(_customThemeButton.default, (0, _extends2.default)({
            appearance: variant,
            autoFocus: !idx
          }, rest), text));
        }))) : null;
      };
      return FooterElement || ActionsElement;
    });
    return _this;
  }
  (0, _createClass2.default)(BenefitsModal, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        actions = _this$props.actions,
        children = _this$props.children,
        heading = _this$props.heading,
        props = (0, _objectWithoutProperties2.default)(_this$props, _excluded2);
      var Header = this.headerComponent(this.props);
      var Footer = this.footerComponent(this.props);
      var CustomHeader = function CustomHeader() {
        var _useModal = (0, _modalDialog.useModal)(),
          titleId = _useModal.titleId;
        return (0, _react2.jsx)(_modal.ModalHeading, {
          id: titleId
        }, heading);
      };
      return (0, _react2.jsx)(_modalDialog.default, (0, _extends2.default)({
        autoFocus: true,
        shouldScrollInViewport: true,
        shouldCloseOnOverlayClick: false,
        shouldCloseOnEscapePress: false
      }, props), (0, _react2.jsx)(Header, null), (0, _react2.jsx)(_modalDialog.ModalBody, null, (0, _react2.jsx)(_modal.ModalBody, null, heading && (0, _react2.jsx)(CustomHeader, null), children)), (0, _react2.jsx)(Footer, null));
    }
  }]);
  return BenefitsModal;
}(_react.Component);
