// @atlaskit/onboarding/dist/cjs/components/spotlight-card.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _customThemeButton = _interopRequireWildcard(require("@atlaskit/button/custom-theme-button"));
var _heading = _interopRequireDefault(require("@atlaskit/heading"));
var _layering = require("@atlaskit/layering");
var _primitives = require("@atlaskit/primitives");
var _dialog = require("../styled/dialog");
var _theme = require("./theme");
var _excluded = ["text", "key"];
/**
 * @jsxRuntime classic
 * @jsx jsx
 */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var bodyStyles = (0, _react2.css)({
  display: 'flex',
  padding: "var(--ds-space-200, 16px)".concat(" ", "var(--ds-space-250, 20px)"),
  flexDirection: 'column'
});
var imageStyles = (0, _react2.css)({
  display: 'block'
});
var defaultHeaderStyles = (0, _react2.css)({
  display: 'flex',
  alignItems: 'baseline',
  justifyContent: 'space-between',
  paddingBlockEnd: "var(--ds-space-100, 8px)"
});
var DefaultHeader = function DefaultHeader(_ref) {
  var children = _ref.children;
  return (0, _react2.jsx)("div", {
    css: defaultHeaderStyles
  }, children);
};
var defaultFooterStyles = (0, _react2.css)({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  paddingBlockStart: "var(--ds-space-100, 8px)"
});
var DefaultFooter = function DefaultFooter(_ref2) {
  var children = _ref2.children;
  return (0, _react2.jsx)("div", {
    css: defaultFooterStyles
  }, children);
};
var containerStyles = (0, _primitives.xcss)({
  height: 'fit-content',
  zIndex: 'spotlight',
  borderRadius: 'border.radius',
  color: 'color.text.inverse',
  overflow: 'auto',
  minWidth: '160px',
  maxWidth: '600px'
});
var containerShadowStyles = (0, _primitives.xcss)({
  boxShadow: 'elevation.shadow.raised'
});
/**
 * __Spotlight card__
 *
 * A spotlight card is for onboarding messages that need a more flexible layout, or don't require a dialog.
 *
 * - [Examples](https://atlassian.design/components/onboarding/spotlight-card/examples)
 * - [Code](https://atlassian.design/components/onboarding/spotlight-card/code)
 * - [Usage](https://atlassian.design/components/onboarding/spotlight-card/usage)
 */
var SpotlightCard = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var _props$actions = props.actions,
    actions = _props$actions === void 0 ? [] : _props$actions,
    actionsBeforeElement = props.actionsBeforeElement,
    children = props.children,
    _props$components = props.components,
    components = _props$components === void 0 ? {} : _props$components,
    heading = props.heading,
    _props$headingLevel = props.headingLevel,
    headingLevel = _props$headingLevel === void 0 ? 4 : _props$headingLevel,
    headingAfterElement = props.headingAfterElement,
    image = props.image,
    innerRef = props.innerRef,
    isFlat = props.isFlat,
    testId = props.testId,
    _props$width = props.width,
    width = _props$width === void 0 ? 400 : _props$width,
    headingId = props.headingId;
  var _components$Header = components.Header,
    Header = _components$Header === void 0 ? DefaultHeader : _components$Header,
    _components$Footer = components.Footer,
    Footer = _components$Footer === void 0 ? DefaultFooter : _components$Footer;
  var _UNSAFE_useLayering = (0, _layering.UNSAFE_useLayering)(),
    currentLevel = _UNSAFE_useLayering.currentLevel;
  return (0, _react2.jsx)(_customThemeButton.Theme.Provider, {
    value: _theme.spotlightButtonTheme
  }, (0, _react2.jsx)(_primitives.Box, {
    backgroundColor: "color.background.discovery.bold",
    xcss: [containerStyles, !isFlat && containerShadowStyles],
    style: {
      width: width
    },
    ref: ref || innerRef,
    testId: testId,
    "data-ds--level": currentLevel
    // temporarily use this data attribute to prevent clicking outside won't close spotlight correctly issue
    ,
    "data-ds--close--type": "single"
  }, typeof image === 'string' ? (0, _react2.jsx)("img", {
    css: imageStyles,
    src: image,
    alt: ""
  }) : image, (0, _react2.jsx)("div", {
    css: bodyStyles
  }, heading || headingAfterElement ? (0, _react2.jsx)(Header, null, (0, _react2.jsx)(_heading.default, {
    id: headingId,
    size: "medium",
    as: "h".concat(headingLevel),
    color: "color.text.inverse"
  }, heading), headingAfterElement) : null, (0, _react2.jsx)(_primitives.Text, null, children), actions.length > 0 || actionsBeforeElement ? (0, _react2.jsx)(Footer, null, actionsBeforeElement || (0, _react2.jsx)("span", null), (0, _react2.jsx)(_dialog.DialogActionItemContainer, null, actions.map(function (_ref3, idx) {
    var text = _ref3.text,
      key = _ref3.key,
      rest = (0, _objectWithoutProperties2.default)(_ref3, _excluded);
    return (0, _react2.jsx)(_dialog.DialogActionItem, {
      key: key || (typeof text === 'string' ? text : "".concat(idx))
    }, (0, _react2.jsx)(_customThemeButton.default, rest, text));
  }))) : null)));
});
SpotlightCard.displayName = 'SpotlightCard';
var _default = exports.default = SpotlightCard;
