// @atlaskit/radio/dist/cjs/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = getRadioCustomProperties;
var _platformFeatureFlags = require("@atlaskit/platform-feature-flags");
var _colors = require("@atlaskit/theme/colors");
var radioThemeColors = {
  light: {
    background: "var(--ds-background-input, ".concat(_colors.N10, ")"),
    backgroundHover: "var(--ds-background-input-hovered, ".concat(_colors.N30, ")"),
    backgroundActive: "var(--ds-background-input-pressed, ".concat(_colors.N30, ")"),
    backgroundChecked: "var(--ds-background-selected-bold, ".concat(_colors.B400, ")"),
    backgroundCheckedHover: "var(--ds-background-selected-bold-hovered, ".concat(_colors.B300, ")"),
    backgroundCheckedActive: "var(--ds-background-selected-bold-pressed, ".concat(_colors.B50, ")"),
    backgroundDisabled: "var(--ds-background-disabled, ".concat(_colors.N20, ")"),
    dotChecked: "var(--ds-icon-inverse, ".concat(_colors.N10, ")"),
    dotDisabled: "var(--ds-icon-disabled, ".concat(_colors.N70, ")"),
    dotActive: "var(--ds-icon-inverse, ".concat(_colors.B400, ")"),
    border: (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.update-border-radio-checkbox_7askv') ? "var(--ds-border-bold, ".concat(_colors.N100, ")") : "var(--ds-border-input, ".concat(_colors.N40, ")"),
    borderHover: (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.update-border-radio-checkbox_7askv') ? "var(--ds-border-bold, ".concat(_colors.N100, ")") : "var(--ds-border-input, ".concat(_colors.N40, ")"),
    borderDisabled: "var(--ds-border-disabled, ".concat(_colors.N20, ")"),
    borderFocus: "var(--ds-border-focused, ".concat(_colors.B100, ")")
  },
  dark: {
    background: "var(--ds-background-input, ".concat(_colors.DN10, ")"),
    backgroundHover: "var(--ds-background-input-hovered, ".concat(_colors.DN30, ")"),
    backgroundActive: "var(--ds-background-input-pressed, ".concat(_colors.B200, ")"),
    backgroundChecked: "var(--ds-background-selected-bold, ".concat(_colors.B400, ")"),
    backgroundCheckedHover: "var(--ds-background-selected-bold-hovered, ".concat(_colors.B75, ")"),
    backgroundCheckedActive: "var(--ds-background-selected-bold-pressed, ".concat(_colors.B200, ")"),
    backgroundDisabled: "var(--ds-background-disabled, ".concat(_colors.DN10, ")"),
    dotChecked: "var(--ds-icon-inverse, ".concat(_colors.DN10, ")"),
    dotDisabled: "var(--ds-icon-disabled, ".concat(_colors.DN90, ")"),
    dotActive: "var(--ds-icon-inverse, ".concat(_colors.DN10, ")"),
    border: (0, _platformFeatureFlags.getBooleanFF)('platform.design-system-team.update-border-radio-checkbox_7askv') ? "var(--ds-border-bold, ".concat(_colors.DN200, ")") : "var(--ds-border-input, ".concat(_colors.DN80, ")"),
    borderHover: "var(--ds-border-input, ".concat(_colors.DN200, ")"),
    borderDisabled: "var(--ds-border-disabled, ".concat(_colors.DN10, ")"),
    borderFocus: "var(--ds-border-focused, ".concat(_colors.B75, ")")
  }
};
function getRadioCustomProperties(mode) {
  var radioColors = radioThemeColors[mode];
  return {
    '--local-background': radioColors.background,
    '--local-background-active': radioColors.backgroundActive,
    '--local-background-checked': radioColors.backgroundChecked,
    '--local-background-checked-active': radioColors.backgroundCheckedActive,
    '--local-background-checked-hover': radioColors.backgroundCheckedHover,
    '--local-background-disabled': radioColors.backgroundDisabled,
    '--local-background-hover': radioColors.backgroundHover,
    '--local-border': radioColors.border,
    '--local-border-disabled': radioColors.borderDisabled,
    '--local-border-hover': radioColors.borderHover,
    '--local-border-focus': radioColors.borderFocus,
    '--local-dot-active': radioColors.dotActive,
    '--local-dot-checked': radioColors.dotChecked,
    '--local-dot-disabled': radioColors.dotDisabled,
    '--local-invalid': "var(--ds-icon-danger, ".concat(_colors.R300, ")")
  };
}
