// @atlaskit/tmp-editor-statsig/dist/cjs/experiments-config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.editorExperimentsConfig = void 0;
var _experimentBuilders = require("./experiment-builders");
/* eslint-disable perfectionist/sort-object-types */
/* eslint-disable @atlaskit/editor/no-re-export */
// Entry file in package.json

// eslint-disable-next-line @typescript-eslint/consistent-type-imports -- Need value import for typeof

/**
 * Extract valid expected values.
 * - For multivariate experiments: returns union of valid string values (inferred from defaultValue type)
 * - For boolean experiments: returns only 'true' literal (cannot use 'false' as expected value)
 */

// Multivariate: use the default value type

/**
 * Extract valid default values.
 * - For boolean experiments: returns only 'false' literal (cannot use 'true' as default value)
 * - For multivariate experiments: returns the default value type (one of the string values)
 */

// Multivariate: use the default value type

/**
 * When adding a new experiment, you need to add it here.
 * Please follow the pattern established in the examples and any
 * existing experiments.
 */
var editorExperimentsConfig = exports.editorExperimentsConfig = {
  // new format to avoid collisions with other users when updating the file

  // Editor Platform experiments
  // lwoollard experiments

  // Added 2026-02-11
  ai_speech_to_text_in_editor: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'ai_speech_to_text_in_editor'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 22-12-2025
  confluence_load_editor_title_on_transition: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_load_editor_title_on_transition'
    },
    param: 'contentPlaceholder',
    defaultValue: false
  }),
  // Added 20-01-2026
  cc_editor_lcm_readonly_initial: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_lcm_readonly_initial'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 02-12-2025
  cc_fix_hydration_ttvc: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_fix_hydration_ttvc'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-25
  'editor_a11y__enghealth-46814_fy26': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_a11y__enghealth-46814_fy26'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-20
  editor_a11y_decision_aria_label: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_a11y_decision_aria_label',
      jira: 'editor_a11y_decision_aria_label'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 03-09-2025
  cc_editor_interactivity_monitoring: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_interactivity_monitoring'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-03
  editor_tinymce_full_width_mode: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_tinymce_full_width_mode'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 05-03-2026
  platform_editor_ai_headingautocomplete: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_headingautocomplete'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-16
  platform_editor_remove_important_in_render_ext: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_remove_important_in_render_ext'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-10
  confluence_max_width_content_appearance: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_max_width_content_appearance'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-15
  confluence_max_width_breakout_extension_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_max_width_breakout_extension_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  confluence_remix_icon_right_side: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_remix_icon_right_side'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  cc_editor_insm_doc_size_stats: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_insm_doc_size_stats'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-26
  cc_editor_fix_insm_inp_buffer: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_fix_insm_inp_buffer'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  cwr_blank_object_experiment: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cwr_blank_object_experiment'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-30
  cc_editor_insm_outlier_events: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'cc_editor_insm_outlier_events'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025-12-09
  cc_editor_hover_link_overlay_css_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_hover_link_overlay_css_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-7
  platform_editor_table_sticky_header_patch_10: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_sticky_header_patch_10'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-17
  platform_editor_renderer_extension_width_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_renderer_extension_width_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-12
  platform_editor_media_vc_fixes: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_media_vc_fixes'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-06
  platform_editor_media_reliability_observability: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_media_reliability_observability'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-24
  platform_editor_media_external_badge_bbc_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_media_external_badge_bbc_fix',
      bitbucket: 'platform_editor_media_external_badge_bbc_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-20
  platform_editor_annotations_sync_on_docchange: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_annotations_sync_on_docchange'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-27
  platform_editor_reduce_noisy_steps_ncs: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_reduce_noisy_steps_ncs'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-02
  platform_editor_block_control_optimise_render: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_block_control_optimise_render'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-08-08
  'example-boolean': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package_boolean'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-08-08
  'example-multivariate': (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package_multivariate'
    },
    param: 'variant',
    values: ['one', 'two', 'three'],
    defaultValue: 'one'
  }),
  // Added 2024-08-08
  'test-new-experiments-package': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_editor_experiment_test_new_package',
      jira: 'jira_editor_experiment_test_new_package'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-09
  platform_editor_react19_migration: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_react19_migration'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-05
  support_table_in_comment: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_support_table_in_comment_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-07
  platform_editor_exp_lazy_node_views: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_exp_lazy_node_views'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-09-18
  platform_renderer_table_sticky_scrollbar: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_renderer_table_sticky_scrollbar'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-05
  'company_hub_carousel_thumbnails-refactor': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'company_hub_carousel_thumbnails-refactor'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-10
  platform_editor_experience_tracking: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_experience_tracking'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05 - gates DOM mutation observer for experience tracking (toolbar + block menu)
  platform_editor_experience_tracking_observer: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_experience_tracking_observer',
      jira: 'platform_editor_experience_tracking_observer'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-26
  platform_editor_prevent_toolbar_layout_shifts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_prevent_toolbar_layout_shifts'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-01
  comment_on_bodied_extensions: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_comment_on_bodied_extensions'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-14
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_advanced_layouts/setup
  advanced_layouts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_advanced_layouts'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-3-15
  // https://console.statsig.com/LqivKg6ADZZaGczRfBKfX/experiments/platform_editor_single_column_layout/setup
  single_column_layouts: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_single_column_layout'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-10-08
  support_table_in_comment_jira: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_support_table_in_comment_jira'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  'platform_editor_ai-prompts-placeholder': (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai-prompts-placeholder'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2024-11-26
  platform_editor_ai_unsplash_page_header: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_unsplash_page_header'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-24
  platform_editor_paste_actions_menu: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_paste_actions_menu'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-03
  platform_editor_aifc_fix_button_viewed_analytics: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_aifc_fix_button_viewed_analytics'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2024-12-05
  platform_editor_blockquote_in_text_formatting_menu: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_blockquote_in_text_formatting_menu'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-13
  platform_editor_element_drag_and_drop_multiselect: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_element_drag_and_drop_multiselect'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-19
  platform_editor_ai_edit_response_in_preview: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_edit_response_in_preview'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-02-10
  platform_editor_controls: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_controls',
      jira: 'platform_editor_controls_jira'
    },
    param: 'cohort',
    values: ['control', 'variant1'],
    defaultValue: 'control'
  }),
  // Added 2025-08-01
  platform_editor_controls_block_controls_state_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_controls_block_controls_state_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-03
  platform_editor_hydratable_ui: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_hydratable_ui'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-02
  platform_editor_inline_card_dispatch_guard: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_inline_card_dispatch_guard'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-14
  // https://switcheroo.atlassian.com/ui/gates/b159b45a-86d9-4f4b-b482-f9aca5b615d6/key/platform_editor_offline_editing_web
  platform_editor_offline_editing_web: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_offline_editing_web'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-31
  platform_editor_lovability_suppress_toolbar_event: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_lovability_suppress_toolbar_event',
      jira: 'platform_editor_lovability_suppress_toolbar_event'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-17
  platform_editor_tables_drag_and_drop: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_tables_drag_and_drop'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-17
  platform_editor_tables_table_selector: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_tables_table_selector'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-23
  platform_editor_usesharedpluginstateselector: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_usesharedpluginstateselector'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_table_display_mode_in_to_dom: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_display_mode_in_to_dom'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-04-23
  platform_renderer_fix_analytics_memo_callback: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_renderer_fix_analytics_memo_callback'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-10
  platform_editor_no_cursor_on_edit_page_init: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_no_cursor_on_edit_page_init'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_clean_up_widget_mark_logic: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_clean_up_widget_mark_logic',
      jira: 'platform_editor_clean_up_widget_mark_logic'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-14
  platform_editor_debounce_portal_provider: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_debounce_portal_provider'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_renderer_shadow_observer_cleanup: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_renderer_shadow_observer_cleanup',
      confluence: 'platform_editor_renderer_shadow_observer_cleanup'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-08 - Jira work sync description comment summary
  'jira-work-sync-desc-comment-summary': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'jira-work-sync-desc-comment-summary'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-15
  platform_editor_breakout_resizing: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_breakout_resizing'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-17
  editor_fix_embed_width_expand: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_fix_embed_width_expand'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-17
  platform_editor_aifc_sync_block_stream_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_aifc_sync_block_stream_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-05-07
  platform_editor_ai_quickstart_command: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_quickstart_command'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-05
  platform_editor_block_controls_perf_optimization: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_block_controls_perf_optimization'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-16
  platform_editor_enable_single_player_step_merging: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_enable_single_player_step_merging'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_toolbar_hide_overflow_menu: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_hide_overflow_menu',
      jira: 'platform_editor_toolbar_hide_overflow_menu'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-02
  platform_editor_code_block_fold_gutter: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_code_block_fold_gutter'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  //Added 2025-07-25
  platform_editor_extension_styles: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_extension_styles'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-24
  platform_editor_find_and_replace_improvements: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_find_and_replace_improvements'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-06-26
  platform_editor_feedback_mandatory_rating: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_feedback_mandatory_rating'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-15
  platform_editor_preview_panel_responsiveness: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_preview_panels_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-18
  platform_editor_preview_panel_linking_exp: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_preview_panel_linking_exp_conf',
      jira: 'platform_editor_preview_panel_linking_exp_jira'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-05
  platform_editor_table_update_table_ref: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_update_table_ref'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-27
  platform_editor_table_excerpts_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_excerpts_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-10
  platform_editor_task_item_styles: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_task_item_styles'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-17
  platform_hover_card_preview_panel: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_hover_card_preview_panel',
      jira: 'platform_hover_card_preview_panel'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025-09-17
  // Added 2025-11-25
  smart_link_confluence_short_link_analytics: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'smart-link-confluence-short-link-analytics',
      jira: 'smart-link-confluence-short-link-analytics'
    },
    param: 'cohort',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2025--8-05
  platform_editor_block_menu: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_block_menu',
      jira: 'platform_editor_block_menu_jira'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-23
  // Added 2025-07-24
  editor_enghealth_hyperlink_toolbar_aria_values: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_enghealth_hyperlink_toolbar_aria_values'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  //Added 2025-07-16
  platform_editor_toolbar_aifc: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_aifc_confluence'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-04
  platform_editor_floating_toolbar_button_aria_label: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_floating_toolbar_button_aria_label'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-31
  platform_editor_breakout_interaction_rerender: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_breakout_interaction_rerender'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  //Added 2025-11-19
  platform_editor_pasting_text_in_panel: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_pasting_text_in_panel'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-19
  platform_editor_comment_editor_border_radius: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_comment_editor_border_radius'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-07-30
  platform_editor_blocktaskitem_node_tenantid: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_blocktaskitem_node_tenantid'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-08
  platform_editor_blocktaskitem_patch_1: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_blocktaskitem_patch_1'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-19
  cc_editor_limited_mode_expanded: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_limited_mode_expanded'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-11
  platform_editor_native_embeds: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_native_embeds'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-05
  platform_editor_native_anchor_with_dnd: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_native_anchor_with_dnd'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  'cc_perf-insights-cards_extension_a11y_list': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_perf-insights-cards_extension_a11y_list'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-12
  ccpi_fix_broken_uploaded_img: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'ccpi_fix_broken_uploaded_img'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-05
  'editor-a11y-fy26-keyboard-move-row-column': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor-a11y-fy26-keyboard-move-row-column',
      jira: 'editor-a11y-fy26-keyboard-move-row-column'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-28
  editor_enable_image_alignment_in_expand: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_enable_image_alignment_in_expand'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-05
  platform_editor_august_a11y: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_august_a11y'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-10
  platform_synced_block: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_synced_block',
      jira: 'platform_synced_block_jira'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  platform_synced_block_use_new_source_nodeview: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_synced_block_use_new_source_nodeview',
      jira: 'platform_synced_block_use_new_source_nodeview'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  platform_synced_block_patch_6: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_synced_block_patch_6',
      jira: 'platform_synced_block_patch_6'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-08-18
  platform_editor_locale_datepicker: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_locale_datepicker'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-16
  platform_editor_prevent_taskitem_remount: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_prevent_taskitem_remount'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-08
  platform_editor_table_toolbar_perf_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_toolbar_perf_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-01
  cc_editor_ai_content_mode: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'cc_editor_ai_content_mode'
    },
    param: 'variant',
    values: ['control', 'test'],
    defaultValue: 'control'
  }),
  // Added 2026-01-09
  platform_editor_render_bodied_extension_as_inline: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_render_bodied_extension_as_inline'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-03
  platform_editor_ai_quick_prompt_iw_cc: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_quick_prompt_iw_cc'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-25
  platform_editor_ai_escape_early_for_unhealthy_node: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_escape_early_for_unhealthy_node'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-04
  confluence_compact_text_format: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_compact_text_format'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_unify_native_dnd_selectors: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_unify_native_dnd_selectors',
      jira: 'platform_editor_unify_native_dnd_selectors'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-10
  platform_editor_fix_button_name_violation_in_table: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_button_name_violation_in_table'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  platform_editor_abort_ufo_on_user_interaction: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_abort_ufo_on_user_interaction'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-11
  platform_editor_add_aria_checked_to_inline_img_btn: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_add_aria_checked_to_inline_image_btn'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-10
  platform_use_llm_space_recommendations: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_use_llm_space_recommendations'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-09-23
  cc_editor_ttvc_release_bundle_one: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_ttvc_release_bundle_one'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-07
  platform_editor_plain_text_support: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_plain_text_support',
      confluence: 'platform_editor_plain_text_support'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-01
  // Added 2025-10-13
  platform_editor_media_error_analytics: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_media_error_analytics'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-22
  platform_editor_remove_bidi_char_warning: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_remove_bidi_char_warning'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-27
  platform_editor_table_sticky_header_improvements: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_sticky_header_improvements'
    },
    param: 'cohort',
    values: ['control', 'test_with_overflow', 'test_without_overflow'],
    defaultValue: 'control'
  }),
  // Added 2025-11-04
  platform_editor_disable_lazy_load_media: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_disable_lazy_load_media'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-10-29
  platform_editor_lovability_inline_code: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_lovability_inline_code'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-16
  platform_editor_disable_lcm_copy_button: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_disable_lcm_copy_button'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-28-01
  platform_editor_primary_toolbar_early_exit: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_primary_toolbar_early_exit',
      jira: 'platform_editor_primary_toolbar_early_exit'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-05
  platform_editor_lovability_emoji_scaling: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_lovability_emoji_scaling'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_flex_based_centering: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_flex_based_centering'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-18
  editor_a11y_7152_profile_card_tab_order: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_a11y_7152_profile_card_tab_order'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-19
  platform_editor_lovability_navigation_fixes: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_lovability_navigation_fixes'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-11-27
  platform_sl_3p_unauth_paste_as_block_card: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'platform_sl_3p_unauth_paste_as_block_card',
      jira: 'platform_sl_3p_unauth_paste_as_block_card'
    },
    values: ['control', 'card_by_default_only', 'card_by_default_and_new_design'],
    param: 'cohort',
    defaultValue: 'control'
  }),
  // Added 2026-03-17 - 3P auth Rovo actions on smart link hover card (boolean: isEnabled)
  platform_sl_3p_auth_rovo_action: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_sl_3p_auth_rovo_action',
      jira: 'platform_sl_3p_auth_rovo_action'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-03
  platform_editor_use_pmr_for_collab_presence_non_ic: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_use_pmr_for_collab_presence_non_ic',
      jira: 'platform_editor_use_pmr_for_collab_presence_non_ic'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-03
  platform_editor_use_pmr_for_collab_presence_in_ic: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_use_pmr_for_collab_presence_in_ic',
      jira: 'platform_editor_use_pmr_for_collab_presence_in_ic'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-23
  platform_editor_to_use_pmr_for_collab_edit_none_ic: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_to_use_pmr_for_collab_edit_none_ic'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-12
  platform_editor_toolbar_delay_render_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_delay_render_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-08
  platform_editor_add_image_editing: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_add_image_editing'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_toolbar_aifc_use_editor_typography: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_aifc_use_editor_typography',
      jira: 'platform_editor_toolbar_aifc_use_editor_typography'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_toolbar_submenu_open_click: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_submenu_open_click'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-30
  platform_editor_table_sticky_header_patch_9: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_sticky_header_patch_9'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-22
  platform_editor_localid_ime_composition_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_localid_ime_composition_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-12-22
  platform_editor_sample_renderer_rendered_event: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_sample_renderer_rendered_event'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2025-01-21
  platform_editor_copy_link_a11y_inconsistency_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_copy_link_a11y_inconsistency_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-25
  platform_editor_expand_paste_in_comment_editor: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_expand_paste_in_comment_editor'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-12
  platform_editor_table_sticky_header_patch_11: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_sticky_header_patch_11'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-16
  platform_editor_ai_fix_insert_after_selection: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_fix_insert_after_selection'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-15
  platform_editor_fix_advanced_codeblocks_crlf: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_advanced_codeblocks_crlf'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-08
  platform_editor_focus_on_chromeless_editor: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_focus_on_chromeless_editor'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-02
  platform_editor_ai_disable_bridge_without_ai: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_disable_bridge_without_ai'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-16
  platform_editor_ai_edit_unsupported_content: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_edit_unsupported_content'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-19
  platform_editor_copy_paste_issue_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_copy_paste_issue_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-27 - AI create-page expand collapse fix
  platform_editor_aifc_expand_collapses_oncreate_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_aifc_expand_collapses_oncreate_fix',
      jira: 'platform_editor_aifc_expand_collapses_oncreate_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-22
  platform_editor_ai_loading_responsive_width: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_loading_responsive_width',
      jira: 'platform_editor_ai_loading_responsive_width'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-22
  platform_editor_nested_drag_handle_icon: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_nested_drag_handle_icon'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-23
  collab_bypass_out_of_sync_period_experiment: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'collab_bypass_out_of_sync_period_experiment'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-16
  platform_editor_ignore_metadata_connection_errors: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_collab_provider_suppress_metadata_errors'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-26
  platform_editor_ai_fix_streaming_json_escape: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_fix_streaming_json_escape'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-18
  platform_editor_table_a11y_eslint_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_a11y_eslint_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-29
  platform_editor_renderer_toolbar_updates: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_renderer_toolbar_updates'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-30
  platform_editor_fix_cross_origin_editor_focus: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_cross_origin_editor_focus'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_comment_rovoinlinechat_improvement: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_comment_rovoinlinechat_improvement'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  confluence_ttvc_inline_extensions: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_ttvc_inline_extensions'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-03
  platform_editor_enghealth_a11y_jan_fixes: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_enghealth_a11y_jan_fixes'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-04
  platform_editor_emoji_tooltips_on_hover: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_emoji_tooltips_on_hover'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-16-05
  platform_editor_outdated_browser_update: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_outdated_browser_update'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-05
  cc_fd_db_top_editor_toolbar: (0, _experimentBuilders.createMultivariateExperiment)({
    productKeys: {
      confluence: 'cc_fd_db_top_editor_toolbar'
    },
    param: 'cohort',
    values: ['control', 'new-description', 'orig-description'],
    defaultValue: 'control'
  }),
  // Added 2026-02-05 - A11Y-10416
  editor_a11y_role_textbox: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'editor_a11y_role_textbox'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-01-28
  platform_editor_smartlink_local_cache: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_smartlink_local_cache'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-05
  platform_editor_toolbar_split_button_ui: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_split_button_ui'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-06
  platform_rovo_inline_chat_missing_analytics_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_rovo_inline_chat_missing_analytics_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-12
  platform_editor_bodiedextension_layoutshift_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_bodiedextension_layoutshift_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-16
  platform_safari_cursor_typeahead_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_safari_cursor_typeahead_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-17
  confluence_frontend_fix_date_hydration_error: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_frontend_fix_date_hydration_error'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-19
  platform_editor_a11y_eslint_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_a11y_eslint_fix',
      jira: 'platform_editor_a11y_eslint_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_fix_advanced_codeblocks_crlf_patch: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_advanced_codeblocks_crlf_patch',
      jira: 'platform_editor_fix_advanced_codeblocks_crlf_patch'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-23
  'cc-maui-experiment': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc-maui-experiment'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  'cc-mui-slides-experiment': (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc-mui-slides-experiment'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-27
  confluence_frontend_cwr_outcome_type_picker: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_frontend_cwr_outcome_type_picker'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-12
  platform_editor_a11y_escape_link_dialog: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_a11y_escape_link_dialog',
      jira: 'platform_editor_a11y_escape_link_dialog'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_vc90_transition_table_border: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_vc90_transition_table_border'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_vc90_transition_expand_icon: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_vc90_transition_expand_icon'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_vc90_transition_panel_icon: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_vc90_transition_panel_icon'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_vc90_transition_mentions: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_vc90_transition_mentions'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-11
  platform_editor_prosemirror_rendered_data: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_prosemirror_rendered_data',
      jira: 'platform_editor_prosemirror_rendered_data'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-19
  platform_editor_flexible_list_indentation: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_flexible_list_indentation'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-24
  platform_editor_flexible_list_schema: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_flexible_list_schema'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-24
  platform_editor_emoji_default_scale: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_emoji_default_scale',
      jira: 'platform_editor_emoji_default_scale'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-26
  platform_editor_toolbar_update_jira_config: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_toolbar_update_jira_config'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-25
  platform_editor_a11y_typeahead_tab_keypress: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_a11y_typeahead_tab_keypress'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-02-27
  platform_editor_early_exit_return_draft: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_early_exit_return_draft'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-03
  platform_editor_ai_update_actioned_event_timeout: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_update_actioned_event_timeout'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-02
  platform_editor_discarded_unified_analytic_events: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_discarded_unified_analytic_events',
      jira: 'platform_editor_discarded_unified_analytic_events'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  platform_editor_appearance_shared_state: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_appearance_shared_state'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_ai_blockmenu_integration: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_blockmenu_integration'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added for space shortcut plugin: when on, space shortcut is not added if AI opt-in is not enabled
  platform_editor_ai_aifc_space_shortcut_patch: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_ai_aifc_space_shortcut_patch'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  platform_editor_rovobutton_smartlink_toolbar_exp: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_rovobutton_smartlink_toolbar_exp'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-02
  platform_editor_a11y_table_wrapper_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_a11y_table_wrapper_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-11
  platform_editor_renderer_error_boundary_stable_key: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_renderer_error_boundary_stable_key',
      confluence: 'platform_editor_renderer_error_boundary_stable_key'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-04
  confluence_fe_renderer_inline_node_mark_color_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_fe_renderer_inline_node_mark_color_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_fix_editor_unhandled_type_errors: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_editor_unhandled_type_errors'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  platform_editor_table_resizer_extended_zone: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_resizer_extended_zone'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_insert_location_check: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      jira: 'platform_editor_insert_location_check',
      confluence: 'platform_editor_insert_location_check'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-05
  cc_drag_and_drop_smart_link_from_content_to_tree: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_drag_and_drop_smart_link_from_content_to_tree'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-19
  platform_editor_fix_table_row_drag_drop_target: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_table_row_drag_drop_target'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-11
  platform_editor_remove_grid_init_reflow: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_remove_grid_init_reflow'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-07
  platform_editor_table_remove_last_cell_decoration: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_remove_last_cell_decoration'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  platform_editor_small_font_size: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_small_font_size',
      jira: 'platform_editor_small_font_size'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_diff_plugin_extended: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_diff_plugin_extended'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_fix_media_toolbar_border_dropdown: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_media_toolbar_border_dropdown'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-24
  platform_editor_fix_media_picker_hidden: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_media_picker_hidden',
      jira: 'platform_editor_fix_media_picker_hidden'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-10
  platform_editor_analyse_table_with_merged_cells: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_analyse_table_with_merged_cells'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  cc_editor_ttvc_media_hold_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'cc_editor_ttvc_media_hold_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-06
  platform_editor_single_player_expand: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_single_player_expand'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-11
  platform_editor_are_nodes_equal_ignore_mark_order: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_are_nodes_equal_ignore_mark_order',
      jira: 'platform_editor_are_nodes_equal_ignore_mark_order'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-11
  platform_editor_disable_last_node_para: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_disable_last_node_para',
      jira: 'platform_editor_disable_last_node_para'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-09
  platform_editor_remove_collab_step_metrics: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_remove_collab_step_metrics'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-13
  platform_editor_editor_centre_content_on_find: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_editor_centre_content_on_find'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-24
  platform_editor_table_ref_optimisation: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_table_ref_optimisation'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-20
  platform_editor_chromeless_expand_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_chromeless_expand_fix',
      jira: 'platform_editor_chromeless_expand_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-23
  platform_editor_inline_media_copy_paste_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_inline_media_copy_paste_fix',
      jira: 'platform_editor_inline_media_copy_paste_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-13
  platform_editor_misaligned_ai_screens_firefox_fix: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_misaligned_ai_screens_firefox_fix',
      jira: 'platform_editor_misaligned_ai_screens_firefox_fix'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-19
  platform_editor_toolbar_two_stage_hydration: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_toolbar_two_stage_hydration'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-15
  platform_editor_sync_block_ssr_config: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_sync_block_ssr_config'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-18
  platform_editor_selection_toolbar_block_handle: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_selection_toolbar_block_handle'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-23
  confluence_toc_nav_a11y: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'confluence_toc_nav_a11y'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-25
  platform_editor_improve_preset_builder_logging: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_improve_preset_builder_logging'
    },
    param: 'isEnabled',
    defaultValue: false
  }),
  // Added 2026-03-26
  platform_editor_fix_comment_border: (0, _experimentBuilders.createBooleanExperiment)({
    productKeys: {
      confluence: 'platform_editor_fix_comment_border'
    },
    param: 'isEnabled',
    defaultValue: false
  })
};
