// @atlaskit/onboarding/dist/es2019/styled/modal.js
/* eslint-disable @atlaskit/design-system/no-nested-styles */
/**
 * @jsxRuntime classic
 * @jsx jsx
 */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
import { h600 } from '@atlaskit/theme/typography';
const modalBodyStyles = css({
  padding: `${"var(--ds-space-500, 40px)"} ${"var(--ds-space-200, 16px)"}`,
  textAlign: 'center'
});

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage, @atlaskit/ui-styling-standard/no-array-arguments -- Ignored via go/DSP-18766
const modalHeadingStyles = css([
// eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
css(h600({
  theme: {
    mode: 'light'
  }
})), {
  marginBottom: "var(--ds-space-100, 8px)",
  color: 'inherit'
}]);
const modalImageStyles = css({
  width: '100%',
  height: 'auto',
  borderStartEndRadius: "var(--ds-border-radius, 3px)",
  borderStartStartRadius: "var(--ds-border-radius, 3px)",
  '@media (min-width: 320px) and (max-width: 480px)': {
    borderRadius: 0
  }
});
const modalActionContainerStyles = css({
  display: 'flex',
  padding: `${"var(--ds-space-0, 0px)"} ${"var(--ds-space-500, 40px)"} 36px`,
  justifyContent: 'center',
  flexDirection: 'row',
  flexFlow: 'wrap'
});
const modalActionContainerReversedStyles = css({
  flexDirection: 'row-reverse'
});
const modalActionItemStyles = css({
  margin: `${"var(--ds-space-0, 0px)"} ${"var(--ds-space-050, 4px)"} ${"var(--ds-space-050, 4px)"}`
});

/**
 * __Modal body__
 *
 * @internal
 */
export const ModalBody = ({
  children
}) => jsx("div", {
  css: modalBodyStyles
}, children);

/**
 * __Modal heading__
 *
 * @internal
 */
export const ModalHeading = ({
  children,
  id
}) => {
  return jsx("h1", {
    css: modalHeadingStyles,
    id: id
  }, children);
};

/**
 * __Modal image__
 *
 * @internal
 */
export const ModalImage = ({
  alt,
  src
}) => jsx("img", {
  css: modalImageStyles,
  alt: alt,
  src: src
});

/**
 * __Modal action container__
 *
 * @internal
 */
export const ModalActionContainer = ({
  children,
  shouldReverseButtonOrder
}) => jsx("div", {
  css: [modalActionContainerStyles, shouldReverseButtonOrder && modalActionContainerReversedStyles]
}, children);

/**
 * __Modal action item__
 *
 * @internal
 */
export const ModalActionItem = ({
  children
}) => jsx("div", {
  css: modalActionItemStyles
}, children);
