// @atlaskit/react-ufo/dist/es2019/vc/vc-observer-new/metric-calculator/fy25_03/index.js
import { fg } from '@atlaskit/platform-feature-flags';
import { expVal } from '../../../expVal';
import { containsDnDMutationInStyle } from '../../../vc-observer/observers/non-visual-styles/is-dnd-style-mutation';
import AbstractVCCalculatorBase from '../abstract-base-vc-calculator';
import { KNOWN_ATTRIBUTES_THAT_DOES_NOT_CAUSE_LAYOUT_SHIFTS, NON_VISUAL_ARIA_ATTRIBUTES, THIRD_PARTY_BROWSER_EXTENSION_ATTRIBUTES, DARK_READER_BROWSER_EXTENSION_ATTRIBUTES } from '../utils/constants';
import { isEntrySmartAnswersInSearch } from '../utils/is-entry-smart-answers-in-search';
import isViewportEntryData from '../utils/is-viewport-entry-data';
const ABORTING_WINDOW_EVENT = ['wheel', 'scroll', 'keydown', 'resize'];
const REVISION_NO = 'fy25.03';
const getConsideredEntryTypes = (include3p, excludeSmartAnswersInSearch) => {
  const entryTypes = ['mutation:child-element', 'mutation:element', 'mutation:attribute', 'layout-shift', 'layout-shift:same-rect', 'window:event'];

  // Only include third-party elements in TTVC calculation when explicitly requested
  if (include3p) {
    entryTypes.push('mutation:third-party-element');
    entryTypes.push('mutation:third-party-attribute');
  }
  if (!excludeSmartAnswersInSearch && fg('rovo_search_page_ttvc_ignoring_smart_answers_fix')) {
    entryTypes.push('mutation:smart-answers-element');
    entryTypes.push('mutation:smart-answers-attribute');
  }
  entryTypes.push('mutation:media');

  // Still included as part of TTVC v3
  entryTypes.push('mutation:attribute:non-visual-input-name');
  return entryTypes;
};
export default class VCCalculator_FY25_03 extends AbstractVCCalculatorBase {
  constructor(revisionNo) {
    super(revisionNo !== null && revisionNo !== void 0 ? revisionNo : REVISION_NO);
  }
  isEntryIncluded(entry, include3p, excludeSmartAnswersInSearch) {
    if (!getConsideredEntryTypes(include3p, excludeSmartAnswersInSearch).includes(entry.data.type)) {
      return false;
    }
    if (entry.data.type === 'mutation:attribute:framework-routing') {
      return false;
    }
    if (excludeSmartAnswersInSearch && isEntrySmartAnswersInSearch(entry) && !fg('rovo_search_page_ttvc_ignoring_smart_answers_fix')) {
      return false;
    }
    if (entry.data.type === 'mutation:media' && fg('media-perf-uplift-mutation-fix')) {
      const entryData = entry.data;
      const attributeName = entryData.attributeName;
      if (attributeName && (/data-(test|file|context)-\S+/g.test(attributeName) || attributeName === 'data-cursor' || attributeName === 'alt' || (attributeName === 'localid' || attributeName === 'contenteditable' || attributeName === 'anchor-name') && expVal('platform_editor_media_vc_fixes', 'isEnabled', false))) {
        return false;
      }

      // special case for style attribute to ignore only anchor-name changes
      if (expVal('platform_editor_media_vc_fixes', 'isEnabled', false)) {
        if (containsDnDMutationInStyle({
          attributeName: entryData.attributeName,
          oldValue: entryData.oldValue,
          newValue: entryData.newValue
        })) {
          return false;
        }
      }
    }
    if (entry.data.type === 'mutation:attribute') {
      const entryData = entry.data;
      const attributeName = entryData.attributeName;
      if (!attributeName || KNOWN_ATTRIBUTES_THAT_DOES_NOT_CAUSE_LAYOUT_SHIFTS.includes(attributeName)) {
        return false;
      }
      if (attributeName.startsWith('data-test') || NON_VISUAL_ARIA_ATTRIBUTES.includes(attributeName) || THIRD_PARTY_BROWSER_EXTENSION_ATTRIBUTES.includes(attributeName) && fg('platform_ufo_exclude_3p_extensions_from_ttvc') || DARK_READER_BROWSER_EXTENSION_ATTRIBUTES.includes(attributeName) && fg('platform_ufo_exclude_dark_reader_extension')) {
        return false;
      }
      return true;
    }
    if (isViewportEntryData(entry.data) && !entry.data.visible) {
      return false;
    }
    return true;
  }
  getVCCleanStatus(filteredEntries) {
    let dirtyReason = '';
    let abortTimestamp = -1;
    const hasAbortEvent = filteredEntries.some(entry => {
      if (entry.data.type === 'window:event') {
        const data = entry.data;
        const eventType = data.eventType;
        if (ABORTING_WINDOW_EVENT.includes(eventType)) {
          dirtyReason = eventType === 'keydown' ? 'keypress' : eventType;
          abortTimestamp = Math.round(entry.time);
          return true;
        }
      }
      return false;
    });
    if (hasAbortEvent && dirtyReason) {
      return {
        isVCClean: false,
        dirtyReason,
        abortTimestamp
      };
    }
    return {
      isVCClean: true
    };
  }
}
