// @atlaskit/inline-dialog/dist/es2019/inline-dialog-container.js
/* inline-dialog-container.tsx generated by @compiled/babel-plugin v0.39.1 */
import "./inline-dialog-container.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import { forwardRef } from 'react';
const containerStyles = null;
/**
 * __Container__
 *
 * A container is used as a styled wrapper around the contents of an inline dialog.
 * Note that the styles here are merged with the style prop that comes from the popper.js library.
 *
 */
export const Container = /*#__PURE__*/forwardRef(({
  children,
  onBlur,
  onClick,
  onFocus,
  style,
  testId
}, ref) => {
  return /*#__PURE__*/React.createElement("div", {
    "data-testid": testId,
    onBlur: onBlur,
    onClick: onClick,
    onFocus: onFocus,
    ref: ref
    // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
    ,
    style: style,
    className: ax(["_2rkofajl _vchh1ntv _p12flti4 _c71llti4 _1pbyowjs _bfhk1bhr _16qs130s _syazi7uo _85i5pxbi _1q51pxbi _y4ti1ejb _bozg1ejb _nt751r31 _49pcglyw _1hvw1o36"])
  }, typeof children === 'function' ? children() : children);
});
