// @atlaskit/tag/dist/cjs/tag/internal/shared/base.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@emotion/react");
var _components = require("@atlaskit/theme/components");
var _constants = require("../../../constants");
var theme = _interopRequireWildcard(require("../../../theme"));
var _excluded = ["before", "contentElement", "after", "testId", "appearance", "style", "color", "href", "className"];
/** @jsx jsx */
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var baseStyles = (0, _react2.css)({
  display: 'inline-flex',
  height: _constants.tagHeight,
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  margin: _constants.defaultMargin,
  padding: "var(--ds-space-0, 0px)",
  position: 'relative',
  backgroundColor: "var(".concat(_constants.cssVar.color.background.default, ")"),
  borderRadius: "var(".concat(_constants.cssVar.borderRadius, ")"),
  color: "var(".concat(_constants.cssVar.color.text.default, ")"),
  cursor: 'default',
  lineHeight: 1,
  overflow: 'hidden'
});
var interactiveStyles = (0, _react2.css)({
  '&:hover': {
    backgroundColor: "var(".concat(_constants.cssVar.color.background.hover, ")")
  },
  '&:active': {
    backgroundColor: "var(".concat(_constants.cssVar.color.background.active, ")")
  }
});
var focusRingStyles = (0, _react2.css)({
  '&:focus-within': {
    boxShadow: "0 0 0 2px var(".concat(_constants.cssVar.color.focusRing, ")"),
    outline: 'none'
  }
});
var nonStandardLinkStyles = (0, _react2.css)({
  '&:active': {
    color: "var(".concat(_constants.cssVar.color.text.active, ")")
  }
});
var BaseTag = /*#__PURE__*/_react.default.forwardRef(function BaseTag(_ref, ref) {
  var _objectSpread2;
  var before = _ref.before,
    contentElement = _ref.contentElement,
    after = _ref.after,
    testId = _ref.testId,
    _ref$appearance = _ref.appearance,
    appearance = _ref$appearance === void 0 ? 'default' : _ref$appearance,
    style = _ref.style,
    _ref$color = _ref.color,
    color = _ref$color === void 0 ? 'standard' : _ref$color,
    href = _ref.href,
    className = _ref.className,
    other = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _useGlobalTheme = (0, _components.useGlobalTheme)(),
    mode = _useGlobalTheme.mode;
  var isLink = Boolean(href);
  var isRemovable = Boolean(after);
  var isInteractive = isLink || isRemovable;
  var isStandardLink = isLink && color === 'standard';

  // Change link text color if  the tag is standard color
  var textLinkColors = isStandardLink ? theme.textColors['standardLink'][mode] : theme.textColors[color][mode];
  var backgroundHoverColors = isRemovable && !isLink ? theme.backgroundColors[color][mode] : theme.linkHoverBackgroundColors[color][mode];
  var backgroundActiveColors = isRemovable && !isLink ? theme.backgroundColors[color][mode] : theme.linkActiveBackgroundColors[color][mode];
  return (0, _react2.jsx)("span", (0, _extends2.default)({}, other, {
    ref: ref,
    css: [baseStyles, (isRemovable || isLink) && focusRingStyles, isLink && !isStandardLink && nonStandardLinkStyles, isInteractive && interactiveStyles],
    style: _objectSpread((_objectSpread2 = {}, (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.text.default, theme.textColors[color][mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.text.hover, theme.textHoverColors[color][mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.text.active, theme.textActiveColors[color][mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.text.link, textLinkColors), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.background.default, theme.backgroundColors[color][mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.background.hover, backgroundHoverColors), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.background.active, backgroundActiveColors), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.focusRing, theme.focusRingColors[mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.removeButton.default, theme.removeButtonColors[color]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.color.removeButton.hover, theme.removeButtonHoverColors[color][mode]), (0, _defineProperty2.default)(_objectSpread2, _constants.cssVar.borderRadius, theme.borderRadius[appearance]), _objectSpread2), style),
    className: className,
    "data-testid": testId
  }), before, contentElement, after);
});

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = BaseTag;
