// @forge/events/out/appEvents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.appEvents = void 0;
const queries_1 = require("./queries");
const api_1 = require("@forge/api");
const errorTypes = {
    400: 'VALIDATION_ERROR',
    401: 'AUTHENTICATION_ERROR',
    403: 'AUTHORIZATION_ERROR',
    429: 'RATE_LIMIT',
    500: 'SERVICE_ERROR',
    503: 'SERVICE_UNAVAILABLE'
};
const endpoint = '/forge/events/v1/app-events';
exports.appEvents = {
    async publish(events) {
        const eventsArray = Array.isArray(events) ? events : [events];
        const body = {
            events: eventsArray.map((e) => ({
                key: e.key
            }))
        };
        const response = await (0, queries_1.post)(endpoint, body, api_1.__requestAtlassianAsApp);
        const responseBody = await response.json();
        if (!response.ok) {
            return {
                type: 'error',
                errorType: errorTypes[response.status] ?? 'OTHER',
                errorMessage: getErrorMessage(responseBody)
            };
        }
        return {
            type: 'success',
            failedEvents: responseBody.failedEvents ?? []
        };
    }
};
function getErrorMessage(responseBody) {
    if (responseBody.errorMessages && responseBody.errorMessages.length > 0) {
        return responseBody.errorMessages.join(', ');
    }
    else if (responseBody.errors) {
        return Object.entries(responseBody.errors)
            .map(([key, value]) => `${key}: ${value}`)
            .join(', ');
    }
    else {
        return JSON.stringify(responseBody);
    }
}
