// @forge/events/out/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvocationLimitReachedError = exports.JobDoesNotExistError = exports.InternalServerError = exports.PartialSuccessError = exports.RateLimitError = exports.NoEventsToPushError = exports.PayloadTooBigError = exports.TooManyEventsError = exports.InvalidPayloadError = exports.InvalidQueueNameError = exports.InvalidPushSettingsError = exports.EventsError = void 0;
class EventsError extends Error {
    constructor(message) {
        super(message);
    }
}
exports.EventsError = EventsError;
class InvalidPushSettingsError extends EventsError {
}
exports.InvalidPushSettingsError = InvalidPushSettingsError;
class InvalidQueueNameError extends EventsError {
}
exports.InvalidQueueNameError = InvalidQueueNameError;
class InvalidPayloadError extends EventsError {
}
exports.InvalidPayloadError = InvalidPayloadError;
class TooManyEventsError extends EventsError {
}
exports.TooManyEventsError = TooManyEventsError;
class PayloadTooBigError extends EventsError {
}
exports.PayloadTooBigError = PayloadTooBigError;
class NoEventsToPushError extends EventsError {
}
exports.NoEventsToPushError = NoEventsToPushError;
class RateLimitError extends EventsError {
}
exports.RateLimitError = RateLimitError;
class PartialSuccessError extends EventsError {
    result;
    failedEvents;
    constructor(message, result, failedEvents) {
        super(message);
        this.result = result;
        this.failedEvents = failedEvents;
    }
}
exports.PartialSuccessError = PartialSuccessError;
class InternalServerError extends EventsError {
    constructor(message, errorCode, details) {
        super(message);
        this.errorCode = errorCode;
        this.details = details;
    }
    errorCode;
    details;
}
exports.InternalServerError = InternalServerError;
class JobDoesNotExistError extends EventsError {
}
exports.JobDoesNotExistError = JobDoesNotExistError;
class InvocationLimitReachedError extends EventsError {
}
exports.InvocationLimitReachedError = InvocationLimitReachedError;
