// @forge/events/out/jobProgress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobProgress = void 0;
const api_1 = require("@forge/api");
const queries_1 = require("./queries");
const validators_1 = require("./validators");
class JobProgress {
    queueParams;
    id;
    apiClient;
    constructor(queueParams, id, apiClient = api_1.__requestAtlassianAsApp) {
        this.queueParams = queueParams;
        this.id = id;
        this.apiClient = apiClient;
    }
    async getStats() {
        const getStatsRequest = {
            queueName: this.queueParams.key,
            jobId: this.id,
            time: new Date().toISOString()
        };
        (0, validators_1.validateGetStatsPayload)(getStatsRequest);
        const response = await (0, queries_1.post)(queries_1.GET_STATS_PATH, getStatsRequest, this.apiClient);
        await (0, validators_1.validateGetStatsAPIResponse)(response, getStatsRequest);
        const { success, inProgress, failed } = await response.json();
        return { success, inProgress, failed };
    }
    async cancel() {
        const cancelJobRequest = {
            queueName: this.queueParams.key,
            jobId: this.id,
            time: new Date().toISOString()
        };
        (0, validators_1.validateCancelJobRequest)(cancelJobRequest);
        const response = await (0, queries_1.post)(queries_1.CANCEL_JOB_PATH, cancelJobRequest, this.apiClient);
        await (0, validators_1.validateCancelJobAPIResponse)(response, cancelJobRequest);
    }
}
exports.JobProgress = JobProgress;
