// @forge/events/out/queue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Queue = void 0;
const api_1 = require("@forge/api");
const queries_1 = require("./queries");
const validators_1 = require("./validators");
const uuid_1 = require("uuid");
const jobProgress_1 = require("./jobProgress");
class Queue {
    queueParams;
    apiClient;
    constructor(queueParams, apiClient = api_1.__requestAtlassianAsApp) {
        this.queueParams = queueParams;
        this.apiClient = apiClient;
        (0, validators_1.validateQueueKey)(this.queueParams.key);
    }
    async push(events) {
        const validEvents = (0, validators_1.validatePushEvents)(events);
        const queueName = this.queueParams.key;
        const jobId = (0, uuid_1.v4)();
        const pushRequest = {
            queueName: queueName,
            jobId: jobId,
            type: 'avi:forge:app:event',
            schema: 'ari:cloud:ecosystem::forge/app-event-2',
            payload: validEvents,
            time: new Date().toISOString()
        };
        const response = await (0, queries_1.post)(queries_1.PUSH_PATH, pushRequest, this.apiClient);
        const result = { jobId };
        await (0, validators_1.validatePushAPIResponse)(pushRequest, response, result);
        return result;
    }
    getJob(jobId) {
        return new jobProgress_1.JobProgress(this.queueParams, jobId, this.apiClient);
    }
}
exports.Queue = Queue;
