// @forge/events/out/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateCancelJobAPIResponse = exports.validateGetStatsAPIResponse = exports.validatePushAPIResponse = exports.validateAPIResponse = exports.validateCancelJobRequest = exports.validateGetStatsPayload = exports.validatePushEvents = exports.validatePushSettings = exports.validateQueueKey = void 0;
const errors_1 = require("./errors");
const text_1 = require("./text");
const VALID_QUEUE_NAME_PATTERN = /^[a-zA-Z0-9-_]+$/;
const MAXIMUM_EVENTS = 50;
const MAXIMUM_PAYLOAD_SIZE_KB = 200;
const validateQueueKey = (queueName) => {
    if (!queueName || !VALID_QUEUE_NAME_PATTERN.test(queueName)) {
        throw new errors_1.InvalidQueueNameError(text_1.Text.error.invalidQueueName);
    }
};
exports.validateQueueKey = validateQueueKey;
const validatePushSettings = (event) => {
    if ((event.delayInSeconds && event.delayInSeconds > 900) || (event.delayInSeconds && event.delayInSeconds < 0)) {
        throw new errors_1.InvalidPushSettingsError(text_1.Text.error.invalidDelayInSecondsSetting);
    }
};
exports.validatePushSettings = validatePushSettings;
function validatePushEvents(arg) {
    const events = Array.isArray(arg) ? arg : [arg];
    if (events.length === 0) {
        throw new errors_1.NoEventsToPushError(text_1.Text.error.noEventsPushed);
    }
    if (events.length > MAXIMUM_EVENTS) {
        throw new errors_1.TooManyEventsError(text_1.Text.error.maxEventsAllowed(MAXIMUM_EVENTS));
    }
    for (const event of events) {
        if (typeof event !== 'object' || Array.isArray(event) || event === null) {
            throw new errors_1.InvalidPayloadError(text_1.Text.error.invalidEvent);
        }
        if (typeof event.body !== 'object' || Array.isArray(event.body) || event.body === null) {
            throw new errors_1.InvalidPayloadError(text_1.Text.error.invalidEventBody);
        }
        (0, exports.validatePushSettings)(event);
    }
    const payloadSizeKB = Buffer.byteLength(JSON.stringify(events)) / 1024;
    if (payloadSizeKB > MAXIMUM_PAYLOAD_SIZE_KB) {
        throw new errors_1.PayloadTooBigError(text_1.Text.error.maxPayloadAllowed(MAXIMUM_PAYLOAD_SIZE_KB));
    }
    return events;
}
exports.validatePushEvents = validatePushEvents;
const validateGetStatsPayload = (getStatsRequest) => {
    if (!getStatsRequest.jobId) {
        throw new errors_1.JobDoesNotExistError(text_1.Text.error.jobIdEmpty);
    }
    (0, exports.validateQueueKey)(getStatsRequest.queueName);
};
exports.validateGetStatsPayload = validateGetStatsPayload;
const validateCancelJobRequest = (cancelJobRequest) => {
    if (!cancelJobRequest.jobId) {
        throw new errors_1.JobDoesNotExistError(text_1.Text.error.jobIdEmpty);
    }
    (0, exports.validateQueueKey)(cancelJobRequest.queueName);
};
exports.validateCancelJobRequest = validateCancelJobRequest;
const validateAPIResponse = async (response, expectedSuccessStatus) => {
    if (response.status === 429) {
        throw new errors_1.RateLimitError(text_1.Text.error.rateLimitError);
    }
    if (response.status === 405) {
        throw new errors_1.InvocationLimitReachedError(text_1.Text.error.invocationLimitReachedError);
    }
    if (response.status != expectedSuccessStatus && response.status) {
        let internalServerError;
        try {
            const responseBody = await response.json();
            const errorMessage = responseBody.message ? `: ${responseBody.message}` : '';
            const errors = responseBody.errors ? `: ${responseBody.errors.join(', ')}` : '';
            internalServerError = new errors_1.InternalServerError(`${response.status} ${response.statusText}${errorMessage}${errors}`, responseBody.code, responseBody.details);
        }
        catch (ignore) {
            internalServerError = new errors_1.InternalServerError(`${response.status} ${response.statusText}`, response.status);
        }
        throw internalServerError;
    }
};
exports.validateAPIResponse = validateAPIResponse;
const validatePushAPIResponse = async (requestBody, response, result) => {
    if (response.status === 413) {
        const responseBody = await response.json();
        throw new errors_1.PayloadTooBigError(responseBody.errorMessage);
    }
    if (response.status === 202) {
        const responseBody = await response.json();
        const defaultErrorMessage = 'Failed to process some events.';
        const partialSuccessError = new errors_1.PartialSuccessError(defaultErrorMessage, result, []);
        if (responseBody.failedEvents && responseBody.failedEvents.length > 0) {
            partialSuccessError.message = `Failed to process ${responseBody.failedEvents.length} event(s).`;
            partialSuccessError.failedEvents = responseBody.failedEvents.map((failedEvent) => {
                return {
                    errorMessage: failedEvent.errorMessage,
                    payload: requestBody.payload[+failedEvent.index]
                };
            });
        }
        if (responseBody.errorMessage) {
            partialSuccessError.message =
                partialSuccessError.message !== defaultErrorMessage
                    ? `${partialSuccessError.message} ${responseBody.errorMessage}`
                    : responseBody.errorMessage;
        }
        throw partialSuccessError;
    }
    await (0, exports.validateAPIResponse)(response, 201);
};
exports.validatePushAPIResponse = validatePushAPIResponse;
const validateGetStatsAPIResponse = async (response, getStatsRequest) => {
    if (response.status === 404) {
        throw new errors_1.JobDoesNotExistError(text_1.Text.error.jobDoesNotExit(getStatsRequest.jobId, getStatsRequest.queueName));
    }
    await (0, exports.validateAPIResponse)(response, 200);
};
exports.validateGetStatsAPIResponse = validateGetStatsAPIResponse;
const validateCancelJobAPIResponse = async (response, cancelJobRequest) => {
    if (response.status === 404) {
        throw new errors_1.JobDoesNotExistError(text_1.Text.error.jobDoesNotExit(cancelJobRequest.jobId, cancelJobRequest.queueName));
    }
    await (0, exports.validateAPIResponse)(response, 204);
};
exports.validateCancelJobAPIResponse = validateCancelJobAPIResponse;
