// @atlaskit/page-layout/dist/esm/components/slots/right-panel.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
/** @jsx jsx */
import { useEffect } from 'react';
import { css, jsx } from '@emotion/react';
import { DEFAULT_RIGHT_PANEL_WIDTH, RIGHT_PANEL, RIGHT_PANEL_WIDTH, VAR_RIGHT_PANEL_WIDTH } from '../../common/constants';
import { getPageLayoutSlotSelector, resolveDimension } from '../../common/utils';
import { publishGridState, useSkipLink } from '../../controllers';
import SlotFocusRing from './internal/slot-focus-ring';
import SlotDimensions from './slot-dimensions';
var baseStyles = css({
  gridArea: RIGHT_PANEL
});
var fixedStyles = css({
  width: RIGHT_PANEL_WIDTH,
  position: 'fixed',
  top: 0,
  right: 0,
  bottom: 0
});

/**
 * __Right panel__
 *
 * Provides a slot for a right panel within the PageLayout.
 *
 * - [Examples](https://atlassian.design/components/page-layout/examples)
 * - [Code](https://atlassian.design/components/page-layout/code)
 */
var RightPanel = function RightPanel(props) {
  var children = props.children,
    isFixed = props.isFixed,
    _props$width = props.width,
    width = _props$width === void 0 ? DEFAULT_RIGHT_PANEL_WIDTH : _props$width,
    shouldPersistWidth = props.shouldPersistWidth,
    testId = props.testId,
    id = props.id,
    skipLinkTitle = props.skipLinkTitle;
  var rightPanelWidth = resolveDimension(VAR_RIGHT_PANEL_WIDTH, width, shouldPersistWidth);
  useEffect(function () {
    publishGridState(_defineProperty({}, VAR_RIGHT_PANEL_WIDTH, rightPanelWidth));
    return function () {
      publishGridState(_defineProperty({}, VAR_RIGHT_PANEL_WIDTH, 0));
    };
  }, [rightPanelWidth]);
  useSkipLink(id, skipLinkTitle);
  return jsx(SlotFocusRing, null, function (_ref) {
    var className = _ref.className;
    return jsx("div", _extends({
      css: [baseStyles, isFixed && fixedStyles],
      className: className,
      "data-testid": testId,
      id: id
    }, getPageLayoutSlotSelector('right-panel')), jsx(SlotDimensions, {
      variableName: VAR_RIGHT_PANEL_WIDTH,
      value: rightPanelWidth
    }), children);
  });
};
export default RightPanel;
