// @atlaskit/atlassian-navigation/dist/cjs/components/Create/skeleton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreateSkeleton = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _constants = require("../../common/constants");
var _theme = require("../../theme");
var _skeleton = require("../IconButton/skeleton");
/** @jsx jsx */

var buttonHeight = "var(--ds-space-400, 32px)";
var skeletonStyles = (0, _react2.css)({
  width: 68,
  height: buttonHeight,
  borderRadius: "var(--ds-border-radius, 3px)",
  marginInlineStart: "var(--ds-space-150, 12px)",
  opacity: 0.15
});
var mobileStyles = (0, _react2.css)((0, _defineProperty2.default)({}, "@media (max-width: ".concat(_constants.CREATE_BREAKPOINT - 1, "px)"), {
  display: 'none !important'
}));

// Not exported to consumers, only used in NavigationSkeleton
// eslint-disable-next-line @repo/internal/react/require-jsdoc
var CreateSkeleton = exports.CreateSkeleton = function CreateSkeleton() {
  var theme = (0, _theme.useTheme)();
  return (0, _react2.jsx)(_react.Fragment, null, (0, _react2.jsx)("div", {
    style: theme.mode.skeleton,
    css: [skeletonStyles, mobileStyles]
  }), (0, _react2.jsx)(_skeleton.IconButtonSkeleton, {
    css: mobileStyles,
    size: 26
  }));
};
