// @atlaskit/avatar/dist/es2019/internal/presence-wrapper.js
/* presence-wrapper.tsx generated by @compiled/babel-plugin v0.39.1 */
import "./presence-wrapper.compiled.css";
import * as React from 'react';
import { ax, ix } from "@compiled/react/runtime";
import AvatarPresence from '../presence';
const styles = {
  root: "_kqswstnw _lcxvglyw"
};
const iconSizeMap = {
  small: "_4t3i1crf _1bsb1crf",
  medium: "_4t3idlk8 _1bsbdlk8",
  large: "_4t3io7ao _1bsbo7ao",
  xlarge: "_4t3if6fq _1bsbf6fq"
};
const circleIconOffsetMap = {
  small: "_94n5idpf _1xi2idpf",
  medium: "_94n5idpf _1xi2idpf",
  large: "_94n5t94y _1xi2t94y",
  xlarge: "_94n51v6z _1xi21v6z"
};
const squareIconOffsetMap = {
  root: "_94n5myb0 _1xi2myb0"
};
const hexagonIconOffsetMap = {
  small: "_94n51n1a _1xi21n1a",
  medium: "_94n51n1a _1xi21n1a",
  large: "_94n51y44 _1xi2myb0",
  xlarge: "_94n5l52n _1xi21lpd"
};

/**
 * __Presence wrapper__
 *
 * A presence wrapper is used internally to position presence ontop of the avatar.
 */
const PresenceWrapper = ({
  size,
  appearance,
  children,
  borderColor,
  presence,
  testId
}) => {
  return /*#__PURE__*/React.createElement("span", {
    "aria-hidden": "true",
    "data-testid": testId && `${testId}--presence`,
    className: ax([styles.root, iconSizeMap[size], circleIconOffsetMap[size], appearance === 'square' && squareIconOffsetMap.root, appearance === 'hexagon' && hexagonIconOffsetMap[size]])
  }, /*#__PURE__*/React.createElement(AvatarPresence, {
    borderColor: borderColor,
    presence: !children ? presence : undefined
  }, children));
};
export default PresenceWrapper;
